---
title: "Gene Enrichment Analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll identify enriched genes for mRNA regions, exons, and introns that overlap with DMLs. To do this, I'll use Uniprot accession codes with a [custom-built Gene Enrichment tool](https://meta.yeastrc.org/compgo_yaamini_oyster/pages/goAnalysisForm.jsp).


## Session information

```{r}
sessionInfo()
```

## Obtain GOterms

Before I do any sort of gene enrichment analysis, I need to get Uniprot codes for all genes. To do this, I will use `blastx`.

### Make a Uniprot Database

The Uniprot database was downloaded from this link as a .fasta file: http://www.uniprot.org/uniprot/?query=&fil=reviewed%3Ayes&columns=id%2Centry%20name%2Creviewed%2Cprotein%20names%2Cgenes%2Corganism%2Clength%2Cgo(biological%20process)%2Cgo-id%2Ccomment(PATHWAY)%2Cdatabase(UniPathway)%2Cdatabase(CDD)%2Cdatabase(Pfam.

```{bash}
head uniprot-filtered-reviewed.fasta
```

```{bash}
/Users/Shared/Apps/ncbi-blast-2.2.29+/bin/makeblastdb -help #Get the help menu
```

The first step involves taking the uniprot fasta file and making a new database. To do this, I will use the following code:

1. Path to `makeblastdb` to create the database needed
2. `in` to select the input file for the database
3. `dbtype` specifies the type of file which in our case, is a protein file

```{bash}
/Users/Shared/Apps/ncbi-blast-2.2.29+/bin/makeblastdb \
-in uniprot-filtered-reviewed.fasta \
-dbtype 'prot'
```

```{bash}
/Users/Shared/Apps/ncbi-blast-2.2.29+/bin/blastx -help
```

Now I can perform a `blastx` to compare my query, the *C. virginica* genome, to our protein database.

1. Path to `blastx`
2. `-query` provides the file we want to blast
3.`-db` specifies database created in the previous step
4. `-outfmt` specifies the type of output file. I will use 6, a tabular file

I ran this section in the Terminal using the following code:

![code](https://raw.githubusercontent.com/RobertsLab/project-virginica-oa/master/images/Lab-Notebook/2018-06-19-blastx-Code-in-Terminal.png)

Once my `blastx` completed, I had the genome matched with Uniprot codes. It can be [found on OWL](http://owl.fish.washington.edu/spartina/Virginica-MBD/2018-07-25-Gene-Enrichment/uniprot-filtered-reviewed.fasta).

## mRNA overlaps

### Merge Uniprot codes with overlaps

The first thing I need to do is merge the txt file of DML-mRNA overlaps with the *C. virginica* genome-Uniprot codes. I should probably do this in bash, as opposed to R.

```{bash}
pwd
```

```{bash}
ls
```

```{bash}
head 2018-06-11-DML-mRNA.txt
```


### `compGO`

## Exon overlaps

### Merge Uniprot codes with overlaps

### `compGO`

## Intron overlaps

### Merge Uniprot codes with overlaps

### `compGO`