---
title: "MethylKit Parameter Testing"
author: "Yaamini Venkataraman"
date: "10/11/2018"
output: html_document
---

In this file, I'll try 1x, 3x, and 5x coverage with subset data from with [new `-score_min` parameter](https://yaaminiv.github.io/DML-Analysis-Part10/) to see how it affects clustering.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo()
```

# Install packages

```{r}
install.packages("devtools") #Install the devtools package
library(devtools) #Load devtools

source("https://bioconductor.org/biocLite.R") #Source package from bioconductor
biocLite("methylKit") #Install methylkit
install_github("al2na/methylKit", build_vignettes = FALSE, repos=BiocInstaller::biocinstallRepos(), dependencies = TRUE) #Install more methylKit options
library(methylKit) #Load methylkit
```

# Process methylation data

I need to access the files on [gannett](http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/). I previously downloaded these files using `wget` and checked file quality with `shasum` in [this notebook](https://github.com/RobertsLab/project-virginica-oa/blob/master/notebooks/2018-10-03-Bismark-Parameter-Testing.ipynb).

```{r}
getwd()
```

```{r}
analysisFiles <- list("../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_dedup.sorted.bam",
                      "../2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_dedup.sorted.bam") #Put all .bam files into a list for analysis.
```

```{r}
sample.IDs <- list("1", "2", "3", "4", "5", "6", "7", "8", "9", "10") #Create list of sample IDs
treatmentSpecification <- c(rep(0, times = 5), rep(1, times = 5)) #Specify which treatment the samples were from. 1 is the treatment (high pCO2) and 0 is the control (ambient pCO2)
```

# Try different coverage metrics

I will use `processBismarkAln` to set different coverage metrics in the `mincov` argument. I'll text 1x, 3x, and 5x, coverage.

```{r}
processedFilesCov1 <- processBismarkAln(location = analysisFiles, sample.id = sample.IDs, assembly = "v3", read.context = "CpG", mincov = 1, treatment = treatmentSpecification) #Process files for CpG meetehylation using 1x coverage. First 5 files were from ambient conditions, and the second from high pCO2 conditions.
```

```{r}
processedFilesCov3 <- processBismarkAln(location = analysisFiles, sample.id = sample.IDs, assembly = "v3", read.context = "CpG", mincov = 3, treatment = treatmentSpecification) #Process files for CpG meetehylation using 3x coverage. First 5 files were from ambient conditions, and the second from high pCO2 conditions.
```

```{r}
processedFilesCov5 <- processBismarkAln(location = analysisFiles, sample.id = sample.IDs, assembly = "v3", read.context = "CpG", mincov = 5, treatment = treatmentSpecification) #Process files for CpG meetehylation using 1x coverage. First 5 files were from ambient conditions, and the second from high pCO2 conditions.
```

Working through 5x coverage with a subset just didn't give me enough alignments, so I'll continue to cmopare 1x and 3x coverage.

# Analyze methylation data

## Obtain methylation and coverage plots

```{r}
nFiles <- length(sample.IDs) #Count number of samples
fileName <- data.frame("nameBase" = rep("2018-10-11-Percent-CpG-Methylation", times = nFiles),
                       "nameBase2" = rep("2018-10-11-Percent-CpG-Coverage", times = nFiles),
                       "sample.ID" = 1:10) #Create new dataframe for filenames
head(fileName) #Confirm dataframe creation
```

```{r}
fileName$actualFileName <- paste(fileName$nameBase, "-1xCoverage", "-Sample", fileName$sample.ID, ".jpeg", sep = "") #Create a new column for the full filename for 1x coverage + specific sample's percent CpG methylation plot
fileName$actualFileName2 <- paste(fileName$nameBase2, "-1xCoverage", "-Sample", fileName$sample.ID, ".jpeg", sep = "") #Create a new column for the full filename for 1x coverage + specific sample's percent CpG coverage plot
fileName$actualFileName3 <- paste(fileName$nameBase, "-3xCoverage", "-Sample", fileName$sample.ID, ".jpeg", sep = "") #Create a new column for the full filename for 3x coverage + specific sample's percent CpG methylation plot
fileName$actualFileName4 <- paste(fileName$nameBase2, "-3xCoverage", "-Sample", fileName$sample.ID, ".jpeg", sep = "") #Create a new column for the full filename for 3x coverage + specific sample's percent CpG coverage plot
head(fileName) #Confirm column creation
```

```{r}
for(i in 1:nFiles) { #For each data file
  jpeg(filename = fileName$actualFileName[i], height = 1000, width = 1000) #Save file with designated name
  getMethylationStats(processedFilesCov1[[i]], plot = TRUE, both.strands = FALSE) #Get %CpG methylation information
  dev.off() #Turn off plotting device
} #Plot and save %CpG methylation information
```


```{r}
for(i in 1:nFiles) { #For each data file
  jpeg(filename = fileName$actualFileName2[i], height = 1000, width = 1000) #Save file with designated name
  getCoverageStats(processedFilesCov1[[i]], plot = TRUE, both.strands = FALSE) #Get CpG coverage information
  dev.off() #Turn off plotting device
} #Plot and save CpG coverage information
```

```{r}
for(i in 1:nFiles) { #For each data file
  jpeg(filename = fileName$actualFileName3[i], height = 1000, width = 1000) #Save file with designated name
  getMethylationStats(processedFilesCov3[[i]], plot = TRUE, both.strands = FALSE) #Get %CpG methylation information
  dev.off() #Turn off plotting device
} #Plot and save %CpG methylation information
```

```{r}
for(i in 1:nFiles) { #For each data file
  jpeg(filename = fileName$actualFileName4[i], height = 1000, width = 1000) #Save file with designated name
  getCoverageStats(processedFilesCov3[[i]], plot = TRUE, both.strands = FALSE) #Get CpG coverage information
  dev.off() #Turn off plotting device
} #Plot and save CpG coverage information
```

## Obtain correlation plots