---
title: "Gene Enrichment Analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll use `topGo` to identify enriched genes for mRNA regions, exons, and introns that overlap with DMLs. I'm following instructions from [this manual](https://bioconductor.riken.jp/packages/3.3/bioc/vignettes/topGO/inst/doc/topGO.pdf) and [this blog entry](http://avrilomics.blogspot.com/2015/07/using-topgo-to-test-for-go-term.html).

## Session information

```{r}
sessionInfo()
```

## Install packages

```{r, echo = FALSE}
source("http://bioconductor.org/biocLite.R") #Install source package

biocLite("GO.db") #Install GO.db
biocLite("topGO") #Install topGO
biocLite("GOstats") #Install GOstats
biocLite("org.Hs.eg.db") #Install org.Hs.eg.db

#Load all installed packages
library("GO.db")
library("topGO")
library("GOstats")
library("org.Hs.eg.db")
```

## Obtain GOterms

Before I do any sort of gene enrichment analysis, I need to get usable Gene IDs and GOterms for all genes. To do this, I will use `blastx`.

### Make a Uniprot Database

The Uniprot database was downloaded from this link as a .fasta file: http://www.uniprot.org/uniprot/?query=&fil=reviewed%3Ayes&columns=id%2Centry%20name%2Creviewed%2Cprotein%20names%2Cgenes%2Corganism%2Clength%2Cgo(biological%20process)%2Cgo-id%2Ccomment(PATHWAY)%2Cdatabase(UniPathway)%2Cdatabase(CDD)%2Cdatabase(Pfam.

```{bash}
head uniprot-filtered-reviewed.fasta
```

```{bash}
/Users/Shared/Apps/ncbi-blast-2.2.29+/bin/makeblastdb -help #Get the help menu
```

The first step involves taking the uniprot fasta file and making a new database. To do this, I will use the following code:

1. Path to `makeblastdb` to create the database needed
2. `in` to select the input file for the database
3. `dbtype` specifies the type of file which in our case, is a protein file

```{bash}
/Users/Shared/Apps/ncbi-blast-2.2.29+/bin/makeblastdb \
-in uniprot-filtered-reviewed.fasta \
-dbtype 'prot'
```

```{bash}
/Users/Shared/Apps/ncbi-blast-2.2.29+/bin/blastx -help
```

Now I can perform a `blastx` to compare my query, the *C. virginica* genome, to our protein database.

1. Path to `blastx`
2. `-query` provides the file we want to blast
3.`-db` specifies database created in the previous step
4. `-outfmt` specifies the type of output file. I will use 6, a tabular file

I ran this section in the Terminal using the following code:

![code](https://raw.githubusercontent.com/RobertsLab/project-virginica-oa/master/images/Lab-Notebook/2018-06-19-blastx-Code-in-Terminal.png)

## mRNA overlaps

### Load and reformat data

```{r}
mRNAandDMLoverlaps <- read.delim(file = "2018-06-14-DML-mRNA-unfolded.txt", header = FALSE) #Read in data
head(mRNAandDMLoverlaps) #Confirm import
```

For `topGO`, the file of gene-to-GO mappings needs to be a tab-delimited file with the first column as the gene name, and the second column as a list of GO terms separated by a comma and space. I need to remove any extraneous information and add GOterm information.

```{r}
genesAndGOTerms <- data.frame("geneID" = mRNAandDMLoverlaps$V17,
                              "geneName" = mRNAandDMLoverlaps$V20) #Create new dataframe for genes and GOterms. Gene IDs are **NOT** Entrez gene IDs from NCBI.
head(genesAndGOTerms) #Confirm dataframe creation
```

Notice how there are some geneNames called "unclassified transcription discrepancy." I'm not sure what this means, but there is an associated geneID. I should be able to proceed with the analysis.

### DAVID

### `topGO`

## Exon overlaps

### Load and reformat data

### DAVID

### `topGO`

## Intron overlaps

### Load and reformat data

### DAVID

### `topGO`