{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bismark\n",
    "\n",
    "In this notebook I'll complete the five steps of Bismark on a subset of my data. The steps from the [user guide](https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#i-bismark-genome-preparation) are as follows:\n",
    "\n",
    "1. Genome Preparation\n",
    "2. Alignment\n",
    "3. Methlation Extractor\n",
    "4. HTML Processing Report\n",
    "5. Summary Report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Genome Preparation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1a. Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/yaamini/Documents/project-virginica-oa/notebooks'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses\n"
     ]
    }
   ],
   "source": [
    "cd ../analyses/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-04-27-Bismark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-01-23-MBDSeq-Labwork\u001b[m\u001b[m/           \u001b[34m2018-04-27-Bismark\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-26-Gonad-Methylation-FastQC\u001b[m\u001b[m/ README.md\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark\n"
     ]
    }
   ],
   "source": [
    "cd 2018-04-27-Bismark/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1b. Download *C. virginica* genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-04-27-Bismark-Inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-04-27-Bismark-Inputs\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  202M  100  202M    0     0  1082k      0  0:03:11  0:03:11 --:--:-- 1110k\n"
     ]
    }
   ],
   "source": [
    "!curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/GCF_002022765.2_C_virginica-3.0_genomic.fna.gz \\\n",
    "    > 2018-04-27-Bismark-Inputs/C_virginica-3.0_genomic.fna.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I unzipped the file in Finder because I couldn't get gunzip to work. Will need to sort that out later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C_virginica-3.0_genomic.fna     C_virginica-3.0_genomic.fna.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls 2018-04-27-Bismark-Inputs/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The file from NCBI is a .fna, which means FASTA nucleic acid. However, Bismark will not recognize an .fna, and needs either an .fa or .fasta. I will need to change the file extension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "! cp 2018-04-27-Bismark-Inputs/C_virginica-3.0_genomic.fna 2018-04-27-Bismark-Inputs/C_virginica-3.0_genomic.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C_virginica-3.0_genomic.fa      C_virginica-3.0_genomic.fna.gz\r\n",
      "C_virginica-3.0_genomic.fna\r\n"
     ]
    }
   ],
   "source": [
    "ls 2018-04-27-Bismark-Inputs/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Success! Now I can actually run `bismark`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run `bismark_genome_preparation`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "This script is supposed to convert a specified reference genome into two different bisulfite\r\n",
      "converted versions and index them for alignments with Bowtie 2 (default), or Bowtie 1. The first\r\n",
      "bisulfite genome will have all Cs converted to Ts (C->T), and the other one will have all Gs\r\n",
      "converted to As (G->A). Both bisulfite genomes will be stored in subfolders within the reference\r\n",
      "genome folder. Once the bisulfite conversion has been completed the program will fork and launch\r\n",
      "two simultaneous instances of the Bowtie 1 or 2 indexer (bowtie-build or bowtie2-build). Be aware\r\n",
      "that the indexing process can take up to several hours; this will mainly depend on genome size\r\n",
      "and system resources.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "The following is a brief description of command line options and arguments to control the\r\n",
      "Bismark Genome Preparation:\r\n",
      "\r\n",
      "\r\n",
      "USAGE: bismark_genome_preparation [options] <arguments>\r\n",
      "\r\n",
      "\r\n",
      "OPTIONS:\r\n",
      "\r\n",
      "--help/--man             Displays this help filea and exits.\r\n",
      "\r\n",
      "--version                Displays version information and exits.\r\n",
      "\r\n",
      "--verbose                Print verbose output for more details or debugging.\r\n",
      "\r\n",
      "--path_to_bowtie </../>  The full path to the Bowtie 1 or Bowtie 2 installation on your system\r\n",
      "                         (depending on which aligner/indexer you intend to use). Unless this path\r\n",
      "                         is specified it is assumed that Bowtie is in the PATH.\r\n",
      "\r\n",
      "--bowtie2                This will create bisulfite indexes for Bowtie 2. (Default: ON).\r\n",
      "\r\n",
      "--bowtie1                This will create bisulfite indexes for Bowtie 1. (Default: OFF).\r\n",
      "\r\n",
      "--single_fasta           Instruct the Bismark Indexer to write the converted genomes into\r\n",
      "                         single-entry FastA files instead of making one multi-FastA file (MFA)\r\n",
      "                         per chromosome. This might be useful if individual bisulfite converted\r\n",
      "                         chromosomes are needed (e.g. for debugging), however it can cause a\r\n",
      "                         problem with indexing if the number of chromosomes is vast (this is likely\r\n",
      "                         to be in the range of several thousand files; the operating system can\r\n",
      "                         only handle lists up to a certain length, and some newly assembled\r\n",
      "                         genomes may contain 20000-500000 contigs of scaffold files which do exceed\r\n",
      "                         this list length limit).\r\n",
      "\r\n",
      "--genomic_composition    Calculate and extract the genomic sequence composition for mono and di-nucleotides\r\n",
      "                         and write the genomic composition table 'genomic_nucleotide_frequencies.txt' to the\r\n",
      "                         genome folder. This may be useful later on when using bam2nuc or the Bismark option\r\n",
      "                         --nucleotide_coverage.\r\n",
      "\r\n",
      "\r\n",
      "ARGUMENTS:\r\n",
      "\r\n",
      "<path_to_genome_folder>  The path to the folder containing the genome to be bisulfite converted.\r\n",
      "                         The Bismark Genome Preparation expects one or more fastA files in the folder\r\n",
      "                         (with the file extension: .fa or .fasta). Specifying this path is mandatory.\r\n",
      "\r\n",
      "\r\n",
      "This script was last modified on 15 August 2017.\r\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark_genome_preparation -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here are the inputs for my command:\n",
    "\n",
    "1. Path to `bismark_genome_preparation`\n",
    "2. --path_to_bowtie + the path to the folder with bowtie2, not the application itself\n",
    "3. --verbose: Prints detailed output for debugging\n",
    "4. Path to *C virginica* genome folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to genome folder specified as: 2018-04-27-Bismark-Inputs/\n",
      "Aligner to be used: Bowtie 2 (default)\n",
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bismark Genome Preparation - Step I: Preparing folders\n",
      "\n",
      "Path to Bowtie 2 specified: /Users/Shared/Apps/anaconda/bin/\n",
      "Bisulfite Genome Indexer version v0.19.0 (last modified 07 November 2016)\n",
      "Created Bisulfite Genome folder /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/\n",
      "Created Bisulfite Genome folder /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/\n",
      "Created Bisulfite Genome folder /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step II: Bisulfite converting reference genome\n",
      "\n",
      "conversions performed:\n",
      "chromosome\tC->T\tG->A\n",
      "NC_035780.1\t11492316\t11528162\n",
      "NC_035781.1\t10879823\t10858285\n",
      "NC_035782.1\t13475475\t13496235\n",
      "NC_035783.1\t10484268\t10482193\n",
      "NC_035784.1\t17306783\t17253434\n",
      "NC_035785.1\t8878985\t8876470\n",
      "NC_035786.1\t9961761\t9989843\n",
      "NC_035787.1\t13145205\t13164212\n",
      "NC_035788.1\t17925605\t17932896\n",
      "NC_035789.1\t5676071\t5682065\n",
      "NC_007175.2\t2827\t3882\n",
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t119229119\n",
      "G->A:\t119267677\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "\n",
      "Preparing indexing of CT converted genome in /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/\n",
      "Parent process: Starting to index C->T converted genome with the following command:\n",
      "\n",
      "/Users/Shared/Apps/anaconda/bin/bowtie2-build -f genome_mfa.CT_conversion.fa BS_CT\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "Preparing indexing of GA converted genome in /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/\n",
      "Child process: Starting to index G->A converted genome with the following command:\n",
      "\n",
      "/Users/Shared/Apps/anaconda/bin/bowtie2-build -f genome_mfa.GA_conversion.fa BS_GA\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:46\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:47\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:07\n",
      "bmax according to bmaxDivN setting: 171168832\n",
      "Using parameters --bmax 128376624 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 128376624 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 171168832\n",
      "Using parameters --bmax 128376624 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 128376624 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:21\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:21\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:09\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 6.84675e+08 (target: 128376623)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:09\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 6.84675e+08 (target: 128376623)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 684675328 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block of length 684675328 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:09:47\n",
      "Returning block of 684675329 for bucket 1\n",
      "  Sorting block time: 00:11:39\n",
      "Returning block of 684675329 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 342402452\n",
      "fchr[G]: 461631571\n",
      "fchr[T]: 461631571\n",
      "fchr[$]: 684675328\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 232427948 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 171168840 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 684675328\n",
      "    bwtLen: 684675329\n",
      "    sz: 171168832\n",
      "    bwtSz: 171168833\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 42792209\n",
      "    offsSz: 171168836\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 3566018\n",
      "    numLines: 3566018\n",
      "    ebwtTotLen: 228225152\n",
      "    ebwtTotSz: 228225152\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:13:17\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:07\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 171168832\n",
      "Using parameters --bmax 128376624 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 128376624 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:20\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 6.84675e+08 (target: 128376623)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 684675328 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 223134775\n",
      "fchr[G]: 223134775\n",
      "fchr[T]: 342402452\n",
      "fchr[$]: 684675328\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 232427948 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 171168840 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 684675328\n",
      "    bwtLen: 684675329\n",
      "    sz: 171168832\n",
      "    bwtSz: 171168833\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 42792209\n",
      "    offsSz: 171168836\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 3566018\n",
      "    numLines: 3566018\n",
      "    ebwtTotLen: 228225152\n",
      "    ebwtTotSz: 228225152\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:15:12\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:07\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 171168832\n",
      "Using parameters --bmax 128376624 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 128376624 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:20\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 6.84675e+08 (target: 128376623)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 684675328 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:09:37\n",
      "Returning block of 684675329 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 342402452\n",
      "fchr[G]: 461631571\n",
      "fchr[T]: 461631571\n",
      "fchr[$]: 684675328\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 232427948 bytes to primary EBWT file: BS_GA.rev.1.bt2\n",
      "Wrote 171168840 bytes to secondary EBWT file: BS_GA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 684675328\n",
      "    bwtLen: 684675329\n",
      "    sz: 171168832\n",
      "    bwtSz: 171168833\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 42792209\n",
      "    offsSz: 171168836\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 3566018\n",
      "    numLines: 3566018\n",
      "    ebwtTotLen: 228225152\n",
      "    ebwtTotSz: 228225152\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:12:19\n",
      "  Sorting block time: 00:11:26\n",
      "Returning block of 684675329 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 223134775\n",
      "fchr[G]: 223134775\n",
      "fchr[T]: 342402452\n",
      "fchr[$]: 684675328\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 232427948 bytes to primary EBWT file: BS_CT.rev.1.bt2\n",
      "Wrote 171168840 bytes to secondary EBWT file: BS_CT.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 684675328\n",
      "    bwtLen: 684675329\n",
      "    sz: 171168832\n",
      "    bwtSz: 171168833\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 42792209\n",
      "    offsSz: 171168836\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 3566018\n",
      "    numLines: 3566018\n",
      "    ebwtTotLen: 228225152\n",
      "    ebwtTotSz: 228225152\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:14:05\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark_genome_preparation \\\n",
    "--path_to_bowtie /Users/Shared/Apps/anaconda/bin/ \\\n",
    "--verbose \\\n",
    "2018-04-27-Bismark-Inputs/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "     This program is free software: you can redistribute it and/or modify\r\n",
      "     it under the terms of the GNU General Public License as published by\r\n",
      "     the Free Software Foundation, either version 3 of the License, or\r\n",
      "     (at your option) any later version.\r\n",
      "\r\n",
      "     This program is distributed in the hope that it will be useful,\r\n",
      "     but WITHOUT ANY WARRANTY; without even the implied warranty of\r\n",
      "     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\r\n",
      "     GNU General Public License for more details.\r\n",
      "     You should have received a copy of the GNU General Public License\r\n",
      "     along with this program.  If not, see <http://www.gnu.org/licenses/>.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "\r\n",
      "The following is a brief description of command line options and arguments to control the Bismark\r\n",
      "bisulfite mapper and methylation caller. Bismark takes in FastA or FastQ files and aligns the\r\n",
      "reads to a specified bisulfite genome. Sequence reads are transformed into a bisulfite converted forward strand\r\n",
      "version (C->T conversion) or into a bisulfite treated reverse strand (G->A conversion of the forward strand).\r\n",
      "Each of these reads are then aligned to bisulfite treated forward strand index of a reference genome\r\n",
      "(C->T converted) and a bisulfite treated reverse strand index of the genome (G->A conversion of the\r\n",
      "forward strand, by doing this alignments will produce the same positions). These 4 instances of Bowtie (1 or 2)\r\n",
      "are run in parallel. The sequence file(s) are then read in again sequence by sequence to pull out the original\r\n",
      "sequence from the genome and determine if there were any protected C's present or not.\r\n",
      "\r\n",
      "As of version 0.7.0 Bismark will only run 2 alignment threads for OT and OB in parallel, the 4 strand mode can be\r\n",
      "re-enabled by using --non_directional.\r\n",
      "\r\n",
      "The final output of Bismark is in SAM format by default. For Bowtie 1 one can alos choose to report the old\r\n",
      "'vanilla' output format, which is a single tab delimited file with all sequences that have a unique best\r\n",
      "alignment to any of the 4 possible strands of a bisulfite PCR product. Both formats are described in more detail below.\r\n",
      "\r\n",
      "\r\n",
      "USAGE: bismark [options] <genome_folder> {-1 <mates1> -2 <mates2> | <singles>}\r\n",
      "\r\n",
      "\r\n",
      "ARGUMENTS:\r\n",
      "\r\n",
      "<genome_folder>          The path to the folder containing the unmodified reference genome\r\n",
      "                         as well as the subfolders created by the Bismark_Genome_Preparation\r\n",
      "                         script (/Bisulfite_Genome/CT_conversion/ and /Bisulfite_Genome/GA_conversion/).\r\n",
      "                         Bismark expects one or more fastA files in this folder (file extension: .fa\r\n",
      "                         or .fasta). The path can be relative or absolute. The path may also be set as\r\n",
      "                         '--genome_folder /path/to/genome/folder/'.\r\n",
      "\r\n",
      "-1 <mates1>              Comma-separated list of files containing the #1 mates (filename usually includes\r\n",
      "                         \"_1\"), e.g. flyA_1.fq,flyB_1.fq). Sequences specified with this option must\r\n",
      "                         correspond file-for-file and read-for-read with those specified in <mates2>.\r\n",
      "                         Reads may be a mix of different lengths. Bismark will produce one mapping result\r\n",
      "                         and one report file per paired-end input file pair.\r\n",
      "\r\n",
      "-2 <mates2>              Comma-separated list of files containing the #2 mates (filename usually includes\r\n",
      "                         \"_2\"), e.g. flyA_1.fq,flyB_1.fq). Sequences specified with this option must\r\n",
      "                         correspond file-for-file and read-for-read with those specified in <mates1>.\r\n",
      "                         Reads may be a mix of different lengths.\r\n",
      "\r\n",
      "<singles>                A comma- or space-separated list of files containing the reads to be aligned (e.g.\r\n",
      "                         lane1.fq,lane2.fq lane3.fq). Reads may be a mix of different lengths. Bismark will\r\n",
      "                         produce one mapping result and one report file per input file.\r\n",
      "\r\n",
      "\r\n",
      "OPTIONS:\r\n",
      "\r\n",
      "\r\n",
      "Input:\r\n",
      "\r\n",
      "--se/--single_end <list> Sets single-end mapping mode explicitly giving a list of file names as <list>.\r\n",
      "                         The filenames may be provided as a comma [,] or colon [:] separated list.\r\n",
      "\r\n",
      "-q/--fastq               The query input files (specified as <mate1>,<mate2> or <singles> are FASTQ\r\n",
      "                         files (usually having extension .fg or .fastq). This is the default. See also\r\n",
      "                         --solexa-quals.\r\n",
      "\r\n",
      "-f/--fasta               The query input files (specified as <mate1>,<mate2> or <singles> are FASTA\r\n",
      "                         files (usually having extensions .fa, .mfa, .fna or similar). All quality values\r\n",
      "                         are assumed to be 40 on the Phred scale. FASTA files are expected to contain both\r\n",
      "                         the read name and the sequence on a single line (and not spread over several lines).\r\n",
      "\r\n",
      "-s/--skip <int>          Skip (i.e. do not align) the first <int> reads or read pairs from the input.\r\n",
      "\r\n",
      "-u/--upto <int>          Only aligns the first <int> reads or read pairs from the input. Default: no limit.\r\n",
      "\r\n",
      "--phred33-quals          FASTQ qualities are ASCII chars equal to the Phred quality plus 33. Default: on.\r\n",
      "\r\n",
      "--phred64-quals          FASTQ qualities are ASCII chars equal to the Phred quality plus 64. Default: off.\r\n",
      "\r\n",
      "--solexa-quals           Convert FASTQ qualities from solexa-scaled (which can be negative) to phred-scaled\r\n",
      "                         (which can't). The formula for conversion is:\r\n",
      "                         phred-qual = 10 * log(1 + 10 ** (solexa-qual/10.0)) / log(10). Used with -q. This\r\n",
      "                         is usually the right option for use with (unconverted) reads emitted by the GA\r\n",
      "                         Pipeline versions prior to 1.3. Works only for Bowtie 1. Default: off.\r\n",
      "\r\n",
      "--solexa1.3-quals        Same as --phred64-quals. This is usually the right option for use with (unconverted)\r\n",
      "                         reads emitted by GA Pipeline version 1.3 or later. Default: off.\r\n",
      "\r\n",
      "--path_to_bowtie         The full path </../../> to the Bowtie (1 or 2) installation on your system. If not\r\n",
      "                         specified it is assumed that Bowtie (1 or 2) is in the PATH.\r\n",
      "\r\n",
      "\r\n",
      "Alignment:\r\n",
      "\r\n",
      "-n/--seedmms <int>       The maximum number of mismatches permitted in the \"seed\", i.e. the first L base pairs\r\n",
      "                         of the read (where L is set with -l/--seedlen). This may be 0, 1, 2 or 3 and the\r\n",
      "                         default is 1. This option is only available for Bowtie 1 (for Bowtie 2 see -N).\r\n",
      "\r\n",
      "-l/--seedlen             The \"seed length\"; i.e., the number of bases of the high quality end of the read to\r\n",
      "                         which the -n ceiling applies. The default is 28. Bowtie (and thus Bismark) is faster for\r\n",
      "                         larger values of -l. This option is only available for Bowtie 1 (for Bowtie 2 see -L).\r\n",
      "\r\n",
      "-e/--maqerr <int>        Maximum permitted total of quality values at all mismatched read positions throughout\r\n",
      "                         the entire alignment, not just in the \"seed\". The default is 70. Like Maq, bowtie rounds\r\n",
      "                         quality values to the nearest 10 and saturates at 30. This value is not relevant for\r\n",
      "                         Bowtie 2.\r\n",
      "\r\n",
      "--chunkmbs <int>         The number of megabytes of memory a given thread is given to store path descriptors in\r\n",
      "                         --best mode. Best-first search must keep track of many paths at once to ensure it is\r\n",
      "                         always extending the path with the lowest cumulative cost. Bowtie tries to minimize the\r\n",
      "                         memory impact of the descriptors, but they can still grow very large in some cases. If\r\n",
      "                         you receive an error message saying that chunk memory has been exhausted in --best mode,\r\n",
      "                         try adjusting this parameter up to dedicate more memory to the descriptors. This value\r\n",
      "                         is not relevant for Bowtie 2. Default: 512.\r\n",
      "\r\n",
      "-I/--minins <int>        The minimum insert size for valid paired-end alignments. E.g. if -I 60 is specified and\r\n",
      "                         a paired-end alignment consists of two 20-bp alignments in the appropriate orientation\r\n",
      "                         with a 20-bp gap between them, that alignment is considered valid (as long as -X is also\r\n",
      "                         satisfied). A 19-bp gap would not be valid in that case. Default: 0.\r\n",
      "\r\n",
      "-X/--maxins <int>        The maximum insert size for valid paired-end alignments. E.g. if -X 100 is specified and\r\n",
      "                         a paired-end alignment consists of two 20-bp alignments in the proper orientation with a\r\n",
      "                         60-bp gap between them, that alignment is considered valid (as long as -I is also satisfied).\r\n",
      "                         A 61-bp gap would not be valid in that case. Default: 500.\r\n",
      "\r\n",
      "--parallel <int>         (May also be --multicore <int>) Sets the number of parallel instances of Bismark to be run concurrently.\r\n",
      "                         This forks the Bismark alignment step very early on so that each individual Spawn of Bismark processes\r\n",
      "                         only every n-th sequence (n being set by --parallel). Once all processes have completed,\r\n",
      "                         the individual BAM files, mapping reports, unmapped or ambiguous FastQ files are merged\r\n",
      "                         into single files in very much the same way as they would have been generated running Bismark\r\n",
      "                         conventionally with only a single instance.\r\n",
      "\r\n",
      "                         If system resources are plentiful this is a viable option to speed up the alignment process\r\n",
      "                         (we observed a near linear speed increase for up to --parallel 8 tested). However, please note\r\n",
      "                         that a typical Bismark run will use several cores already (Bismark itself, 2 or 4 threads of\r\n",
      "                         Bowtie/Bowtie2, Samtools, gzip etc...) and ~10-16GB of memory depending on the choice of aligner\r\n",
      "                         and genome. WARNING: Bismark Parallel (BP?) is resource hungry! Each value of --parallel specified\r\n",
      "                         will effectively lead to a linear increase in compute and memory requirements, so --parallel 4 for\r\n",
      "                         e.g. the GRCm38 mouse genome will probably use ~20 cores and eat ~40GB or RAM, but at the same time\r\n",
      "                         reduce the alignment time to ~25-30%. You have been warned.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Bowtie 1 Reporting:\r\n",
      "\r\n",
      "-k <2>                   Due to the way Bismark works Bowtie will report up to 2 valid alignments. This option\r\n",
      "                         will be used by default.\r\n",
      "\r\n",
      "--best                   Make Bowtie guarantee that reported singleton alignments are \"best\" in terms of stratum\r\n",
      "                         (i.e. number of mismatches, or mismatches in the seed in the case if -n mode) and in\r\n",
      "                         terms of the quality; e.g. a 1-mismatch alignment where the mismatch position has Phred\r\n",
      "                         quality 40 is preferred over a 2-mismatch alignment where the mismatched positions both\r\n",
      "                         have Phred quality 10. When --best is not specified, Bowtie may report alignments that\r\n",
      "                         are sub-optimal in terms of stratum and/or quality (though an effort is made to report\r\n",
      "                         the best alignment). --best mode also removes all strand bias. Note that --best does not\r\n",
      "                         affect which alignments are considered \"valid\" by Bowtie, only which valid alignments\r\n",
      "                         are reported by Bowtie. Bowtie is about 1-2.5 times slower when --best is specified.\r\n",
      "                         Default: on.\r\n",
      "\r\n",
      "--no_best                Disables the --best option which is on by default. This can speed up the alignment process,\r\n",
      "                         e.g. for testing purposes, but for credible results it is not recommended to disable --best.\r\n",
      "\r\n",
      "\r\n",
      "Output:\r\n",
      "\r\n",
      "--non_directional        The sequencing library was constructed in a non strand-specific manner, alignments to all four\r\n",
      "                         bisulfite strands will be reported. Default: OFF.\r\n",
      "\r\n",
      "                         (The current Illumina protocol for BS-Seq is directional, in which case the strands complementary\r\n",
      "                         to the original strands are merely theoretical and should not exist in reality. Specifying directional\r\n",
      "                         alignments (which is the default) will only run 2 alignment threads to the original top (OT)\r\n",
      "                         or bottom (OB) strands in parallel and report these alignments. This is the recommended option\r\n",
      "                         for sprand-specific libraries).\r\n",
      "\r\n",
      "--pbat                   This options may be used for PBAT-Seq libraries (Post-Bisulfite Adapter Tagging; Kobayashi et al.,\r\n",
      "                         PLoS Genetics, 2012). This is essentially the exact opposite of alignments in 'directional' mode,\r\n",
      "                         as it will only launch two alignment threads to the CTOT and CTOB strands instead of the normal OT\r\n",
      "                         and OB ones. Use this option only if you are certain that your libraries were constructed following\r\n",
      "                         a PBAT protocol (if you don't know what PBAT-Seq is you should not specify this option). The option\r\n",
      "                         --pbat works only for FastQ files (in both Bowtie and Bowtie 2 mode) and using uncompressed\r\n",
      "                         temporary files only).\r\n",
      "\r\n",
      "--sam-no-hd              Suppress SAM header lines (starting with @). This might be useful when very large input files are\r\n",
      "                         split up into several smaller files to run concurrently and the output files are to be merged.\r\n",
      "\r\n",
      "--rg_tag                 Write out a Read Group tag to the resulting SAM/BAM file. This will write the following line to the\r\n",
      "                         SAM header: @RG PL: ILLUMINA ID:SAMPLE SM:SAMPLE ; to set ID and SM see --rg_id and --rg_sample.\r\n",
      "                         In addition each read receives an RG:Z:RG-ID tag. Default: OFF.\r\n",
      "\r\n",
      "--rg_id <string>         Sets the ID field in the @RG header line. The default is 'SAMPLE'.\r\n",
      "\r\n",
      "--rg_sample <string>     Sets the SM field in the @RG header line; can't be set without setting --rg_id as well. The default is\r\n",
      "                         'SAMPLE'.\r\n",
      "\r\n",
      "--quiet                  Print nothing besides alignments.\r\n",
      "\r\n",
      "--vanilla                Performs bisulfite mapping with Bowtie 1 and prints the 'old' output (as in Bismark 0.5.X) instead\r\n",
      "                         of SAM format output.\r\n",
      "\r\n",
      "-un/--unmapped           Write all reads that could not be aligned to a file in the output directory. Written reads will\r\n",
      "                         appear as they did in the input, without any translation of quality values that may have\r\n",
      "                         taken place within Bowtie or Bismark. Paired-end reads will be written to two parallel files with _1\r\n",
      "                         and _2 inserted in their filenames, i.e. _unmapped_reads_1.txt and unmapped_reads_2.txt. Reads\r\n",
      "                         with more than one valid alignment with the same number of lowest mismatches (ambiguous mapping)\r\n",
      "                         are also written to _unmapped_reads.txt unless the option --ambiguous is specified as well.\r\n",
      "\r\n",
      "--ambiguous              Write all reads which produce more than one valid alignment with the same number of lowest\r\n",
      "                         mismatches or other reads that fail to align uniquely to a file in the output directory.\r\n",
      "                         Written reads will appear as. they did in the input, without any of the translation of quality\r\n",
      "                         values that may have taken place within Bowtie or Bismark. Paired-end reads will be written to two\r\n",
      "                         parallel files with _1 and _2 inserted in their filenames, i.e. _ambiguous_reads_1.txt and\r\n",
      "                         _ambiguous_reads_2.txt. These reads are not written to the file specified with --un.\r\n",
      "\r\n",
      "-o/--output_dir <dir>    Write all output files into this directory. By default the output files will be written into\r\n",
      "                         the same folder as the input file(s). If the specified folder does not exist, Bismark will attempt\r\n",
      "                         to create it first. The path to the output folder can be either relative or absolute.\r\n",
      "\r\n",
      "--temp_dir <dir>         Write temporary files to this directory instead of into the same directory as the input files. If\r\n",
      "                         the specified folder does not exist, Bismark will attempt to create it first. The path to the\r\n",
      "                         temporary folder can be either relative or absolute.\r\n",
      "\r\n",
      "--non_bs_mm              Optionally outputs an extra column specifying the number of non-bisulfite mismatches a read during the\r\n",
      "                         alignment step. This option is only available for SAM format. In Bowtie 2 context, this value is\r\n",
      "                         just the number of actual non-bisulfite mismatches and ignores potential insertions or deletions.\r\n",
      "                         The format for single-end reads and read 1 of paired-end reads is 'XA:Z:number of mismatches'\r\n",
      "                         and 'XB:Z:number of mismatches' for read 2 of paired-end reads.\r\n",
      "\r\n",
      "--gzip                   Temporary bisulfite conversion files will be written out in a GZIP compressed form to save disk\r\n",
      "                         space. This option is available for most alignment modes but is not available for paired-end FastA\r\n",
      "                         files. This option might be somewhat slower than writing out uncompressed files, but this awaits\r\n",
      "                         further testing.\r\n",
      "\r\n",
      "--sam                    The output will be written out in SAM format instead of the default BAM format. Bismark will\r\n",
      "                         attempt to use the path to Samtools that was specified with '--samtools_path', or, if it hasn't\r\n",
      "                         been specified, attempt to find Samtools in the PATH. If no installation of Samtools can be found,\r\n",
      "                         the SAM output will be compressed with GZIP instead (yielding a .sam.gz output file).\r\n",
      "\r\n",
      "--cram                   Writes the output to a CRAM file instead of BAM. This requires the use of Samtools 1.2 or higher.\r\n",
      "\r\n",
      "--cram_ref <ref_file>    CRAM output requires you to specify a reference genome as a single FastA file. If this single-FastA\r\n",
      "                         reference file is not supplied explicitly it will be regenerated from the genome .fa sequence(s)\r\n",
      "                         used for the Bismark run and written to a file called 'Bismark_genome_CRAM_reference.mfa' into the\r\n",
      "                         oputput directory.\r\n",
      "\r\n",
      "--samtools_path          The path to your Samtools installation, e.g. /home/user/samtools/. Does not need to be specified\r\n",
      "                         explicitly if Samtools is in the PATH already.\r\n",
      "\r\n",
      "--prefix <prefix>        Prefixes <prefix> to the output filenames. Trailing dots will be replaced by a single one. For\r\n",
      "                         example, '--prefix test' with 'file.fq' would result in the output file 'test.file.fq_bismark.sam' etc.\r\n",
      "\r\n",
      "-B/--basename <basename> Write all output to files starting with this base file name. For example, '--basename foo'\r\n",
      "                         would result in the files 'foo.bam' and 'foo_SE_report.txt' (or its paired-end equivalent). Takes\r\n",
      "                         precedence over --prefix.\r\n",
      "\r\n",
      "--old_flag               Only in paired-end SAM mode, uses the FLAG values used by Bismark v0.8.2 and before. In addition,\r\n",
      "                         this options appends /1 and /2 to the read IDs for reads 1 and 2 relative to the input file. Since\r\n",
      "                         both the appended read IDs and custom FLAG values may cause problems with some downstream tools\r\n",
      "                         such as Picard, new defaults were implemented as of version 0.8.3.\r\n",
      "\r\n",
      "\r\n",
      "                                             default                         old_flag\r\n",
      "                                       ===================              ===================\r\n",
      "                                       Read 1       Read 2              Read 1       Read 2\r\n",
      "\r\n",
      "                              OT:         99          147                  67          131\r\n",
      "\r\n",
      "                              OB:         83          163                 115          179\r\n",
      "\r\n",
      "                              CTOT:      147           99                  67          131\r\n",
      "\r\n",
      "                              CTOB:      163           83                 115          179\r\n",
      "\r\n",
      "--ambig_bam              For reads that have multiple alignments a random alignment is written out to a special file ending in\r\n",
      "                         '.ambiguous.bam'. The alignments are in Bowtie2 format and do not any contain Bismark specific\r\n",
      "                         entries such as the methylation call etc. These ambiguous BAM files are intended to be used as\r\n",
      "                         coverage estimators for variant callers.\r\n",
      "\r\n",
      "--nucleotide_coverage    Calculates the mono- and di-nucleotide sequence composition of covered positions in the analysed BAM\r\n",
      "                         file and compares it to the genomic average composition once alignments are complete by calling 'bam2nuc'.\r\n",
      "                         Since this calculation may take a while, bam2nuc attempts to write the genomic sequence composition\r\n",
      "                         into a file called 'genomic_nucleotide_frequencies.txt' indside the reference genome folder so it can\r\n",
      "                         be re-used the next time round instead of calculating it once again. If a file 'nucleotide_stats.txt' is\r\n",
      "                         found with the Bismark reports it will be automatically detected and used for the Bismark HTML report.\r\n",
      "                         This option works only for BAM or CRAM files.\r\n",
      "\r\n",
      "\r\n",
      "Other:\r\n",
      "\r\n",
      "-h/--help                Displays this help file.\r\n",
      "\r\n",
      "-v/--version             Displays version information.\r\n",
      "\r\n",
      "\r\n",
      "BOWTIE 2 SPECIFIC OPTIONS\r\n",
      "\r\n",
      "--bowtie1                Uses Bowtie 1 instead of Bowtie 2, which might be a good choice for faster and very short\r\n",
      "                         alignments. Bismark assumes that raw sequence data is adapter and/or quality trimmed where\r\n",
      "                         appropriate. Default: off.\r\n",
      "\r\n",
      "--bowtie2                Default: ON. Uses Bowtie 2 instead of Bowtie 1. Bismark limits Bowtie 2 to only perform end-to-end\r\n",
      "                         alignments, i.e. searches for alignments involving all read characters (also called\r\n",
      "                         untrimmed or unclipped alignments). Bismark assumes that raw sequence data is adapter\r\n",
      "                         and/or quality trimmed where appropriate. Both small (.bt2) and large (.bt2l) Bowtie 2\r\n",
      "                         indexes are supported.\r\n",
      "\r\n",
      "Bowtie 2 alignment options:\r\n",
      "\r\n",
      "-N <int>                 Sets the number of mismatches to allowed in a seed alignment during multiseed alignment.\r\n",
      "                         Can be set to 0 or 1. Setting this higher makes alignment slower (often much slower)\r\n",
      "                         but increases sensitivity. Default: 0. This option is only available for Bowtie 2 (for\r\n",
      "                         Bowtie 1 see -n).\r\n",
      "\r\n",
      "-L <int>                 Sets the length of the seed substrings to align during multiseed alignment. Smaller values\r\n",
      "                         make alignment slower but more senstive. Default: the --sensitive preset of Bowtie 2 is\r\n",
      "                         used by default, which sets -L to 20. maximum of L can be set to 32. The length of the seed\r\n",
      "                         would effect the alignment speed dramatically while the larger L, the faster the aligment.\r\n",
      "                         This option is only available for Bowtie 2 (for Bowtie 1 see -l).\r\n",
      "\r\n",
      "--ignore-quals           When calculating a mismatch penalty, always consider the quality value at the mismatched\r\n",
      "                         position to be the highest possible, regardless of the actual value. I.e. input is treated\r\n",
      "                         as though all quality values are high. This is also the default behavior when the input\r\n",
      "                         doesn't specify quality values (e.g. in -f mode). This option is invariable and on by default.\r\n",
      "\r\n",
      "\r\n",
      "Bowtie 2 paired-end options:\r\n",
      "\r\n",
      "--no-mixed               This option disables Bowtie 2's behavior to try to find alignments for the individual mates if\r\n",
      "                         it cannot find a concordant or discordant alignment for a pair. This option is invariable and\r\n",
      "                         and on by default.\r\n",
      "\r\n",
      "--no-discordant          Normally, Bowtie 2 looks for discordant alignments if it cannot find any concordant alignments.\r\n",
      "                         A discordant alignment is an alignment where both mates align uniquely, but that does not\r\n",
      "                         satisfy the paired-end constraints (--fr/--rf/--ff, -I, -X). This option disables that behavior\r\n",
      "                         and it is on by default.\r\n",
      "\r\n",
      "--no_dovetail            It is possible, though unusual, for the mates to \"dovetail\", with the mates seemingly extending\r\n",
      "                         \"past\" each other as in this example:\r\n",
      "\r\n",
      "                         Mate 1:                 GTCAGCTACGATATTGTTTGGGGTGACACATTACGC\r\n",
      "                         Mate 2:            TATGAGTCAGCTACGATATTGTTTGGGGTGACACAT\r\n",
      "                         Reference: GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGCGTCTTTGAC\r\n",
      "\r\n",
      "                         By default, dovetailing is considered inconsistent with concordant alignment, but by default\r\n",
      "                         Bismark calls Bowtie 2 with --dovetail, causing it to consider dovetailing alignments as\r\n",
      "                         concordant. This becomes relevant whenever reads are clipped from their 5' end prior to mapping,\r\n",
      "                         e.g. because of quality or bias issues.\r\n",
      "\r\n",
      "                         Specify --no_dovetail to turn off this behaviour for paired-end libraries. Default: OFF.\r\n",
      "\r\n",
      "\r\n",
      "Bowtie 2 effort options:\r\n",
      "\r\n",
      "-D <int>                 Up to <int> consecutive seed extension attempts can \"fail\" before Bowtie 2 moves on, using\r\n",
      "                         the alignments found so far. A seed extension \"fails\" if it does not yield a new best or a\r\n",
      "                         new second-best alignment. Default: 15.\r\n",
      "\r\n",
      "-R <int>                 <int> is the maximum number of times Bowtie 2 will \"re-seed\" reads with repetitive seeds.\r\n",
      "                         When \"re-seeding,\" Bowtie 2 simply chooses a new set of reads (same length, same number of\r\n",
      "                         mismatches allowed) at different offsets and searches for more alignments. A read is considered\r\n",
      "                         to have repetitive seeds if the total number of seed hits divided by the number of seeds\r\n",
      "                         that aligned at least once is greater than 300. Default: 2.\r\n",
      "\r\n",
      "Bowtie 2 parallelization options:\r\n",
      "\r\n",
      "\r\n",
      "-p NTHREADS              Launch NTHREADS parallel search threads (default: 1). Threads will run on separate processors/cores\r\n",
      "                         and synchronize when parsing reads and outputting alignments. Searching for alignments is highly\r\n",
      "                         parallel, and speedup is close to linear. Increasing -p increases Bowtie 2's memory footprint.\r\n",
      "                         E.g. when aligning to a human genome index, increasing -p from 1 to 8 increases the memory footprint\r\n",
      "                         by a few hundred megabytes. This option is only available if bowtie is linked with the pthreads\r\n",
      "                         library (i.e. if BOWTIE_PTHREADS=0 is not specified at build time). In addition, this option will\r\n",
      "                         automatically use the option '--reorder', which guarantees that output SAM records are printed in\r\n",
      "                         an order corresponding to the order of the reads in the original input file, even when -p is set\r\n",
      "                         greater than 1 (Bismark requires the Bowtie 2 output to be this way). Specifying --reorder and\r\n",
      "                         setting -p greater than 1 causes Bowtie 2 to run somewhat slower and use somewhat more memory then\r\n",
      "                         if --reorder were not specified. Has no effect if -p is set to 1, since output order will naturally\r\n",
      "                         correspond to input order in that case.\r\n",
      "\r\n",
      "Bowtie 2 Scoring options:\r\n",
      "\r\n",
      "--score_min <func>       Sets a function governing the minimum alignment score needed for an alignment to be considered\r\n",
      "                         \"valid\" (i.e. good enough to report). This is a function of read length. For instance, specifying\r\n",
      "                         L,0,-0.2 sets the minimum-score function f to f(x) = 0 + -0.2 * x, where x is the read length.\r\n",
      "                         See also: setting function options at http://bowtie-bio.sourceforge.net/bowtie2. The default is\r\n",
      "                         L,0,-0.2.\r\n",
      "\r\n",
      "--rdg <int1>,<int2>      Sets the read gap open (<int1>) and extend (<int2>) penalties. A read gap of length N gets a penalty\r\n",
      "                         of <int1> + N * <int2>. Default: 5, 3.\r\n",
      "\r\n",
      "--rfg <int1>,<int2>      Sets the reference gap open (<int1>) and extend (<int2>) penalties. A reference gap of length N gets\r\n",
      "                         a penalty of <int1> + N * <int2>. Default: 5, 3.\r\n",
      "\r\n",
      "\r\n",
      "Bowtie 2 Reporting options:\r\n",
      "\r\n",
      "-most_valid_alignments <int> This used to be the Bowtie 2 parameter -M. As of Bowtie 2 version 2.0.0 beta7 the option -M is\r\n",
      "                         deprecated. It will be removed in subsequent versions. What used to be called -M mode is still the\r\n",
      "                         default mode, but adjusting the -M setting is deprecated.  Use the -D and -R options to adjust the\r\n",
      "                         effort expended to find valid alignments.\r\n",
      "\r\n",
      "                         For reference, this used to be the old (now deprecated) description of -M:\r\n",
      "                         Bowtie 2 searches for at most <int>+1 distinct, valid alignments for each read. The search terminates when it\r\n",
      "                         can't find more distinct valid alignments, or when it finds <int>+1 distinct alignments, whichever\r\n",
      "                         happens first. Only the best alignment is reported. Information from the other alignments is used to\r\n",
      "                         estimate mapping quality and to set SAM optional fields, such as AS:i and XS:i. Increasing -M makes\r\n",
      "                         Bowtie 2 slower, but increases the likelihood that it will pick the correct alignment for a read that\r\n",
      "                         aligns many places. For reads that have more than <int>+1 distinct, valid alignments, Bowtie 2 does not\r\n",
      "                         guarantee that the alignment reported is the best possible in terms of alignment score. -M is\r\n",
      "                         always used and its default value is set to 10.\r\n",
      "\r\n",
      "\r\n",
      "'VANILLA' Bismark  OUTPUT:\r\n",
      "\r\n",
      "Single-end output format (tab-separated):\r\n",
      "\r\n",
      " (1) <seq-ID>\r\n",
      " (2) <read alignment strand>\r\n",
      " (3) <chromosome>\r\n",
      " (4) <start position>\r\n",
      " (5) <end position>\r\n",
      " (6) <observed bisulfite sequence>\r\n",
      " (7) <equivalent genomic sequence>\r\n",
      " (8) <methylation call>\r\n",
      " (9) <read conversion\r\n",
      "(10) <genome conversion>\r\n",
      "(11) <read quality score (Phred33)>\r\n",
      "\r\n",
      "\r\n",
      "Paired-end output format (tab-separated):\r\n",
      " (1) <seq-ID>\r\n",
      " (2) <read 1 alignment strand>\r\n",
      " (3) <chromosome>\r\n",
      " (4) <start position>\r\n",
      " (5) <end position>\r\n",
      " (6) <observed bisulfite sequence 1>\r\n",
      " (7) <equivalent genomic sequence 1>\r\n",
      " (8) <methylation call 1>\r\n",
      " (9) <observed bisulfite sequence 2>\r\n",
      "(10) <equivalent genomic sequence 2>\r\n",
      "(11) <methylation call 2>\r\n",
      "(12) <read 1 conversion\r\n",
      "(13) <genome conversion>\r\n",
      "(14) <read 1 quality score (Phred33)>\r\n",
      "(15) <read 2 quality score (Phred33)>\r\n",
      "\r\n",
      "\r\n",
      "Bismark SAM OUTPUT (default):\r\n",
      "\r\n",
      " (1) QNAME  (seq-ID)\r\n",
      " (2) FLAG   (this flag tries to take the strand a bisulfite read originated from into account (this is different from ordinary DNA alignment flags!))\r\n",
      " (3) RNAME  (chromosome)\r\n",
      " (4) POS    (start position)\r\n",
      " (5) MAPQ   (always 255 for use with Bowtie)\r\n",
      " (6) CIGAR\r\n",
      " (7) RNEXT\r\n",
      " (8) PNEXT\r\n",
      " (9) TLEN\r\n",
      "(10) SEQ\r\n",
      "(11) QUAL   (Phred33 scale)\r\n",
      "(12) NM-tag (edit distance to the reference)\r\n",
      "(13) MD-tag (base-by-base mismatches to the reference (handles indels)\r\n",
      "(14) XM-tag (methylation call string)\r\n",
      "(15) XR-tag (read conversion state for the alignment)\r\n",
      "(16) XG-tag (genome conversion state for the alignment)\r\n",
      "(17) XA/XB-tag (non-bisulfite mismatches) (optional!)\r\n",
      "\r\n",
      "Each read of paired-end alignments is written out in a separate line in the above format.\r\n",
      "\r\n",
      "\r\n",
      "Last modified on 10 October 2017\r\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What I need for this command:\n",
    "\n",
    "1. Path to `bismark`\n",
    "2. -u 10000: Only aligns the first 10000 reads or read pairs (basically allows me to subset data)\n",
    "3. --non_directional: See [this issue](https://github.com/RobertsLab/resources/issues/216)\n",
    "4. --score_min L,0,-1.2: Allow for mismatches\n",
    "4. --genome + path to the folder with the .fa genome, which also has all of the bisulfite genome directories prepared in step 1c.\n",
    "5. Path to sequence files for alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to Bowtie 2 specified as: bowtie2\n",
      "Bowtie seems to be working fine (tested command 'bowtie2 --version' [2.3.0])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/usr/local/bin/samtools'\n",
      "Reference genome folder provided is 2018-04-27-Bismark-Inputs/\t(absolute path is '/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/)'\n",
      "FastQ format assumed (by default)\n",
      "Processing sequences up to read no. 10000 from the input file\n",
      "\n",
      "Input files to be analysed (in current folder '/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark'):\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz\n",
      "/Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\n",
      "Library was specified to be not strand-specific (non-directional), therefore alignments to all four possible bisulfite strands (OT, CTOT, OB and CTOB) will be reported\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/\n",
      "\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_10_s1_R1.fastq.gz to zr2096_10_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_10_s1_R1.fastq.gz to zr2096_10_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_10_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_10_s1_R1.fastq.gz_C_to_T.fastq and zr2096_10_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_10_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_1:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tGTTATATATAATTGTATTAATTATAAAAAATATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATTTATTATTAAAAATATTAAATATATTATTTTA\tBBBBBFFFFFFFFFFFFFFBFFFFFFFFFFFFF#################################<</<FFFFFFFFFFFFFFFFFFFFFFFFFF<FFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_10_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_1:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tGTTATATATAATTGTATTAATTATAAAAAATATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATTTATTATTAAAAATATTAAATATATTATTTTA\tBBBBBFFFFFFFFFFFFFFBFFFFFFFFFFFFF#################################<</<FFFFFFFFFFFFFFFFFFFFFFFFFF<FFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_10_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_1:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tACCATATACAATCATATCAACTATAAAAAATACNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATCTACTATCAAAAATACTAAACACACTACCTTA\tBBBBBFFFFFFFFFFFFFFBFFFFFFFFFFFFF#################################<</<FFFFFFFFFFFFFFFFFFFFFFFFFF<FFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_10_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_1:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tACCATATACAATCATATCAACTATAAAAAATACNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATCTACTATCAAAAATACTAAACACACTACCTTA\tBBBBBFFFFFFFFFFFFFFBFFFFFFFFFFFFF#################################<</<FFFFFFFFFFFFFFFFFFFFFFFFFF<FFF\tYT:Z:UU\tYF:Z:NS\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_10_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6709 (67.09%) aligned 0 times\n",
      "    1341 (13.41%) aligned exactly 1 time\n",
      "    1950 (19.50%) aligned >1 times\n",
      "32.91% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6203 (62.03%) aligned 0 times\n",
      "    1586 (15.86%) aligned exactly 1 time\n",
      "    2211 (22.11%) aligned >1 times\n",
      "37.97% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6204 (62.04%) aligned 0 times\n",
      "    1552 (15.52%) aligned exactly 1 time\n",
      "    2244 (22.44%) aligned >1 times\n",
      "37.96% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6579 (65.79%) aligned 0 times\n",
      "    1472 (14.72%) aligned exactly 1 time\n",
      "    1949 (19.49%) aligned >1 times\n",
      "34.21% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_10_s1_R1.fastq.gz_C_to_T.fastq and zr2096_10_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6363\n",
      "Mapping efficiency:\t63.6%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1342\n",
      "Sequences did not map uniquely:\t2295\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1534\t((converted) top strand)\n",
      "CT/GA:\t1417\t((converted) bottom strand)\n",
      "GA/CT:\t1703\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1709\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t110692\n",
      "\n",
      "Total methylated C's in CpG context:\t10598\n",
      "Total methylated C's in CHG context:\t834\n",
      "Total methylated C's in CHH context:\t2849\n",
      "Total methylated C's in Unknown context:\t400\n",
      "\n",
      "Total unmethylated C's in CpG context:\t5034\n",
      "Total unmethylated C's in CHG context:\t24437\n",
      "Total unmethylated C's in CHH context:\t66940\n",
      "Total unmethylated C's in Unknown context:\t1800\n",
      "\n",
      "C methylated in CpG context:\t67.8%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.1%\n",
      "C methylated in Unknown context (CN or CHN):\t18.2%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 0m 44s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_10_s1_R2.fastq.gz to zr2096_10_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_10_s1_R2.fastq.gz to zr2096_10_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_10_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_10_s1_R2.fastq.gz_C_to_T.fastq and zr2096_10_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_10_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_2:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNGTTTTGTATTTTTGATAGTAGATATGATTGTATATTAGGGTAAGGTAGTGTGTTTAGTATTTTTGATAGTAGATATGATTGAATATTGTGGTAA\t#####BB<FFFFFFFFFFFFFBFFFFFFFFFFFFFFFFBFFBFFFFFFFFFBFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_10_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_2:N:0:GATCAG\t16\tNC_035781.1_GA_converted\t5622442\t7\t99M\t*\t0\t0\tTTACCACAATATTCAATCATATCTACTATCAAAAATACTAAACACACTACCTTACCCTAATATACAATCATATCTACTATCAAAAATACAAAACNNNNN\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFBFFFFFFFFFBFFBFFFFFFFFFFFFFFFFBFFFFFFFFFFFFF<BB#####\tAS:i:-23\tXS:i:-35\tXN:i:0\tXM:i:8\tXO:i:0\tXG:i:0\tNM:i:8\tMD:Z:23A14C35A19A0C0A0C0T0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_10_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_2:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNATTTTATATTTTTAATAATAAATATAATTATATATTAAAATAAAATAATATATTTAATATTTTTAATAATAAATACAATTAAATATCATAATAA\t#####BB<FFFFFFFFFFFFFBFFFFFFFFFFFFFFFFBFFBFFFFFFFFFBFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_10_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1232:2155_2:N:0:GATCAG\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNATTTTATATTTTTAATAATAAATATAATTATATATTAAAATAAAATAATATATTTAATATTTTTAATAATAAATACAATTAAATATCATAATAA\t#####BB<FFFFFFFFFFFFFBFFFFFFFFFFFFFFFFBFFBFFFFFFFFFBFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_10_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6052 (60.52%) aligned 0 times\n",
      "    1681 (16.81%) aligned exactly 1 time\n",
      "    2267 (22.67%) aligned >1 times\n",
      "39.48% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6655 (66.55%) aligned 0 times\n",
      "    1401 (14.01%) aligned exactly 1 time\n",
      "    1944 (19.44%) aligned >1 times\n",
      "33.45% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6143 (61.43%) aligned 0 times\n",
      "    1594 (15.94%) aligned exactly 1 time\n",
      "    2263 (22.63%) aligned >1 times\n",
      "38.57% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6636 (66.36%) aligned 0 times\n",
      "    1396 (13.96%) aligned exactly 1 time\n",
      "    1968 (19.68%) aligned >1 times\n",
      "33.64% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_10_s1_R2.fastq.gz_C_to_T.fastq and zr2096_10_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6457\n",
      "Mapping efficiency:\t64.6%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1202\n",
      "Sequences did not map uniquely:\t2341\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1456\t((converted) top strand)\n",
      "CT/GA:\t1458\t((converted) bottom strand)\n",
      "GA/CT:\t1849\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1694\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t111568\n",
      "\n",
      "Total methylated C's in CpG context:\t10678\n",
      "Total methylated C's in CHG context:\t827\n",
      "Total methylated C's in CHH context:\t2954\n",
      "Total methylated C's in Unknown context:\t411\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4977\n",
      "Total unmethylated C's in CHG context:\t24762\n",
      "Total unmethylated C's in CHH context:\t67370\n",
      "Total unmethylated C's in Unknown context:\t1776\n",
      "\n",
      "C methylated in CpG context:\t68.2%\n",
      "C methylated in CHG context:\t3.2%\n",
      "C methylated in CHH context:\t4.2%\n",
      "C methylated in Unknown context (CN or CHN):\t18.8%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 0m 52s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_1_s1_R1.fastq.gz to zr2096_1_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_1_s1_R1.fastq.gz to zr2096_1_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_1_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_1_s1_R1.fastq.gz_C_to_T.fastq and zr2096_1_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_1_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_1:N:0:CGATGT\t0\tNC_035783.1_CT_converted\t41658294\t1\t100M\t*\t0\t0\tGAGTTTTTTTGATTATTTGTTGTTTGTTGTTTGTTTGNNNNTNNNTNNGTTTGTTTGTTTGTTTGTTTGTTTGTAAATTTTTTATATTTTTATTTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<##<<<FFFFFFFFFFFFFFFF<FFFFFF/</FFFFFF/B/FFFFF/FFFFFFFF\tAS:i:-9\tXS:i:-9\tXN:i:0\tXM:i:9\tXO:i:0\tXG:i:0\tNM:i:9\tMD:Z:37T0T0T0G1T0T0G1T0T52\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_1_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_1:N:0:CGATGT\t16\tNC_035784.1_GA_converted\t16863647\t1\t100M\t*\t0\t0\tAAAAAAAATAAAAATATAAAAAATTTACAAACAAACAAACAAACAAACAAACNNANNNANNNNCAAACAAACAACAAACAACAAATAATCAAAAAAACTC\tFFFFFFFF/FFFFF/B/FFFFFF/</FFFFFF<FFFFFFFFFFFFFFFF<<<##<###<####FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-9\tXS:i:-9\tXN:i:0\tXM:i:9\tXO:i:0\tXG:i:0\tNM:i:9\tMD:Z:52A0A1C0A0A1C0A0A0A37\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_1_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_1:N:0:CGATGT\t16\tNC_035780.1_CT_converted\t31013177\t0\t11M4I63M2I20M\t*\t0\t0\tAAAAAAAATAAAAATATAAAAAATTTATAAATGAATAAATAAATAAATAAATNNANNNANNNNTAAATAAATGATGAATGATAAATAATTAAAAAAATTT\tFFFFFFFF/FFFFF/B/FFFFFF/</FFFFFF<FFFFFFFFFFFFFFFF<<<##<###<####FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-115\tXN:i:0\tXM:i:22\tXO:i:2\tXG:i:6\tNM:i:28\tMD:Z:0T3T11T2A1A2G23A0A1T0A0A1T0A0A0A9A1A0T10A3G2T3G0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_1_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_1:N:0:CGATGT\t0\tNC_035783.1_GA_converted\t30506405\t2\t12M1D8M2I2M4I14M1D38M1D20M\t*\t0\t0\tAAATTTTTTTAATTATTTATCATTCATCATTTATTTANNNNTNNNTNNATTTATTTATTTATTTATTCATTTATAAATTTTTTATATTTTTATTTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<##<<<FFFFFFFFFFFFFFFF<FFFFFF/</FFFFFF/B/FFFFF/FFFFFFFF\tAS:i:-109\tXS:i:-120\tXN:i:0\tXM:i:17\tXO:i:5\tXG:i:9\tNM:i:26\tMD:Z:1T10^T19A0A0A0T1^A0T0T0A1T0T19T6T0T4^A2C2C1A8A3\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_1_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8165 (81.65%) aligned 0 times\n",
      "    744 (7.44%) aligned exactly 1 time\n",
      "    1091 (10.91%) aligned >1 times\n",
      "18.35% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8168 (81.68%) aligned 0 times\n",
      "    767 (7.67%) aligned exactly 1 time\n",
      "    1065 (10.65%) aligned >1 times\n",
      "18.32% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8253 (82.53%) aligned 0 times\n",
      "    724 (7.24%) aligned exactly 1 time\n",
      "    1023 (10.23%) aligned >1 times\n",
      "17.47% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8307 (83.07%) aligned 0 times\n",
      "    671 (6.71%) aligned exactly 1 time\n",
      "    1022 (10.22%) aligned >1 times\n",
      "16.93% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_1_s1_R1.fastq.gz_C_to_T.fastq and zr2096_1_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t3083\n",
      "Mapping efficiency:\t30.8%\n",
      "\n",
      "Sequences with no alignments under any condition:\t5801\n",
      "Sequences did not map uniquely:\t1116\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t723\t((converted) top strand)\n",
      "CT/GA:\t759\t((converted) bottom strand)\n",
      "GA/CT:\t808\t(complementary to (converted) top strand)\n",
      "GA/GA:\t793\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t50039\n",
      "\n",
      "Total methylated C's in CpG context:\t4195\n",
      "Total methylated C's in CHG context:\t390\n",
      "Total methylated C's in CHH context:\t2635\n",
      "Total methylated C's in Unknown context:\t322\n",
      "\n",
      "Total unmethylated C's in CpG context:\t2109\n",
      "Total unmethylated C's in CHG context:\t10153\n",
      "Total unmethylated C's in CHH context:\t30557\n",
      "Total unmethylated C's in Unknown context:\t1075\n",
      "\n",
      "C methylated in CpG context:\t66.5%\n",
      "C methylated in CHG context:\t3.7%\n",
      "C methylated in CHH context:\t7.9%\n",
      "C methylated in Unknown context (CN or CHN):\t23.0%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 1s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_1_s1_R2.fastq.gz to zr2096_1_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_1_s1_R2.fastq.gz to zr2096_1_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_1_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_1_s1_R2.fastq.gz_C_to_T.fastq and zr2096_1_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_1_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_2:N:0:CGATGT\t4\t*\t0\t0\t*\t*\t0\t0\tTTTTTTAAAAAAAAATGAAATTTTTTATTTAAATAAATTTAAATTTTTTTTATTTAAAAATATTTTATATTAAATTTAATTAAAATTAATTTAATAATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_1_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_2:N:0:CGATGT\t4\t*\t0\t0\t*\t*\t0\t0\tTTTTTTAAAAAAAAATGAAATTTTTTATTTAAATAAATTTAAATTTTTTTTATTTAAAAATATTTTATATTAAATTTAATTAAAATTAATTTAATAATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_1_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_2:N:0:CGATGT\t16\tNC_035787.1_CT_converted\t60683662\t6\t100M\t*\t0\t0\tGAATTATTGGGTTAATTTTAATTAAATTTGATATAAAGTATTTTTAGGTGAAGGGGATTTAAGTTTGTTTGAATGAAGGGTTTTGTTTTTTTTTAAGGGG\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-6\tXS:i:-12\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:45G54\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_1_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1249:2156_2:N:0:CGATGT\t0\tNC_035780.1_GA_converted\t8090028\t12\t100M\t*\t0\t0\tCCCCTTAAAAAAAAACAAAACCCTTCATTCAAACAAACTTAAATCCCCTTCACCTAAAAATACTTTATATCAAATTTAATTAAAATTAACCCAATAATTC\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:-6\tXS:i:-30\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:29T70\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_1_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8074 (80.74%) aligned 0 times\n",
      "    800 (8.00%) aligned exactly 1 time\n",
      "    1126 (11.26%) aligned >1 times\n",
      "19.26% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8079 (80.79%) aligned 0 times\n",
      "    818 (8.18%) aligned exactly 1 time\n",
      "    1103 (11.03%) aligned >1 times\n",
      "19.21% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8356 (83.56%) aligned 0 times\n",
      "    641 (6.41%) aligned exactly 1 time\n",
      "    1003 (10.03%) aligned >1 times\n",
      "16.44% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    8297 (82.97%) aligned 0 times\n",
      "    729 (7.29%) aligned exactly 1 time\n",
      "    974 (9.74%) aligned >1 times\n",
      "17.03% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_1_s1_R2.fastq.gz_C_to_T.fastq and zr2096_1_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t3211\n",
      "Mapping efficiency:\t32.1%\n",
      "\n",
      "Sequences with no alignments under any condition:\t5683\n",
      "Sequences did not map uniquely:\t1106\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t756\t((converted) top strand)\n",
      "CT/GA:\t690\t((converted) bottom strand)\n",
      "GA/CT:\t906\t(complementary to (converted) top strand)\n",
      "GA/GA:\t859\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t52598\n",
      "\n",
      "Total methylated C's in CpG context:\t4485\n",
      "Total methylated C's in CHG context:\t417\n",
      "Total methylated C's in CHH context:\t2492\n",
      "Total methylated C's in Unknown context:\t324\n",
      "\n",
      "Total unmethylated C's in CpG context:\t2134\n",
      "Total unmethylated C's in CHG context:\t10794\n",
      "Total unmethylated C's in CHH context:\t32276\n",
      "Total unmethylated C's in Unknown context:\t1129\n",
      "\n",
      "C methylated in CpG context:\t67.8%\n",
      "C methylated in CHG context:\t3.7%\n",
      "C methylated in CHH context:\t7.2%\n",
      "C methylated in Unknown context (CN or CHN):\t22.3%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 12s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_2_s1_R1.fastq.gz to zr2096_2_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_2_s1_R1.fastq.gz to zr2096_2_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_2_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_2_s1_R1.fastq.gz_C_to_T.fastq and zr2096_2_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_2_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_1:N:0:TGACCA\t4\t*\t0\t0\t*\t*\t0\t0\tTTTTTTATTAAAATTTATTATTTTAAAAATAAAAATTNNNNTNNNTTTTATTATTTTTTTTAATATTTTTTAATATAATTAAATAATAAAAAAAATTAAT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_2_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_1:N:0:TGACCA\t4\t*\t0\t0\t*\t*\t0\t0\tTTTTTTATTAAAATTTATTATTTTAAAAATAAAAATTNNNNTNNNTTTTATTATTTTTTTTAATATTTTTTAATATAATTAAATAATAAAAAAAATTAAT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_2_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_1:N:0:TGACCA\t16\tNC_035782.1_CT_converted\t11804233\t6\t7M1D93M\t*\t0\t0\tATTAATTTTTTTTGTTATTTAGTTATATTGGGAGATGTTAGAAGGAATGGTGGAANNNGNNNNAGTTTTTATTTTTGGAATAGTAGATTTTAATAAGGAG\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFF<<<###<####FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-27\tXS:i:-33\tXN:i:0\tXM:i:9\tXO:i:1\tXG:i:1\tNM:i:10\tMD:Z:3T0T2^G48G0A0A1A0G0A0A37\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_2_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_1:N:0:TGACCA\t4\t*\t0\t0\t*\t*\t0\t0\tCTCCTTATTAAAATCTACTATTCCAAAAATAAAAACTNNNNCNNNTTCCACCATTCCTTCTAACATCTCCCAATATAACTAAATAACAAAAAAAATTAAT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_2_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6684 (66.84%) aligned 0 times\n",
      "    1393 (13.93%) aligned exactly 1 time\n",
      "    1923 (19.23%) aligned >1 times\n",
      "33.16% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6679 (66.79%) aligned 0 times\n",
      "    1378 (13.78%) aligned exactly 1 time\n",
      "    1943 (19.43%) aligned >1 times\n",
      "33.21% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6868 (68.68%) aligned 0 times\n",
      "    1266 (12.66%) aligned exactly 1 time\n",
      "    1866 (18.66%) aligned >1 times\n",
      "31.32% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6879 (68.79%) aligned 0 times\n",
      "    1246 (12.46%) aligned exactly 1 time\n",
      "    1875 (18.75%) aligned >1 times\n",
      "31.21% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_2_s1_R1.fastq.gz_C_to_T.fastq and zr2096_2_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t5451\n",
      "Mapping efficiency:\t54.5%\n",
      "\n",
      "Sequences with no alignments under any condition:\t2569\n",
      "Sequences did not map uniquely:\t1980\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1311\t((converted) top strand)\n",
      "CT/GA:\t1303\t((converted) bottom strand)\n",
      "GA/CT:\t1420\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1417\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t86892\n",
      "\n",
      "Total methylated C's in CpG context:\t8081\n",
      "Total methylated C's in CHG context:\t711\n",
      "Total methylated C's in CHH context:\t3049\n",
      "Total methylated C's in Unknown context:\t415\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3617\n",
      "Total unmethylated C's in CHG context:\t19399\n",
      "Total unmethylated C's in CHH context:\t52035\n",
      "Total unmethylated C's in Unknown context:\t1576\n",
      "\n",
      "C methylated in CpG context:\t69.1%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t5.5%\n",
      "C methylated in Unknown context (CN or CHN):\t20.8%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 21s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_2_s1_R2.fastq.gz to zr2096_2_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_2_s1_R2.fastq.gz to zr2096_2_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_2_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_2_s1_R2.fastq.gz_C_to_T.fastq and zr2096_2_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_2_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_2:N:0:TGACCA\t0\tNC_035786.1_CT_converted\t53592787\t0\t93M2I4M\t*\t0\t0\tTTGTTTTGGAAAAATAATGGATAAAAAAGGTTGATGTGGTGATGTTTTTAGTGATGTGGTGGTGTTTGAGAAATTATTTATTAATTTTTTTTGTTATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFBFFFFFFFFBFF/BFF\tAS:i:-41\tXS:i:-41\tXN:i:0\tXM:i:5\tXO:i:1\tXG:i:2\tNM:i:7\tMD:Z:0A0A0A41A48G3\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_2_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_2:N:0:TGACCA\t16\tNC_035784.1_GA_converted\t62327061\t6\t6M3D93M\t*\t0\t0\tAAATAACAAAAAAAATTAATAAATAATTTCTCAAACACCACCACATCACTAAAAACATCACCACATCAACCTTTTTTATCCATTATTTTTCCAAAACAA\tFFB/FFBFFFFFFFFBFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-38\tXS:i:-41\tXN:i:0\tXM:i:4\tXO:i:1\tXG:i:3\tNM:i:7\tMD:Z:6^AAC48T41T0T0T0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_2_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_2:N:0:TGACCA\t4\t*\t0\t0\t*\t*\t0\t0\tTCACTTCAAAAAAATAATAAATAAAAAAAATTAATACAACAATATTTTTAATAATACAACAATATTTAAAAAATTATTTATTAATTTTTTTTATTATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFBFFFFFFFFBFF/BFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_2_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1148:2157_2:N:0:TGACCA\t4\t*\t0\t0\t*\t*\t0\t0\tTCACTTCAAAAAAATAATAAATAAAAAAAATTAATACAACAATATTTTTAATAATACAACAATATTTAAAAAATTATTTATTAATTTTTTTTATTATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFBFFFFFFFFBFF/BFF\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_2_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6884 (68.84%) aligned 0 times\n",
      "    1324 (13.24%) aligned exactly 1 time\n",
      "    1792 (17.92%) aligned >1 times\n",
      "31.16% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6606 (66.06%) aligned 0 times\n",
      "    1374 (13.74%) aligned exactly 1 time\n",
      "    2020 (20.20%) aligned >1 times\n",
      "33.94% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6903 (69.03%) aligned 0 times\n",
      "    1282 (12.82%) aligned exactly 1 time\n",
      "    1815 (18.15%) aligned >1 times\n",
      "30.97% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6671 (66.71%) aligned 0 times\n",
      "    1320 (13.20%) aligned exactly 1 time\n",
      "    2009 (20.09%) aligned >1 times\n",
      "33.29% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_2_s1_R2.fastq.gz_C_to_T.fastq and zr2096_2_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t5481\n",
      "Mapping efficiency:\t54.8%\n",
      "\n",
      "Sequences with no alignments under any condition:\t2531\n",
      "Sequences did not map uniquely:\t1988\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1295\t((converted) top strand)\n",
      "CT/GA:\t1317\t((converted) bottom strand)\n",
      "GA/CT:\t1452\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1417\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t87388\n",
      "\n",
      "Total methylated C's in CpG context:\t8135\n",
      "Total methylated C's in CHG context:\t666\n",
      "Total methylated C's in CHH context:\t3294\n",
      "Total methylated C's in Unknown context:\t448\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3627\n",
      "Total unmethylated C's in CHG context:\t19502\n",
      "Total unmethylated C's in CHH context:\t52164\n",
      "Total unmethylated C's in Unknown context:\t1554\n",
      "\n",
      "C methylated in CpG context:\t69.2%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t5.9%\n",
      "C methylated in Unknown context (CN or CHN):\t22.4%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 31s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_3_s1_R1.fastq.gz to zr2096_3_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_3_s1_R1.fastq.gz to zr2096_3_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_3_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_3_s1_R1.fastq.gz_C_to_T.fastq and zr2096_3_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_3_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_1:N:0:ACAGTG\t4\t*\t0\t0\t*\t*\t0\t0\tTNTTTTAAAATAATTTAAATAATTTTATATAAATNNNNNNNNNNNNNNNTATTTAAATTAAAAATTTAAATAAATTTTTTTAATTATTTATTGTTTAATG\tB#<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFF###############<<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_3_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_1:N:0:ACAGTG\t4\t*\t0\t0\t*\t*\t0\t0\tTNTTTTAAAATAATTTAAATAATTTTATATAAATNNNNNNNNNNNNNNNTATTTAAATTAAAAATTTAAATAAATTTTTTTAATTATTTATTGTTTAATG\tB#<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFF###############<<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_3_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_1:N:0:ACAGTG\t4\t*\t0\t0\t*\t*\t0\t0\tCNCTTCAAAATAATTTAAATAATCTTATATAAATNNNNNNNNNNNNNNNCACCTAAACTAAAAACTCAAATAAACTTTTCTAATCACCTATCATCCAACA\tB#<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFF###############<<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_3_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_1:N:0:ACAGTG\t4\t*\t0\t0\t*\t*\t0\t0\tCNCTTCAAAATAATTTAAATAATCTTATATAAATNNNNNNNNNNNNNNNCACCTAAACTAAAAACTCAAATAAACTTTTCTAATCACCTATCATCCAACA\tB#<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFF###############<<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_3_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6063 (60.63%) aligned 0 times\n",
      "    1617 (16.17%) aligned exactly 1 time\n",
      "    2320 (23.20%) aligned >1 times\n",
      "39.37% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6145 (61.45%) aligned 0 times\n",
      "    1529 (15.29%) aligned exactly 1 time\n",
      "    2326 (23.26%) aligned >1 times\n",
      "38.55% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6537 (65.37%) aligned 0 times\n",
      "    1347 (13.47%) aligned exactly 1 time\n",
      "    2116 (21.16%) aligned >1 times\n",
      "34.63% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6492 (64.92%) aligned 0 times\n",
      "    1405 (14.05%) aligned exactly 1 time\n",
      "    2103 (21.03%) aligned >1 times\n",
      "35.08% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_3_s1_R1.fastq.gz_C_to_T.fastq and zr2096_3_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6272\n",
      "Mapping efficiency:\t62.7%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1237\n",
      "Sequences did not map uniquely:\t2491\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1401\t((converted) top strand)\n",
      "CT/GA:\t1483\t((converted) bottom strand)\n",
      "GA/CT:\t1666\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1722\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t109168\n",
      "\n",
      "Total methylated C's in CpG context:\t12011\n",
      "Total methylated C's in CHG context:\t784\n",
      "Total methylated C's in CHH context:\t2675\n",
      "Total methylated C's in Unknown context:\t400\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3734\n",
      "Total unmethylated C's in CHG context:\t24189\n",
      "Total unmethylated C's in CHH context:\t65775\n",
      "Total unmethylated C's in Unknown context:\t1825\n",
      "\n",
      "C methylated in CpG context:\t76.3%\n",
      "C methylated in CHG context:\t3.1%\n",
      "C methylated in CHH context:\t3.9%\n",
      "C methylated in Unknown context (CN or CHN):\t18.0%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 44s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_3_s1_R2.fastq.gz to zr2096_3_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_3_s1_R2.fastq.gz to zr2096_3_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_3_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_3_s1_R2.fastq.gz_C_to_T.fastq and zr2096_3_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_3_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_2:N:0:ACAGTG\t0\tNC_035784.1_CT_converted\t71250717\t2\t80M1D9M1I9M\t*\t0\t0\tNNNNAGATAGATGTTGGATGATAGGTGATTAGAAAAGTTTATTTGAGTTTTTAGTTTAGGTGAGTTAAAAATATTGAATTTATATAAGATTATTTAAAT\t####<BBFFFFFFFFFFFFF<FFFFFF<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFB\tAS:i:-56\tXS:i:-57\tXN:i:0\tXM:i:10\tXO:i:2\tXG:i:2\tNM:i:12\tMD:Z:0G0G0A0T0G8A66^A0A1G6A1A6\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_3_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_2:N:0:ACAGTG\t16\tNC_035788.1_GA_converted\t71743390\t21\t99M\t*\t0\t0\tATTTAAATAATCTTATATAAATTCAATATTTTTAACTCACCTAAACTAAAAACTCAAATAAACTTTTCTAATCACCTATCATCCAACATCTATCTNNNN\tBFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFF<FFFFFF<FFFFFFFFFFFFFBB<####\tAS:i:-16\tXS:i:-72\tXN:i:0\tXM:i:6\tXO:i:0\tXG:i:0\tNM:i:6\tMD:Z:0T0A93A0T0C0C0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_3_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_2:N:0:ACAGTG\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNAAATAAACATTAAACAATAAATAATTAAAAAAATTTATTTAAATTTTTAATTTAAATAAATTAAAAATATTAAATTTATATAAAATTATTTAAAT\t####<BBFFFFFFFFFFFFF<FFFFFF<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFB\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_3_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1059:2153_2:N:0:ACAGTG\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNAAATAAACATTAAACAATAAATAATTAAAAAAATTTATTTAAATTTTTAATTTAAATAAATTAAAAATATTAAATTTATATAAAATTATTTAAAT\t####<BBFFFFFFFFFFFFF<FFFFFF<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFB\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_3_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6460 (64.60%) aligned 0 times\n",
      "    1466 (14.66%) aligned exactly 1 time\n",
      "    2074 (20.74%) aligned >1 times\n",
      "35.40% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6523 (65.23%) aligned 0 times\n",
      "    1417 (14.17%) aligned exactly 1 time\n",
      "    2060 (20.60%) aligned >1 times\n",
      "34.77% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6037 (60.37%) aligned 0 times\n",
      "    1605 (16.05%) aligned exactly 1 time\n",
      "    2358 (23.58%) aligned >1 times\n",
      "39.63% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6055 (60.55%) aligned 0 times\n",
      "    1600 (16.00%) aligned exactly 1 time\n",
      "    2345 (23.45%) aligned >1 times\n",
      "39.45% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_3_s1_R2.fastq.gz_C_to_T.fastq and zr2096_3_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6387\n",
      "Mapping efficiency:\t63.9%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1162\n",
      "Sequences did not map uniquely:\t2451\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1457\t((converted) top strand)\n",
      "CT/GA:\t1497\t((converted) bottom strand)\n",
      "GA/CT:\t1712\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1721\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t111400\n",
      "\n",
      "Total methylated C's in CpG context:\t12185\n",
      "Total methylated C's in CHG context:\t810\n",
      "Total methylated C's in CHH context:\t2682\n",
      "Total methylated C's in Unknown context:\t401\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3947\n",
      "Total unmethylated C's in CHG context:\t24833\n",
      "Total unmethylated C's in CHH context:\t66943\n",
      "Total unmethylated C's in Unknown context:\t1934\n",
      "\n",
      "C methylated in CpG context:\t75.5%\n",
      "C methylated in CHG context:\t3.2%\n",
      "C methylated in CHH context:\t3.9%\n",
      "C methylated in Unknown context (CN or CHN):\t17.2%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 53s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_4_s1_R1.fastq.gz to zr2096_4_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_4_s1_R1.fastq.gz to zr2096_4_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_4_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_4_s1_R1.fastq.gz_C_to_T.fastq and zr2096_4_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_4_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_1:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNGATTAGTGTATAGTGGTTGTTATGGAAATTAGNNNNNNNNNNNNNNNNNNNNNNNNNTGNANTNTTTTTATTAGAATTGTTAAGGTTTATTTGATTTAT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF#########################<<#<#/#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_4_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_1:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNGATTAGTGTATAGTGGTTGTTATGGAAATTAGNNNNNNNNNNNNNNNNNNNNNNNNNTGNANTNTTTTTATTAGAATTGTTAAGGTTTATTTGATTTAT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF#########################<<#<#/#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_4_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_1:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNAATTAATATATAATAATTATTATAAAAATTAANNNNNNNNNNNNNNNNNNNNNNNNNTANANTNTTTTTATTAAAATTATTAAAATTTATTTAATTTAT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF#########################<<#<#/#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_4_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_1:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNAATTAATATATAATAATTATTATAAAAATTAANNNNNNNNNNNNNNNNNNNNNNNNNTANANTNTTTTTATTAAAATTATTAAAATTTATTTAATTTAT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF#########################<<#<#/#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_4_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6392 (63.92%) aligned 0 times\n",
      "    1501 (15.01%) aligned exactly 1 time\n",
      "    2107 (21.07%) aligned >1 times\n",
      "36.08% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6314 (63.14%) aligned 0 times\n",
      "    1521 (15.21%) aligned exactly 1 time\n",
      "    2165 (21.65%) aligned >1 times\n",
      "36.86% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6344 (63.44%) aligned 0 times\n",
      "    1491 (14.91%) aligned exactly 1 time\n",
      "    2165 (21.65%) aligned >1 times\n",
      "36.56% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6365 (63.65%) aligned 0 times\n",
      "    1496 (14.96%) aligned exactly 1 time\n",
      "    2139 (21.39%) aligned >1 times\n",
      "36.35% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_4_s1_R1.fastq.gz_C_to_T.fastq and zr2096_4_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6169\n",
      "Mapping efficiency:\t61.7%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1547\n",
      "Sequences did not map uniquely:\t2284\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1561\t((converted) top strand)\n",
      "CT/GA:\t1508\t((converted) bottom strand)\n",
      "GA/CT:\t1514\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1586\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t99485\n",
      "\n",
      "Total methylated C's in CpG context:\t8450\n",
      "Total methylated C's in CHG context:\t690\n",
      "Total methylated C's in CHH context:\t2940\n",
      "Total methylated C's in Unknown context:\t390\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4523\n",
      "Total unmethylated C's in CHG context:\t21587\n",
      "Total unmethylated C's in CHH context:\t61295\n",
      "Total unmethylated C's in Unknown context:\t1734\n",
      "\n",
      "C methylated in CpG context:\t65.1%\n",
      "C methylated in CHG context:\t3.1%\n",
      "C methylated in CHH context:\t4.6%\n",
      "C methylated in Unknown context (CN or CHN):\t18.4%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 2m 2s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_4_s1_R2.fastq.gz to zr2096_4_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_4_s1_R2.fastq.gz to zr2096_4_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_4_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_4_s1_R2.fastq.gz_C_to_T.fastq and zr2096_4_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_4_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_2:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNTTTATTATTAATTTTTAAAAAAAAAAAATTATTTTTAATATTAATTTTATAAATTAAATAAATTTTAATAATTTTAATAAAAATATTTTATATAT\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFF/BBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_4_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_2:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNTTTATTATTAATTTTTAAAAAAAAAAAATTATTTTTAATATTAATTTTATAAATTAAATAAATTTTAATAATTTTAATAAAAATATTTTATATAT\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFF/BBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_4_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_2:N:0:GCCAAT\t16\tNC_035785.1_CT_converted\t35734902\t40\t67M1I32M\t*\t0\t0\tATATATGAAGTGTTTTTATTAGAATTGTTAAGGTTTATTTGATTTATAAAATTGATATTAAGGATGGTTTTTTTTTTTTAAAAATTAATGATGGANNNNN\tFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBB/FFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBB#####\tAS:i:-25\tXN:i:0\tXM:i:7\tXO:i:1\tXG:i:1\tNM:i:8\tMD:Z:40A48T4A0G0T0G0A0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_4_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1343:2133_2:N:0:GCCAAT\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNTCCATCATTAATTTTTAAAAAAAAAAAACCATCCTTAATATCAATTTTATAAATCAAATAAACCTTAACAATTCTAATAAAAACACTTCATATAT\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFF/BBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFF\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_4_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6075 (60.75%) aligned 0 times\n",
      "    1601 (16.01%) aligned exactly 1 time\n",
      "    2324 (23.24%) aligned >1 times\n",
      "39.25% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6718 (67.18%) aligned 0 times\n",
      "    1344 (13.44%) aligned exactly 1 time\n",
      "    1938 (19.38%) aligned >1 times\n",
      "32.82% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6636 (66.36%) aligned 0 times\n",
      "    1410 (14.10%) aligned exactly 1 time\n",
      "    1954 (19.54%) aligned >1 times\n",
      "33.64% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6038 (60.38%) aligned 0 times\n",
      "    1616 (16.16%) aligned exactly 1 time\n",
      "    2346 (23.46%) aligned >1 times\n",
      "39.62% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_4_s1_R2.fastq.gz_C_to_T.fastq and zr2096_4_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6188\n",
      "Mapping efficiency:\t61.9%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1494\n",
      "Sequences did not map uniquely:\t2318\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1384\t((converted) top strand)\n",
      "CT/GA:\t1382\t((converted) bottom strand)\n",
      "GA/CT:\t1716\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1706\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t99874\n",
      "\n",
      "Total methylated C's in CpG context:\t8463\n",
      "Total methylated C's in CHG context:\t657\n",
      "Total methylated C's in CHH context:\t2768\n",
      "Total methylated C's in Unknown context:\t385\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4531\n",
      "Total unmethylated C's in CHG context:\t21673\n",
      "Total unmethylated C's in CHH context:\t61782\n",
      "Total unmethylated C's in Unknown context:\t1781\n",
      "\n",
      "C methylated in CpG context:\t65.1%\n",
      "C methylated in CHG context:\t2.9%\n",
      "C methylated in CHH context:\t4.3%\n",
      "C methylated in Unknown context (CN or CHN):\t17.8%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 2m 11s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_5_s1_R1.fastq.gz to zr2096_5_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_5_s1_R1.fastq.gz to zr2096_5_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_5_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNAATAAAATATTATTTAAATTTTATTATTTATANNNNNNNNNNNNNNNNNNATTNTTTTTATTATATATATAAAATTATTTTAAAATTATTATATTTTTT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNAATAAAATATTATTTAAATTTTATTATTTATANNNNNNNNNNNNNNNNNNATTNTTTTTATTATATATATAAAATTATTTTAAAATTATTATATTTTTT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNAACAAAACACCACTCAAATCTTACTATCCACANNNNNNNNNNNNNNNNNNATTNCTTCTATCATATATATAAAATCATTCCAAAATTACTATATTTTTC\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNAACAAAACACCACTCAAATCTTACTATCCACANNNNNNNNNNNNNNNNNNATTNCTTCTATCATATATATAAAATCATTCCAAAATTACTATATTTTTC\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_5_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6392 (63.92%) aligned 0 times\n",
      "    1534 (15.34%) aligned exactly 1 time\n",
      "    2074 (20.74%) aligned >1 times\n",
      "36.08% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6280 (62.80%) aligned 0 times\n",
      "    1608 (16.08%) aligned exactly 1 time\n",
      "    2112 (21.12%) aligned >1 times\n",
      "37.20% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6720 (67.20%) aligned 0 times\n",
      "    1360 (13.60%) aligned exactly 1 time\n",
      "    1920 (19.20%) aligned >1 times\n",
      "32.80% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6718 (67.18%) aligned 0 times\n",
      "    1333 (13.33%) aligned exactly 1 time\n",
      "    1949 (19.49%) aligned >1 times\n",
      "32.82% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6164\n",
      "Mapping efficiency:\t61.6%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1621\n",
      "Sequences did not map uniquely:\t2215\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1435\t((converted) top strand)\n",
      "CT/GA:\t1456\t((converted) bottom strand)\n",
      "GA/CT:\t1690\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1583\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t103425\n",
      "\n",
      "Total methylated C's in CpG context:\t8677\n",
      "Total methylated C's in CHG context:\t799\n",
      "Total methylated C's in CHH context:\t2907\n",
      "Total methylated C's in Unknown context:\t450\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4421\n",
      "Total unmethylated C's in CHG context:\t21895\n",
      "Total unmethylated C's in CHH context:\t64726\n",
      "Total unmethylated C's in Unknown context:\t1840\n",
      "\n",
      "C methylated in CpG context:\t66.2%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.3%\n",
      "C methylated in Unknown context (CN or CHN):\t19.7%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 2m 21s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_5_s1_R2.fastq.gz to zr2096_5_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_5_s1_R2.fastq.gz to zr2096_5_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_5_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_5_s1_R2.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_5_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNNNTTGAGTATATTGAAGAGGAATTATTGATAGTTTTTGTTGAAAAATATAGTAATTTTGGAATGATTTTATATATATGATAGAAGAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_5_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC\t16\tNC_035782.1_GA_converted\t24522137\t12\t99M\t*\t0\t0\tATTATTATTATTTCTTCTATCATATATATAAAATCATTCCAAAATTACTATATTTTTCAACAAAAACTATCAATAATTCCTCTTCAATATACTCAANNN\tBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB<<###\tAS:i:-3\tXS:i:-20\tXN:i:0\tXM:i:3\tXO:i:0\tXG:i:0\tNM:i:3\tMD:Z:96A0C0T0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_5_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNNNTCAAATATATTAAAAAAAAATTATTAATAATTTTCATTAAAAAATATAATAATTTTAAAATAATTTTATATATATAATAAAAAAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_5_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC\t4\t*\t0\t0\t*\t*\t0\t0\tNNNTCAAATATATTAAAAAAAAATTATTAATAATTTTCATTAAAAAATATAATAATTTTAAAATAATTTTATATATATAATAAAAAAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_5_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6196 (61.96%) aligned 0 times\n",
      "    1538 (15.38%) aligned exactly 1 time\n",
      "    2266 (22.66%) aligned >1 times\n",
      "38.04% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6166 (61.66%) aligned 0 times\n",
      "    1629 (16.29%) aligned exactly 1 time\n",
      "    2205 (22.05%) aligned >1 times\n",
      "38.34% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6737 (67.37%) aligned 0 times\n",
      "    1376 (13.76%) aligned exactly 1 time\n",
      "    1887 (18.87%) aligned >1 times\n",
      "32.63% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6782 (67.82%) aligned 0 times\n",
      "    1374 (13.74%) aligned exactly 1 time\n",
      "    1844 (18.44%) aligned >1 times\n",
      "32.18% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_5_s1_R2.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6202\n",
      "Mapping efficiency:\t62.0%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1537\n",
      "Sequences did not map uniquely:\t2261\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1394\t((converted) top strand)\n",
      "CT/GA:\t1366\t((converted) bottom strand)\n",
      "GA/CT:\t1706\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1736\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t103940\n",
      "\n",
      "Total methylated C's in CpG context:\t8868\n",
      "Total methylated C's in CHG context:\t757\n",
      "Total methylated C's in CHH context:\t2856\n",
      "Total methylated C's in Unknown context:\t433\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4481\n",
      "Total unmethylated C's in CHG context:\t22166\n",
      "Total unmethylated C's in CHH context:\t64812\n",
      "Total unmethylated C's in Unknown context:\t1860\n",
      "\n",
      "C methylated in CpG context:\t66.4%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.2%\n",
      "C methylated in Unknown context (CN or CHN):\t18.9%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 2m 29s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_6_s1_R1.fastq.gz to zr2096_6_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_6_s1_R1.fastq.gz to zr2096_6_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_6_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_6_s1_R1.fastq.gz_C_to_T.fastq and zr2096_6_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_6_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_1:N:0:CTTGTA\t4\t*\t0\t0\t*\t*\t0\t0\tNTTATTAAATTTAATTTTATTATTAAATAAAAANNNNNNNNNNNNNNNNNNNNNNANNTANANAAATATTAATTAAATATTTTTAAAATAATAAATTTAA\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF######################<##<<#<#<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_6_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_1:N:0:CTTGTA\t4\t*\t0\t0\t*\t*\t0\t0\tNTTATTAAATTTAATTTTATTATTAAATAAAAANNNNNNNNNNNNNNNNNNNNNNANNTANANAAATATTAATTAAATATTTTTAAAATAATAAATTTAA\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF######################<##<<#<#<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_6_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_1:N:0:CTTGTA\t4\t*\t0\t0\t*\t*\t0\t0\tNTCATCAAATCCAACTTTATCACCAAATAAAAANNNNNNNNNNNNNNNNNNNNNNANNCANANAAACATCAATCAAACATCTCTAAAACAATAAACTTAA\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF######################<##<<#<#<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_6_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_1:N:0:CTTGTA\t4\t*\t0\t0\t*\t*\t0\t0\tNTCATCAAATCCAACTTTATCACCAAATAAAAANNNNNNNNNNNNNNNNNNNNNNANNCANANAAACATCAATCAAACATCTCTAAAACAATAAACTTAA\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF######################<##<<#<#<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\tYF:Z:NS\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_6_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6340 (63.40%) aligned 0 times\n",
      "    1545 (15.45%) aligned exactly 1 time\n",
      "    2115 (21.15%) aligned >1 times\n",
      "36.60% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6302 (63.02%) aligned 0 times\n",
      "    1544 (15.44%) aligned exactly 1 time\n",
      "    2154 (21.54%) aligned >1 times\n",
      "36.98% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6263 (62.63%) aligned 0 times\n",
      "    1487 (14.87%) aligned exactly 1 time\n",
      "    2250 (22.50%) aligned >1 times\n",
      "37.37% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6326 (63.26%) aligned 0 times\n",
      "    1524 (15.24%) aligned exactly 1 time\n",
      "    2150 (21.50%) aligned >1 times\n",
      "36.74% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_6_s1_R1.fastq.gz_C_to_T.fastq and zr2096_6_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6310\n",
      "Mapping efficiency:\t63.1%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1354\n",
      "Sequences did not map uniquely:\t2336\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1594\t((converted) top strand)\n",
      "CT/GA:\t1635\t((converted) bottom strand)\n",
      "GA/CT:\t1540\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1541\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t102194\n",
      "\n",
      "Total methylated C's in CpG context:\t10011\n",
      "Total methylated C's in CHG context:\t703\n",
      "Total methylated C's in CHH context:\t2760\n",
      "Total methylated C's in Unknown context:\t409\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3765\n",
      "Total unmethylated C's in CHG context:\t22238\n",
      "Total unmethylated C's in CHH context:\t62717\n",
      "Total unmethylated C's in Unknown context:\t1860\n",
      "\n",
      "C methylated in CpG context:\t72.7%\n",
      "C methylated in CHG context:\t3.1%\n",
      "C methylated in CHH context:\t4.2%\n",
      "C methylated in Unknown context (CN or CHN):\t18.0%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 2m 38s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_6_s1_R2.fastq.gz to zr2096_6_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_6_s1_R2.fastq.gz to zr2096_6_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_6_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_6_s1_R2.fastq.gz_C_to_T.fastq and zr2096_6_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_6_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_2:N:0:CTTGTA\t0\tNC_035787.1_CT_converted\t44054397\t23\t30M1D1M1D69M\t*\t0\t0\tNNNNNTAAGAATTTTTTTAAGTTTATTGTTTTAGAGATGTTTGATTGATGTTTATTTGTATTTTGTAATTAGTGGTGGTTGTTTTTTTATTTGGTGATAA\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFBFFFFBFFFFB/\tAS:i:-57\tXN:i:0\tXM:i:11\tXO:i:2\tXG:i:2\tNM:i:13\tMD:Z:0A0G0A0T0G25^A1^G4T28A6A1A24T0G0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_6_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_2:N:0:CTTGTA\t16\tNC_035788.1_GA_converted\t79269189\t8\t4M2I64M2D30M\t*\t0\t0\tTTATCACCAAATAAAAAAACAACCACCACTAATTACAAAATACAAATAAACATCAATCAAACATCTCTAAAACAATAAACTTAAAAAAATTCTTANNNNN\t/BFFFFBFFFFBFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBB#####\tAS:i:-63\tXN:i:0\tXM:i:11\tXO:i:2\tXG:i:4\tNM:i:15\tMD:Z:2C4T16T1T6T34^AT2T22C0C0T0T0T0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_6_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_2:N:0:CTTGTA\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNTAAAAATTTTTTTAAATTTATTATTTTAAAAATATTTAATTAATATTTATTTATATTTTATAATTAACAATAATTATTTTTTTATTTAATAATAA\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFBFFFFBFFFFB/\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_6_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1093:2134_2:N:0:CTTGTA\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNTAAAAATTTTTTTAAATTTATTATTTTAAAAATATTTAATTAATATTTATTTATATTTTATAATTAACAATAATTATTTTTTTATTTAATAATAA\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFBFFFFBFFFFB/\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_6_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    5846 (58.46%) aligned 0 times\n",
      "    1704 (17.04%) aligned exactly 1 time\n",
      "    2450 (24.50%) aligned >1 times\n",
      "41.54% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6601 (66.01%) aligned 0 times\n",
      "    1406 (14.06%) aligned exactly 1 time\n",
      "    1993 (19.93%) aligned >1 times\n",
      "33.99% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    5898 (58.98%) aligned 0 times\n",
      "    1693 (16.93%) aligned exactly 1 time\n",
      "    2409 (24.09%) aligned >1 times\n",
      "41.02% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6638 (66.38%) aligned 0 times\n",
      "    1381 (13.81%) aligned exactly 1 time\n",
      "    1981 (19.81%) aligned >1 times\n",
      "33.62% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_6_s1_R2.fastq.gz_C_to_T.fastq and zr2096_6_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6416\n",
      "Mapping efficiency:\t64.2%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1253\n",
      "Sequences did not map uniquely:\t2331\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1371\t((converted) top strand)\n",
      "CT/GA:\t1388\t((converted) bottom strand)\n",
      "GA/CT:\t1845\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1812\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t103399\n",
      "\n",
      "Total methylated C's in CpG context:\t9835\n",
      "Total methylated C's in CHG context:\t712\n",
      "Total methylated C's in CHH context:\t2742\n",
      "Total methylated C's in Unknown context:\t382\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3932\n",
      "Total unmethylated C's in CHG context:\t22726\n",
      "Total unmethylated C's in CHH context:\t63452\n",
      "Total unmethylated C's in Unknown context:\t1889\n",
      "\n",
      "C methylated in CpG context:\t71.4%\n",
      "C methylated in CHG context:\t3.0%\n",
      "C methylated in CHH context:\t4.1%\n",
      "C methylated in Unknown context (CN or CHN):\t16.8%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 2m 52s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_7_s1_R1.fastq.gz to zr2096_7_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_7_s1_R1.fastq.gz to zr2096_7_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_7_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_7_s1_R1.fastq.gz_C_to_T.fastq and zr2096_7_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_7_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_1:N:0:ATCACG\t4\t*\t0\t0\t*\t*\t0\t0\tAAAAAAATGATTATTATGATGAATTTAAATATTAATAATAAAATTTAAAAAAAATATAAATGTTAAATATTAAATTTTTTAAAAAATGATTATGG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_7_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_1:N:0:ATCACG\t4\t*\t0\t0\t*\t*\t0\t0\tAAAAAAATGATTATTATGATGAATTTAAATATTAATAATAAAATTTAAAAAAAATATAAATGTTAAATATTAAATTTTTTAAAAAATGATTATGG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_7_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_1:N:0:ATCACG\t16\tNC_035780.1_CT_converted\t42548681\t17\t95M\t*\t0\t0\tTTGTAATTGTTTTTTGAAGAATTTGATATTTAATGTTTATGTTTTTTTTAAATTTTGTTATTAATATTTGAATTTGTTGTAGTAGTTGTTTTTTT\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-6\tXS:i:-30\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:0A94\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_7_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_1:N:0:ATCACG\t0\tNC_035781.1_GA_converted\t24539746\t6\t95M\t*\t0\t0\tAAAAAAACAACTACTACAACAAATTCAAATATTAATAACAAAATTTAAAAAAAACATAAACATTAAATATCAAATTCTTCAAAAAACAATTACAA\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:-6\tXS:i:-12\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:94T0\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_7_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6284 (62.84%) aligned 0 times\n",
      "    1581 (15.81%) aligned exactly 1 time\n",
      "    2135 (21.35%) aligned >1 times\n",
      "37.16% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6334 (63.34%) aligned 0 times\n",
      "    1541 (15.41%) aligned exactly 1 time\n",
      "    2125 (21.25%) aligned >1 times\n",
      "36.66% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6656 (66.56%) aligned 0 times\n",
      "    1423 (14.23%) aligned exactly 1 time\n",
      "    1921 (19.21%) aligned >1 times\n",
      "33.44% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6721 (67.21%) aligned 0 times\n",
      "    1398 (13.98%) aligned exactly 1 time\n",
      "    1881 (18.81%) aligned >1 times\n",
      "32.79% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_7_s1_R1.fastq.gz_C_to_T.fastq and zr2096_7_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6352\n",
      "Mapping efficiency:\t63.5%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1477\n",
      "Sequences did not map uniquely:\t2171\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1461\t((converted) top strand)\n",
      "CT/GA:\t1520\t((converted) bottom strand)\n",
      "GA/CT:\t1648\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1723\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t107029\n",
      "\n",
      "Total methylated C's in CpG context:\t9550\n",
      "Total methylated C's in CHG context:\t804\n",
      "Total methylated C's in CHH context:\t2977\n",
      "Total methylated C's in Unknown context:\t402\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4133\n",
      "Total unmethylated C's in CHG context:\t23247\n",
      "Total unmethylated C's in CHH context:\t66318\n",
      "Total unmethylated C's in Unknown context:\t1966\n",
      "\n",
      "C methylated in CpG context:\t69.8%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.3%\n",
      "C methylated in Unknown context (CN or CHN):\t17.0%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 3m 1s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_7_s1_R2.fastq.gz to zr2096_7_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_7_s1_R2.fastq.gz to zr2096_7_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_7_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_7_s1_R2.fastq.gz_C_to_T.fastq and zr2096_7_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_7_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_2:N:0:ATCACG\t0\tNC_035780.1_CT_converted\t42548681\t17\t95M\t*\t0\t0\tTTGTAATTGTTTTTTGAAGAATTTGATATTTAATGTTTATGTTTTTTTTAAATTTTGTTATTAATATTTGAATTTGTTGTAGTAGTTGTTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/FFBBFFFFFFBBFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:-6\tXS:i:-30\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:0A94\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_7_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_2:N:0:ATCACG\t16\tNC_035781.1_GA_converted\t24539746\t6\t95M\t*\t0\t0\tAAAAAAACAACTACTACAACAAATTCAAATATTAATAACAAAATTTAAAAAAAACATAAACATTAAATATCAAATTCTTCAAAAAACAATTACAA\tFFFFFFFFFFFFFFFFFFFFFFFBBFFFFFFBBFF/FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-6\tXS:i:-12\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:94T0\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_7_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_2:N:0:ATCACG\t4\t*\t0\t0\t*\t*\t0\t0\tCCATAATCATTTTTTAAAAAATTTAATATTTAACATTTATATTTTTTTTAAATTTTATTATTAATATTTAAATTCATCATAATAATCATTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/FFBBFFFFFFBBFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_7_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1132:2189_2:N:0:ATCACG\t4\t*\t0\t0\t*\t*\t0\t0\tCCATAATCATTTTTTAAAAAATTTAATATTTAACATTTATATTTTTTTTAAATTTTATTATTAATATTTAAATTCATCATAATAATCATTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/FFBBFFFFFFBBFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_7_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6141 (61.41%) aligned 0 times\n",
      "    1674 (16.74%) aligned exactly 1 time\n",
      "    2185 (21.85%) aligned >1 times\n",
      "38.59% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6130 (61.30%) aligned 0 times\n",
      "    1661 (16.61%) aligned exactly 1 time\n",
      "    2209 (22.09%) aligned >1 times\n",
      "38.70% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6817 (68.17%) aligned 0 times\n",
      "    1343 (13.43%) aligned exactly 1 time\n",
      "    1840 (18.40%) aligned >1 times\n",
      "31.83% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6784 (67.84%) aligned 0 times\n",
      "    1365 (13.65%) aligned exactly 1 time\n",
      "    1851 (18.51%) aligned >1 times\n",
      "32.16% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_7_s1_R2.fastq.gz_C_to_T.fastq and zr2096_7_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6387\n",
      "Mapping efficiency:\t63.9%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1433\n",
      "Sequences did not map uniquely:\t2180\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1394\t((converted) top strand)\n",
      "CT/GA:\t1406\t((converted) bottom strand)\n",
      "GA/CT:\t1772\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1815\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t106977\n",
      "\n",
      "Total methylated C's in CpG context:\t9312\n",
      "Total methylated C's in CHG context:\t821\n",
      "Total methylated C's in CHH context:\t2823\n",
      "Total methylated C's in Unknown context:\t424\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4043\n",
      "Total unmethylated C's in CHG context:\t22916\n",
      "Total unmethylated C's in CHH context:\t67062\n",
      "Total unmethylated C's in Unknown context:\t1926\n",
      "\n",
      "C methylated in CpG context:\t69.7%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.0%\n",
      "C methylated in Unknown context (CN or CHN):\t18.0%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 3m 11s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_8_s1_R1.fastq.gz to zr2096_8_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_8_s1_R1.fastq.gz to zr2096_8_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_8_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_8_s1_R1.fastq.gz_C_to_T.fastq and zr2096_8_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_8_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_1:N:0:TTAGGC\t4\t*\t0\t0\t*\t*\t0\t0\tTAATTTAAAATTAAAAAATTTATTAAATTTATTATTATTTNTTATTTTTTATAAAAGAAAATAGAGAGAGAGAGAGAGAGAGAGAGAG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_8_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_1:N:0:TTAGGC\t4\t*\t0\t0\t*\t*\t0\t0\tTAATTTAAAATTAAAAAATTTATTAAATTTATTATTATTTNTTATTTTTTATAAAAGAAAATAGAGAGAGAGAGAGAGAGAGAGAGAG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFF\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_8_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_1:N:0:TTAGGC\t16\tNC_007175.2_CT_converted\t14567\t0\t2M17I69M\t*\t0\t0\tTTTTTTTTTTTTTTTTTTTTTTTTTTATTTTTTTTTGTGAAGAATGGNGAATGATGATAGATTTAATGAGTTTTTTAATTTTGGATTA\tFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<<#FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-81\tXN:i:0\tXM:i:5\tXO:i:1\tXG:i:17\tNM:i:22\tMD:Z:1A5A0G5A15T40\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_8_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_1:N:0:TTAGGC\t4\t*\t0\t0\t*\t*\t0\t0\tTAATCCAAAATTAAAAAACTCATTAAATCTATCATCATTCNCCATTCTTCACAAAAAAAAATAAAAAAAAAAAAAAAAAAAAAAAAAA\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFF\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_8_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    7000 (70.00%) aligned 0 times\n",
      "    1302 (13.02%) aligned exactly 1 time\n",
      "    1698 (16.98%) aligned >1 times\n",
      "30.00% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6987 (69.87%) aligned 0 times\n",
      "    1298 (12.98%) aligned exactly 1 time\n",
      "    1715 (17.15%) aligned >1 times\n",
      "30.13% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6572 (65.72%) aligned 0 times\n",
      "    1442 (14.42%) aligned exactly 1 time\n",
      "    1986 (19.86%) aligned >1 times\n",
      "34.28% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6652 (66.52%) aligned 0 times\n",
      "    1457 (14.57%) aligned exactly 1 time\n",
      "    1891 (18.91%) aligned >1 times\n",
      "33.48% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_8_s1_R1.fastq.gz_C_to_T.fastq and zr2096_8_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t5715\n",
      "Mapping efficiency:\t57.1%\n",
      "\n",
      "Sequences with no alignments under any condition:\t2299\n",
      "Sequences did not map uniquely:\t1986\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1348\t((converted) top strand)\n",
      "CT/GA:\t1329\t((converted) bottom strand)\n",
      "GA/CT:\t1528\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1510\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t94042\n",
      "\n",
      "Total methylated C's in CpG context:\t7628\n",
      "Total methylated C's in CHG context:\t699\n",
      "Total methylated C's in CHH context:\t2957\n",
      "Total methylated C's in Unknown context:\t373\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4076\n",
      "Total unmethylated C's in CHG context:\t20470\n",
      "Total unmethylated C's in CHH context:\t58212\n",
      "Total unmethylated C's in Unknown context:\t1652\n",
      "\n",
      "C methylated in CpG context:\t65.2%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.8%\n",
      "C methylated in Unknown context (CN or CHN):\t18.4%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 3m 19s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_8_s1_R2.fastq.gz to zr2096_8_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_8_s1_R2.fastq.gz to zr2096_8_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_8_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_8_s1_R2.fastq.gz_C_to_T.fastq and zr2096_8_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_8_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_2:N:0:TTAGGC\t0\tNC_007175.2_CT_converted\t14567\t0\t2M17I69M\t*\t0\t0\tTTTTTTTTTTTTTTTTTTTTTTTTTTATTTTTTTTTGTGAAGAATGGTGAATGATGATAGATTTAATGAGTTTTTTAATTTTGGATTA\tBBBBBFFFFFFFFFFFFFFFFFFFFF/FFFF<FFFFBF</F</<<<<FF/<FF/<<</F//<FF</BFF/FFFFFF//<FFFB//BF/\tAS:i:-80\tXN:i:0\tXM:i:4\tXO:i:1\tXG:i:17\tNM:i:21\tMD:Z:1A5A0G5A56\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_8_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_2:N:0:TTAGGC\t4\t*\t0\t0\t*\t*\t0\t0\tTTTTTTTTTTTTTTTTTTTTTTTTTTATTTTTTTTTGTGAAGAATGGTGAATGATGATAGATTTAATGAGTTTTTTAATTTTGGATTA\tBBBBBFFFFFFFFFFFFFFFFFFFFF/FFFF<FFFFBF</F</<<<<FF/<FF/<<</F//<FF</BFF/FFFFFF//<FFFB//BF/\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_8_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_2:N:0:TTAGGC\t4\t*\t0\t0\t*\t*\t0\t0\tCTCTCTCTCTCTCTCTCTCTCTCTCTATTTTCTTTTATAAAAAATAATAAATAATAATAAATTTAATAAATTTTTTAATTTTAAATTA\tBBBBBFFFFFFFFFFFFFFFFFFFFF/FFFF<FFFFBF</F</<<<<FF/<FF/<<</F//<FF</BFF/FFFFFF//<FFFB//BF/\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_8_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:1:1101:1281:2170_2:N:0:TTAGGC\t4\t*\t0\t0\t*\t*\t0\t0\tCTCTCTCTCTCTCTCTCTCTCTCTCTATTTTCTTTTATAAAAAATAATAAATAATAATAAATTTAATAAATTTTTTAATTTTAAATTA\tBBBBBFFFFFFFFFFFFFFFFFFFFF/FFFF<FFFFBF</F</<<<<FF/<FF/<<</F//<FF</BFF/FFFFFF//<FFFB//BF/\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_8_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6568 (65.68%) aligned 0 times\n",
      "    1479 (14.79%) aligned exactly 1 time\n",
      "    1953 (19.53%) aligned >1 times\n",
      "34.32% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6584 (65.84%) aligned 0 times\n",
      "    1452 (14.52%) aligned exactly 1 time\n",
      "    1964 (19.64%) aligned >1 times\n",
      "34.16% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    7071 (70.71%) aligned 0 times\n",
      "    1252 (12.52%) aligned exactly 1 time\n",
      "    1677 (16.77%) aligned >1 times\n",
      "29.29% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6986 (69.86%) aligned 0 times\n",
      "    1225 (12.25%) aligned exactly 1 time\n",
      "    1789 (17.89%) aligned >1 times\n",
      "30.14% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_8_s1_R2.fastq.gz_C_to_T.fastq and zr2096_8_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t5645\n",
      "Mapping efficiency:\t56.5%\n",
      "\n",
      "Sequences with no alignments under any condition:\t2318\n",
      "Sequences did not map uniquely:\t2037\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1260\t((converted) top strand)\n",
      "CT/GA:\t1253\t((converted) bottom strand)\n",
      "GA/CT:\t1552\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1580\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t92445\n",
      "\n",
      "Total methylated C's in CpG context:\t7495\n",
      "Total methylated C's in CHG context:\t721\n",
      "Total methylated C's in CHH context:\t2989\n",
      "Total methylated C's in Unknown context:\t372\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3847\n",
      "Total unmethylated C's in CHG context:\t19920\n",
      "Total unmethylated C's in CHH context:\t57473\n",
      "Total unmethylated C's in Unknown context:\t1544\n",
      "\n",
      "C methylated in CpG context:\t66.1%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.9%\n",
      "C methylated in Unknown context (CN or CHN):\t19.4%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 3m 29s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_9_s1_R1.fastq.gz to zr2096_9_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_9_s1_R1.fastq.gz to zr2096_9_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_9_s1_R1.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_9_s1_R1.fastq.gz_C_to_T.fastq and zr2096_9_s1_R1.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_9_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_1:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNATTAAATAAAAAATAATAAATAATAAATTANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTATAGG\t#/<<</<<B<<B<FF<BBFFBFFFF<B/FFF####################################<<<<FBF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_9_s1_R1.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_1:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNATTAAATAAAAAATAATAAATAATAAATTANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTATAGG\t#/<<</<<B<<B<FF<BBFFBFFFF<B/FFF####################################<<<<FBF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_9_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_1:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNACCAAATAAAAAATAACAAATAATAAATCANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTATAAA\t#/<<</<<B<<B<FF<BBFFBFFFF<B/FFF####################################<<<<FBF\tYT:Z:UU\tYF:Z:NS\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_9_s1_R1.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_1:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNACCAAATAAAAAATAACAAATAATAAATCANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTATAAA\t#/<<</<<B<<B<FF<BBFFBFFFF<B/FFF####################################<<<<FBF\tYT:Z:UU\tYF:Z:NS\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_9_s1_R1_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6591 (65.91%) aligned 0 times\n",
      "    1449 (14.49%) aligned exactly 1 time\n",
      "    1960 (19.60%) aligned >1 times\n",
      "34.09% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6519 (65.19%) aligned 0 times\n",
      "    1458 (14.58%) aligned exactly 1 time\n",
      "    2023 (20.23%) aligned >1 times\n",
      "34.81% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6584 (65.84%) aligned 0 times\n",
      "    1508 (15.08%) aligned exactly 1 time\n",
      "    1908 (19.08%) aligned >1 times\n",
      "34.16% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6555 (65.55%) aligned 0 times\n",
      "    1510 (15.10%) aligned exactly 1 time\n",
      "    1935 (19.35%) aligned >1 times\n",
      "34.45% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_9_s1_R1.fastq.gz_C_to_T.fastq and zr2096_9_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6115\n",
      "Mapping efficiency:\t61.1%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1660\n",
      "Sequences did not map uniquely:\t2225\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1493\t((converted) top strand)\n",
      "CT/GA:\t1555\t((converted) bottom strand)\n",
      "GA/CT:\t1592\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1475\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t108734\n",
      "\n",
      "Total methylated C's in CpG context:\t11677\n",
      "Total methylated C's in CHG context:\t923\n",
      "Total methylated C's in CHH context:\t3066\n",
      "Total methylated C's in Unknown context:\t462\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3948\n",
      "Total unmethylated C's in CHG context:\t23839\n",
      "Total unmethylated C's in CHH context:\t65281\n",
      "Total unmethylated C's in Unknown context:\t1927\n",
      "\n",
      "C methylated in CpG context:\t74.7%\n",
      "C methylated in CHG context:\t3.7%\n",
      "C methylated in CHH context:\t4.5%\n",
      "C methylated in Unknown context (CN or CHN):\t19.3%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 3m 37s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Single-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "Input file is in FastQ format\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz: uncompress failed\n",
      "Processing reads up to sequence no. 10000 from /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\n",
      "Writing a C -> T converted version of the input file zr2096_9_s1_R2.fastq.gz to zr2096_9_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_9_s1_R2.fastq.gz to zr2096_9_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_9_s1_R2.fastq.gz (10001 sequences in total)\n",
      "\n",
      "Input files are zr2096_9_s1_R2.fastq.gz_C_to_T.fastq and zr2096_9_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ with the specified options: -q --score-min L,0,-1.2 --ignore-quals\n",
      "\n",
      "Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from zr2096_9_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_2:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNAANTAAGAGGATAGGAGATATGTTAGGAATTTATAATTTGATTTATTATTTGTTATTTTTTATTTGGTTAGATTNNNNGAGTGTTGTGTAGGGAA\t#####<<#</BFFFB/FBFBF<FFFFF/FBFF//<FFF/</<FFF//</<<B/FB//FB/<FFFFF/B/F/FFFFB///####<</FFF<FBFF/F</7B\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from zr2096_9_s1_R2.fastq.gz_C_to_T.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_2:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNAANTAAGAGGATAGGAGATATGTTAGGAATTTATAATTTGATTTATTATTTGTTATTTTTTATTTGGTTAGATTNNNNGAGTGTTGTGTAGGGAA\t#####<<#</BFFFB/FBFBF<FFFFF/FBFF//<FFF/</<FFF//</<<B/FB//FB/<FFFFF/B/F/FFFFB///####<</FFF<FBFF/F</7B\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadCTgenome (reading in sequences from zr2096_9_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --nofw)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/CT_conversion/BS_CT\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_2:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNAANTAAAAAAATAAAAAACACATTAAAAATCTATAATTTAATTTATTATTTATTATTTTTTATTTAATTAAATCNNNNAAACATCATATAAAAAA\t#####<<#</BFFFB/FBFBF<FFFFF/FBFF//<FFF/</<FFF//</<<B/FB//FB/<FFFFF/B/F/FFFFB///####<</FFF<FBFF/F</7B\tYT:Z:UU\n",
      "Now starting the Bowtie 2 aligner for GAreadGAgenome (reading in sequences from zr2096_9_s1_R2.fastq.gz_G_to_A.fastq with options -q --score-min L,0,-1.2 --ignore-quals --norc)\n",
      "Using Bowtie 2 index: /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/Bisulfite_Genome/GA_conversion/BS_GA\n",
      "\n",
      "Found first alignment:\tHWI-C00124:321:CC781ANXX:2:1101:1135:2128_2:N:0:ACTTGA\t4\t*\t0\t0\t*\t*\t0\t0\tNNNNNAANTAAAAAAATAAAAAACACATTAAAAATCTATAATTTAATTTATTATTTATTATTTTTTATTTAATTAAATCNNNNAAACATCATATAAAAAA\t#####<<#</BFFFB/FBFBF<FFFFF/FBFF//<FFF/</<FFF//</<<B/FB//FB/<FFFFF/B/F/FFFFB///####<</FFF<FBFF/F</7B\tYT:Z:UU\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_9_s1_R2_bismark_bt2.bam <<<\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz: uncompress failed\n",
      "\n",
      "Reading in the sequence file /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6944 (69.44%) aligned 0 times\n",
      "    1318 (13.18%) aligned exactly 1 time\n",
      "    1738 (17.38%) aligned >1 times\n",
      "30.56% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    5945 (59.45%) aligned 0 times\n",
      "    1771 (17.71%) aligned exactly 1 time\n",
      "    2284 (22.84%) aligned >1 times\n",
      "40.55% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    6996 (69.96%) aligned 0 times\n",
      "    1297 (12.97%) aligned exactly 1 time\n",
      "    1707 (17.07%) aligned >1 times\n",
      "30.04% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were unpaired; of these:\n",
      "    5921 (59.21%) aligned 0 times\n",
      "    1780 (17.80%) aligned exactly 1 time\n",
      "    2299 (22.99%) aligned >1 times\n",
      "40.79% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_9_s1_R2.fastq.gz_C_to_T.fastq and zr2096_9_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequences analysed in total:\t10000\n",
      "Number of alignments with a unique best hit from the different alignments:\t6280\n",
      "Mapping efficiency:\t62.8%\n",
      "\n",
      "Sequences with no alignments under any condition:\t1379\n",
      "Sequences did not map uniquely:\t2341\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t1316\t((converted) top strand)\n",
      "CT/GA:\t1217\t((converted) bottom strand)\n",
      "GA/CT:\t1859\t(complementary to (converted) top strand)\n",
      "GA/GA:\t1888\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t111835\n",
      "\n",
      "Total methylated C's in CpG context:\t11862\n",
      "Total methylated C's in CHG context:\t885\n",
      "Total methylated C's in CHH context:\t3081\n",
      "Total methylated C's in Unknown context:\t434\n",
      "\n",
      "Total unmethylated C's in CpG context:\t3985\n",
      "Total unmethylated C's in CHG context:\t24536\n",
      "Total unmethylated C's in CHH context:\t67486\n",
      "Total unmethylated C's in Unknown context:\t1932\n",
      "\n",
      "C methylated in CpG context:\t74.9%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.4%\n",
      "C methylated in Unknown context (CN or CHN):\t18.3%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 3m 46s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark \\\n",
    "-u 10000 \\\n",
    "--non_directional \\\n",
    "--score_min L,0,-1.2 \\\n",
    "--genome 2018-04-27-Bismark-Inputs/ \\\n",
    "/Volumes/web/nightingales/C_virginica/zr2096_* \\"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "I got a [`gunzip` error](https://yaaminiv.github.io/Gonad-Methylation-Analysis-Part5/) at the end of each Bismark alignment. [Steven said to continue](https://github.com/RobertsLab/resources/issues/236) since I still got output files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Methylation Extractor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The first thing I want to do is create a new directory to store any outputs form this step."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-04-29-Methylation-Extractor-Output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-04-27-Bismark-Inputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Methylation-Extractor-Output\u001b[m\u001b[m/\r\n",
      "zr2096_10_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_10_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R2_bismark_bt2_SE_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "The following is a brief description of all options to control the Bismark\r\n",
      "methylation extractor. The script reads in a bisulfite read alignment results file \r\n",
      "produced by the Bismark bisulfite mapper (in BAM/CRAM/SAM format) and extracts the\r\n",
      "methylation information for individual cytosines. This information is found in the\r\n",
      "methylation call field which can contain the following characters:\r\n",
      "\r\n",
      "       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n",
      "       ~~~   X   for methylated C in CHG context                      ~~~\r\n",
      "       ~~~   x   for not methylated C CHG                             ~~~\r\n",
      "       ~~~   H   for methylated C in CHH context                      ~~~\r\n",
      "       ~~~   h   for not methylated C in CHH context                  ~~~\r\n",
      "       ~~~   Z   for methylated C in CpG context                      ~~~\r\n",
      "       ~~~   z   for not methylated C in CpG context                  ~~~\r\n",
      "       ~~~   U   for methylated C in Unknown context (CN or CHN       ~~~\r\n",
      "       ~~~   u   for not methylated C in Unknown context (CN or CHN)  ~~~\r\n",
      "       ~~~   .   for any bases not involving cytosines                ~~~\r\n",
      "       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n",
      "\r\n",
      "The methylation extractor outputs result files for cytosines in CpG, CHG and CHH\r\n",
      "context (this distinction is actually already made in Bismark itself). As the methylation\r\n",
      "information for every C analysed can produce files which easily have tens or even hundreds of\r\n",
      "millions of lines, file sizes can become very large and more difficult to handle. The C\r\n",
      "methylation info additionally splits cytosine methylation calls up into one of the four possible\r\n",
      "strands a given bisulfite read aligned against:\r\n",
      "\r\n",
      "             OT      original top strand\r\n",
      "             CTOT    complementary to original top strand\r\n",
      "\r\n",
      "             OB      original bottom strand\r\n",
      "             CTOB    complementary to original bottom strand\r\n",
      "\r\n",
      "Thus, by default twelve individual output files are being generated per input file (unless\r\n",
      "--comprehensive is specified, see below). The output files can be imported into a genome\r\n",
      "viewer, such as SeqMonk, and re-combined into a single data group if desired (in fact\r\n",
      "unless the bisulfite reads were generated preserving directionality it doesn't make any\r\n",
      "sense to look at the data in a strand-specific manner). Strand-specific output files can\r\n",
      "optionally be skipped, in which case only three output files for CpG, CHG or CHH context\r\n",
      "will be generated. For both the strand-specific and comprehensive outputs there is also\r\n",
      "the option to merge both non-CpG contexts (CHG and CHH) into one single non-CpG context.\r\n",
      "\r\n",
      "\r\n",
      "The output files are in the following format (tab delimited):\r\n",
      "\r\n",
      "<sequence_id>     <strand>      <chromosome>     <position>     <methylation call>\r\n",
      "\r\n",
      "\r\n",
      "USAGE: bismark_methylation_extractor [options] <filenames>\r\n",
      "\r\n",
      "\r\n",
      "ARGUMENTS:\r\n",
      "==========\r\n",
      "\r\n",
      "<filenames>              A space-separated list of Bismark result files in SAM format from\r\n",
      "                         which methylation information is extracted for every cytosine in\r\n",
      "                         the reads. For alignment files in the older custom Bismark output\r\n",
      "                         see option '--vanilla'.\r\n",
      "\r\n",
      "OPTIONS:\r\n",
      "\r\n",
      "-s/--single-end          Input file(s) are Bismark result file(s) generated from single-end\r\n",
      "                         read data. If neither -s nor -p is set the type of experiment will\r\n",
      "                         be determined automatically.\r\n",
      "\r\n",
      "-p/--paired-end          Input file(s) are Bismark result file(s) generated from paired-end\r\n",
      "                         read data. If neither -s nor -p is set the type of experiment will\r\n",
      "                         be determined automatically.\r\n",
      "\r\n",
      "--vanilla                The Bismark result input file(s) are in the old custom Bismark format\r\n",
      "                         (up to version 0.5.x) and not in SAM format which is the default as\r\n",
      "                         of Bismark version 0.6.x or higher. Default: OFF.\r\n",
      "\r\n",
      "--no_overlap             For paired-end reads it is theoretically possible that read_1 and\r\n",
      "                         read_2 overlap. This option avoids scoring overlapping methylation\r\n",
      "                         calls twice (only methylation calls of read 1 are used for in the process\r\n",
      "                         since read 1 has historically higher quality basecalls than read 2).\r\n",
      "                         Whilst this option removes a bias towards more methylation calls\r\n",
      "                         in the center of sequenced fragments it may de facto remove a sizable\r\n",
      "                         proportion of the data. This option is on by default for paired-end data\r\n",
      "                         but can be disabled using --include_overlap. Default: ON.\r\n",
      "\r\n",
      "--include_overlap        For paired-end data all methylation calls will be extracted irrespective of\r\n",
      "                         of whether they overlap or not. Default: OFF.\r\n",
      "\r\n",
      "--ignore <int>           Ignore the first <int> bp from the 5' end of Read 1 (or single-end alignment\r\n",
      "                         files) when processing the methylation call string. This can remove e.g. a\r\n",
      "                         restriction enzyme site at the start of each read or any other source of\r\n",
      "                         bias (such as PBAT-Seq data).\r\n",
      "\r\n",
      "--ignore_r2 <int>        Ignore the first <int> bp from the 5' end of Read 2 of paired-end sequencing\r\n",
      "                         results only. Since the first couple of bases in Read 2 of BS-Seq experiments\r\n",
      "                         show a severe bias towards non-methylation as a result of end-repairing\r\n",
      "                         sonicated fragments with unmethylated cytosines (see M-bias plot), it is\r\n",
      "                         recommended that the first couple of bp of Read 2 are removed before\r\n",
      "                         starting downstream analysis. Please see the section on M-bias plots in the\r\n",
      "                         Bismark User Guide for more details.\r\n",
      "\r\n",
      "--ignore_3prime <int>    Ignore the last <int> bp from the 3' end of Read 1 (or single-end alignment\r\n",
      "                         files) when processing the methylation call string. This can remove unwanted\r\n",
      "                         biases from the end of reads.\r\n",
      "\r\n",
      "--ignore_3prime_r2 <int> Ignore the last <int> bp from the 3' end of Read 2 of paired-end sequencing\r\n",
      "                         results only. This can remove unwanted biases from the end of reads.\r\n",
      "\r\n",
      "--comprehensive          Specifying this option will merge all four possible strand-specific\r\n",
      "                         methylation info into context-dependent output files. The default\r\n",
      "\r\n",
      "                         contexts are:\r\n",
      "                          - CpG context\r\n",
      "                          - CHG context\r\n",
      "                          - CHH context\r\n",
      "\r\n",
      "--merge_non_CpG          This will produce two output files (in --comprehensive mode) or eight\r\n",
      "                         strand-specific output files (default) for Cs in\r\n",
      "                          - CpG context\r\n",
      "                          - non-CpG context\r\n",
      "\r\n",
      "--report                 Prints out a short methylation summary as well as the paramaters used to run\r\n",
      "                         this script. Default: ON.\r\n",
      "\r\n",
      "--no_header              Suppresses the Bismark version header line in all output files for more convenient\r\n",
      "                         batch processing.\r\n",
      "\r\n",
      "-o/--output DIR          Allows specification of a different output directory (absolute or relative\r\n",
      "                         path). If not specified explicitly, the output will be written to the current directory.\r\n",
      "\r\n",
      "--samtools_path          The path to your Samtools installation, e.g. /home/user/samtools/. Does not need to be specified\r\n",
      "                         explicitly if Samtools is in the PATH already.\r\n",
      "\r\n",
      "--gzip                   The methylation extractor files (CpG_OT_..., CpG_OB_... etc) will be written out in\r\n",
      "                         a GZIP compressed form to save disk space. This option is also passed on to the genome-wide\r\n",
      "                         cytosine report. BedGraph and coverage files will be written out as .gz by default.\r\n",
      "\r\n",
      "--version                Displays version information.\r\n",
      "\r\n",
      "-h/--help                Displays this help file and exits.\r\n",
      "\r\n",
      "--mbias_only             The methylation extractor will read the entire file but only output the M-bias table and plots as \r\n",
      "                         well as a report (optional) and then quit. Default: OFF.\r\n",
      "\r\n",
      "--mbias_off              The methylation extractor will process the entire file as usual but doesn't write out any M-bias report.\r\n",
      "                         Only recommended for users who deliberately want to keep an earlier version of the M-bias report. \r\n",
      "                         Default: OFF.\r\n",
      "\r\n",
      "--parallel <int>         May also be --multicore <int>. Sets the number of cores to be used for the methylation extraction process.\r\n",
      "                         If system resources are plentiful this is a viable option to speed up the extraction process (we observed a\r\n",
      "                         near linear speed increase for up to 10 cores used). Please note that a typical process of extracting a BAM file\r\n",
      "                         and writing out '.gz' output streams will in fact use ~3 cores per value of --parallel <int>\r\n",
      "                         specified (1 for the methylation extractor itself, 1 for a Samtools stream, 1 for GZIP stream), so\r\n",
      "                         --parallel 10 is likely to use around 30 cores of system resources. This option has no bearing\r\n",
      "                         on the bismark2bedGraph or genome-wide cytosine report processes.\r\n",
      "\r\n",
      "--yacht                  This option (Yet Another Context Hunting Tool) writes out additional information about the read a methylation\r\n",
      "                         call belongs to, and its output is meant to be fed into the NOMe_filtering script. This option writes out\r\n",
      "                         a single 'any_C_context' file that contains all methylation calls for a read consecutively. Its intended use\r\n",
      "                         is single-cell NOMe-Seq data, and thus this option works only in single-end mode (paired-end reads often suffer\r\n",
      "                         from chimaera problems...)\r\n",
      "\r\n",
      "                         --yacht will add three additional columns to the standard methylation call files:\r\n",
      "                                    <read start>  <read end>  <read orientation>\r\n",
      "                         For forward reads (+ orientation) the start position is the left-most position wheras for reverse reads\r\n",
      "                         (- orientation) it is the rightmost position.\r\n",
      "\r\n",
      "\r\n",
      "bedGraph specific options:\r\n",
      "==========================\r\n",
      "\r\n",
      "--bedGraph               After finishing the methylation extraction, the methylation output is written into a\r\n",
      "                         sorted bedGraph file that reports the position of a given cytosine and its methylation \r\n",
      "                         state (in %, see details below). The methylation extractor output is temporarily split up into\r\n",
      "                         temporary files, one per chromosome (written into the current directory or folder\r\n",
      "                         specified with -o/--output); these temp files are then used for sorting and deleted\r\n",
      "                         afterwards. By default, only cytosines in CpG context will be sorted. The option\r\n",
      "                         '--CX_context' may be used to report all cytosines irrespective of sequence context\r\n",
      "                         (this will take MUCH longer!). The default folder for temporary files during the sorting\r\n",
      "                         process is the output directory. The bedGraph conversion step is performed by the external\r\n",
      "                         module 'bismark2bedGraph'; this script needs to reside in the same folder as the \r\n",
      "                         bismark_methylation_extractor itself.\r\n",
      "\r\n",
      "--zero_based             Write out an additional coverage file (ending in .zero.cov) that uses 0-based genomic start\r\n",
      "                         and 1-based genomic end coordinates (zero-based, half-open), like used in the bedGraph file,\r\n",
      "                         instead of using 1-based coordinates throughout. Default: OFF.\r\n",
      "\r\n",
      "\r\n",
      "--cutoff [threshold]     The minimum number of times any methylation state (methylated or unmethylated) has to be seen\r\n",
      "                         for a nucleotide before its methylation percentage is reported. Default: 1.\r\n",
      "\r\n",
      "--remove_spaces          Replaces whitespaces in the sequence ID field with underscores to allow sorting.\r\n",
      "\r\n",
      "--CX/--CX_context        The sorted bedGraph output file contains information on every single cytosine that was covered\r\n",
      "                         in the experiment irrespective of its sequence context. This applies to both forward and\r\n",
      "                         reverse strands. Please be aware that this option may generate large temporary and output files\r\n",
      "                         and may take a long time to sort (up to many hours). Default: OFF.\r\n",
      "                         (i.e. Default = CpG context only).\r\n",
      "\r\n",
      "--buffer_size <string>   This allows you to specify the main memory sort buffer when sorting the methylation information.\r\n",
      "                         Either specify a percentage of physical memory by appending % (e.g. --buffer_size 50%) or\r\n",
      "\t\t\t a multiple of 1024 bytes, e.g. 'K' multiplies by 1024, 'M' by 1048576 and so on for 'T' etc. \r\n",
      "                         (e.g. --buffer_size 20G). For more information on sort type 'info sort' on a command line.\r\n",
      "                         Defaults to 2G.\r\n",
      "\r\n",
      "--scaffolds/--gazillion  Users working with unfinished genomes sporting tens or even hundreds of thousands of\r\n",
      "                         scaffolds/contigs/chromosomes frequently encountered errors with pre-sorting reads to\r\n",
      "                         individual chromosome files. These errors were caused by the operating system's limit\r\n",
      "                         of the number of filehandle that can be written to at any one time (typically 1024; to\r\n",
      "                         find out this limit on Linux, type: ulimit -a).\r\n",
      "                         To bypass the limitation of open filehandles, the option --scaffolds does not pre-sort\r\n",
      "                         methylation calls into individual chromosome files. Instead, all input files are\r\n",
      "                         temporarily merged into a single file (unless there is only a single file), and this\r\n",
      "                         file will then be sorted by both chromosome AND position using the Unix sort command.\r\n",
      "                         Please be aware that this option might take a looooong time to complete, depending on\r\n",
      "                         the size of the input files, and the memory you allocate to this process (see --buffer_size).\r\n",
      "                         Nevertheless, it seems to be working.\r\n",
      "\r\n",
      "--ample_memory           Using this option will not sort chromosomal positions using the UNIX 'sort' command, but will\r\n",
      "                         instead use two arrays to sort methylated and unmethylated calls. This may result in a faster\r\n",
      "                         sorting process of very large files, but this comes at the cost of a larger memory footprint\r\n",
      "                         (two arrays of the length of the largest human chromosome 1 (~250M bp) consume around 16GB\r\n",
      "                         of RAM). Due to overheads in creating and looping through these arrays it seems that it will\r\n",
      "                         actually be *slower* for small files (few million alignments), and we are currently testing at\r\n",
      "                         which point it is advisable to use this option. Note that --ample_memory is not compatible\r\n",
      "                         with options '--scaffolds/--gazillion' (as it requires pre-sorted files to begin with).\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Genome-wide cytosine methylation report specific options:\r\n",
      "=========================================================\r\n",
      "\r\n",
      "--cytosine_report        After the conversion to bedGraph has completed, the option '--cytosine_report' produces a\r\n",
      "                         genome-wide methylation report for all cytosines in the genome. By default, the output uses 1-based\r\n",
      "                         chromosome coordinates (zero-based start coords are optional) and reports CpG context only (all\r\n",
      "                         cytosine context is optional). The output considers all Cs on both forward and reverse strands and\r\n",
      "                         reports their position, strand, trinucleotide content and methylation state (counts are 0 if not\r\n",
      "                         covered). The cytosine report conversion step is performed by the external module\r\n",
      "                         'coverage2cytosine'; this script needs to reside in the same folder as the bismark_methylation_extractor\r\n",
      "                         itself.\r\n",
      "\r\n",
      "--CX/--CX_context        The output file contains information on every single cytosine in the genome irrespective of\r\n",
      "                         its context. This applies to both forward and reverse strands. Please be aware that this will\r\n",
      "                         generate output files with > 1.1 billion lines for a mammalian genome such as human or mouse.\r\n",
      "                         Default: OFF (i.e. Default = CpG context only).\r\n",
      "\r\n",
      "--zero_based             Uses 0-based genomic coordinates instead of 1-based coordinates. Default: OFF.\r\n",
      "\r\n",
      "--genome_folder <path>   Enter the genome folder you wish to use to extract sequences from (full path only). Accepted\r\n",
      "                         formats are FastA files ending with '.fa' or '.fasta'. Specifying a genome folder path is mandatory.\r\n",
      "\r\n",
      "--split_by_chromosome    Writes the output into individual files for each chromosome instead of a single output file. Files\r\n",
      "                         will be named to include the input filename and the chromosome number.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "OUTPUT:\r\n",
      "\r\n",
      "The bismark_methylation_extractor output is in the form:\r\n",
      "========================================================\r\n",
      "<seq-ID>  <methylation state*>  <chromosome>  <start position (= end position)>  <methylation call>\r\n",
      "\r\n",
      "* Methylated cytosines receive a '+' orientation,\r\n",
      "* Unmethylated cytosines receive a '-' orientation.\r\n",
      "\r\n",
      "The bismark_methylation_extractor output with --yacht (optional) specified is in the form:\r\n",
      "==========================================================================================\r\n",
      "<seq-ID>  <methylation state*>  <chromosome>  <start position (= end position)>  <methylation call>  <read start>  <read end>  <read orientation>\r\n",
      "\r",
      "\r\n",
      "* Methylated cytosines receive a '+' orientation,\r\n",
      "* Unmethylated cytosines receive a '-' orientation.\r\n",
      "\r\n",
      "\r\n",
      "The bedGraph output (optional) looks like this (tab-delimited; 0-based start coords, 1-based end coords):\r\n",
      "=========================================================================================================\r\n",
      "\r\n",
      "track type=bedGraph (header line)\r\n",
      "\r\n",
      "<chromosome>  <start position>  <end position>  <methylation percentage>\r\n",
      "\r\n",
      "\r\n",
      "The coverage output looks like this (tab-delimited, 1-based genomic coords; zero-based half-open coordinates available with '--zero_based'):\r\n",
      "============================================================================================================================================\r\n",
      "\r\n",
      "<chromosome>  <start position>  <end position>  <methylation percentage>  <count methylated>  <count non-methylated>\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "The genome-wide cytosine methylation output file is tab-delimited in the following format:\r\n",
      "==========================================================================================\r\n",
      "<chromosome>  <position>  <strand>  <count methylated>  <count non-methylated>  <C-context>  <trinucleotide context>\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "This script was last modified on 05 October 2017.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark_methylation_extractor -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "According to the quick guide, I don't need much to extract methylated cytosines.\n",
    "\n",
    "1. Path to `bismark_methylation_extractor`\n",
    "2. -o + path to the output directory\n",
    "3. --gzip: Compress methylation extractor files to save disk space\n",
    "4. --bedGraph: Create `bedGraph` file with results\n",
    "5. Path to .bam files generated in Step 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " *** Bismark methylation extractor version v0.19.0 ***\n",
      "\n",
      "Trying to determine the type of mapping from the SAM header line of file zr2096_10_s1_R1_bismark_bt2.bam\n",
      "Treating file(s) as single-end data (as extracted from @PG line)\n",
      "\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Bismark single-end SAM format specified (default)\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: 2018-04-29-Methylation-Extractor-Output/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>zr2096_10_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_10_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6363 lines in total\n",
      "Total number of methylation call strings processed: 6363\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t110692\n",
      "\n",
      "Total methylated C's in CpG context:\t10598\n",
      "Total methylated C's in CHG context:\t834\n",
      "Total methylated C's in CHH context:\t2849\n",
      "\n",
      "Total C to T conversions in CpG context:\t5034\n",
      "Total C to T conversions in CHG context:\t24437\n",
      "Total C to T conversions in CHH context:\t66940\n",
      "\n",
      "C methylated in CpG context:\t67.8%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.1%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_10_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_10_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_10_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_10_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_10_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6457 lines in total\n",
      "Total number of methylation call strings processed: 6457\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t111568\n",
      "\n",
      "Total methylated C's in CpG context:\t10678\n",
      "Total methylated C's in CHG context:\t827\n",
      "Total methylated C's in CHH context:\t2954\n",
      "\n",
      "Total C to T conversions in CpG context:\t4977\n",
      "Total C to T conversions in CHG context:\t24762\n",
      "Total C to T conversions in CHH context:\t67370\n",
      "\n",
      "C methylated in CpG context:\t68.2%\n",
      "C methylated in CHG context:\t3.2%\n",
      "C methylated in CHH context:\t4.2%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_10_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_10_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_10_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_CTOT_zr2096_10_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_1_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_1_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 3083 lines in total\n",
      "Total number of methylation call strings processed: 3083\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t50039\n",
      "\n",
      "Total methylated C's in CpG context:\t4195\n",
      "Total methylated C's in CHG context:\t390\n",
      "Total methylated C's in CHH context:\t2635\n",
      "\n",
      "Total C to T conversions in CpG context:\t2109\n",
      "Total C to T conversions in CHG context:\t10153\n",
      "Total C to T conversions in CHH context:\t30557\n",
      "\n",
      "C methylated in CpG context:\t66.5%\n",
      "C methylated in CHG context:\t3.7%\n",
      "C methylated in CHH context:\t7.9%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_10_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_1_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_1_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_1_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_1_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_1_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 3211 lines in total\n",
      "Total number of methylation call strings processed: 3211\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t52598\n",
      "\n",
      "Total methylated C's in CpG context:\t4485\n",
      "Total methylated C's in CHG context:\t417\n",
      "Total methylated C's in CHH context:\t2492\n",
      "\n",
      "Total C to T conversions in CpG context:\t2134\n",
      "Total C to T conversions in CHG context:\t10794\n",
      "Total C to T conversions in CHH context:\t32276\n",
      "\n",
      "C methylated in CpG context:\t67.8%\n",
      "C methylated in CHG context:\t3.7%\n",
      "C methylated in CHH context:\t7.2%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_1_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_1_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_1_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_1_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_2_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_2_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 5451 lines in total\n",
      "Total number of methylation call strings processed: 5451\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t86892\n",
      "\n",
      "Total methylated C's in CpG context:\t8081\n",
      "Total methylated C's in CHG context:\t711\n",
      "Total methylated C's in CHH context:\t3049\n",
      "\n",
      "Total C to T conversions in CpG context:\t3617\n",
      "Total C to T conversions in CHG context:\t19399\n",
      "Total C to T conversions in CHH context:\t52035\n",
      "\n",
      "C methylated in CpG context:\t69.1%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t5.5%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_1_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_2_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_2_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_2_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_2_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_2_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 5481 lines in total\n",
      "Total number of methylation call strings processed: 5481\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t87388\n",
      "\n",
      "Total methylated C's in CpG context:\t8135\n",
      "Total methylated C's in CHG context:\t666\n",
      "Total methylated C's in CHH context:\t3294\n",
      "\n",
      "Total C to T conversions in CpG context:\t3627\n",
      "Total C to T conversions in CHG context:\t19502\n",
      "Total C to T conversions in CHH context:\t52164\n",
      "\n",
      "C methylated in CpG context:\t69.2%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t5.9%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_2_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_2_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_2_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_2_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_3_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_3_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6272 lines in total\n",
      "Total number of methylation call strings processed: 6272\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t109168\n",
      "\n",
      "Total methylated C's in CpG context:\t12011\n",
      "Total methylated C's in CHG context:\t784\n",
      "Total methylated C's in CHH context:\t2675\n",
      "\n",
      "Total C to T conversions in CpG context:\t3734\n",
      "Total C to T conversions in CHG context:\t24189\n",
      "Total C to T conversions in CHH context:\t65775\n",
      "\n",
      "C methylated in CpG context:\t76.3%\n",
      "C methylated in CHG context:\t3.1%\n",
      "C methylated in CHH context:\t3.9%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_2_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_3_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_3_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_3_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_CTOT_zr2096_3_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_3_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_3_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6387 lines in total\n",
      "Total number of methylation call strings processed: 6387\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t111400\n",
      "\n",
      "Total methylated C's in CpG context:\t12185\n",
      "Total methylated C's in CHG context:\t810\n",
      "Total methylated C's in CHH context:\t2682\n",
      "\n",
      "Total C to T conversions in CpG context:\t3947\n",
      "Total C to T conversions in CHG context:\t24833\n",
      "Total C to T conversions in CHH context:\t66943\n",
      "\n",
      "C methylated in CpG context:\t75.5%\n",
      "C methylated in CHG context:\t3.2%\n",
      "C methylated in CHH context:\t3.9%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_3_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_3_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_3_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_3_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_4_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_4_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6169 lines in total\n",
      "Total number of methylation call strings processed: 6169\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t99485\n",
      "\n",
      "Total methylated C's in CpG context:\t8450\n",
      "Total methylated C's in CHG context:\t690\n",
      "Total methylated C's in CHH context:\t2940\n",
      "\n",
      "Total C to T conversions in CpG context:\t4523\n",
      "Total C to T conversions in CHG context:\t21587\n",
      "Total C to T conversions in CHH context:\t61295\n",
      "\n",
      "C methylated in CpG context:\t65.1%\n",
      "C methylated in CHG context:\t3.1%\n",
      "C methylated in CHH context:\t4.6%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_3_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_4_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_4_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_4_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_4_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_4_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6188 lines in total\n",
      "Total number of methylation call strings processed: 6188\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t99874\n",
      "\n",
      "Total methylated C's in CpG context:\t8463\n",
      "Total methylated C's in CHG context:\t657\n",
      "Total methylated C's in CHH context:\t2768\n",
      "\n",
      "Total C to T conversions in CpG context:\t4531\n",
      "Total C to T conversions in CHG context:\t21673\n",
      "Total C to T conversions in CHH context:\t61782\n",
      "\n",
      "C methylated in CpG context:\t65.1%\n",
      "C methylated in CHG context:\t2.9%\n",
      "C methylated in CHH context:\t4.3%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_4_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_4_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_4_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_4_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_5_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_5_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6164 lines in total\n",
      "Total number of methylation call strings processed: 6164\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t103425\n",
      "\n",
      "Total methylated C's in CpG context:\t8677\n",
      "Total methylated C's in CHG context:\t799\n",
      "Total methylated C's in CHH context:\t2907\n",
      "\n",
      "Total C to T conversions in CpG context:\t4421\n",
      "Total C to T conversions in CHG context:\t21895\n",
      "Total C to T conversions in CHH context:\t64726\n",
      "\n",
      "C methylated in CpG context:\t66.2%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.3%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_4_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_5_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_5_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_5_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_5_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_5_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6202 lines in total\n",
      "Total number of methylation call strings processed: 6202\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t103940\n",
      "\n",
      "Total methylated C's in CpG context:\t8868\n",
      "Total methylated C's in CHG context:\t757\n",
      "Total methylated C's in CHH context:\t2856\n",
      "\n",
      "Total C to T conversions in CpG context:\t4481\n",
      "Total C to T conversions in CHG context:\t22166\n",
      "Total C to T conversions in CHH context:\t64812\n",
      "\n",
      "C methylated in CpG context:\t66.4%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.2%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_5_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_5_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_5_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_5_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_6_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_6_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6310 lines in total\n",
      "Total number of methylation call strings processed: 6310\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t102194\n",
      "\n",
      "Total methylated C's in CpG context:\t10011\n",
      "Total methylated C's in CHG context:\t703\n",
      "Total methylated C's in CHH context:\t2760\n",
      "\n",
      "Total C to T conversions in CpG context:\t3765\n",
      "Total C to T conversions in CHG context:\t22238\n",
      "Total C to T conversions in CHH context:\t62717\n",
      "\n",
      "C methylated in CpG context:\t72.7%\n",
      "C methylated in CHG context:\t3.1%\n",
      "C methylated in CHH context:\t4.2%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_5_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_6_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_6_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_6_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_6_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_6_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6416 lines in total\n",
      "Total number of methylation call strings processed: 6416\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t103399\n",
      "\n",
      "Total methylated C's in CpG context:\t9835\n",
      "Total methylated C's in CHG context:\t712\n",
      "Total methylated C's in CHH context:\t2742\n",
      "\n",
      "Total C to T conversions in CpG context:\t3932\n",
      "Total C to T conversions in CHG context:\t22726\n",
      "Total C to T conversions in CHH context:\t63452\n",
      "\n",
      "C methylated in CpG context:\t71.4%\n",
      "C methylated in CHG context:\t3.0%\n",
      "C methylated in CHH context:\t4.1%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_6_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_6_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_6_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_6_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_7_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_7_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6352 lines in total\n",
      "Total number of methylation call strings processed: 6352\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t107029\n",
      "\n",
      "Total methylated C's in CpG context:\t9550\n",
      "Total methylated C's in CHG context:\t804\n",
      "Total methylated C's in CHH context:\t2977\n",
      "\n",
      "Total C to T conversions in CpG context:\t4133\n",
      "Total C to T conversions in CHG context:\t23247\n",
      "Total C to T conversions in CHH context:\t66318\n",
      "\n",
      "C methylated in CpG context:\t69.8%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.3%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_6_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_7_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_7_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_7_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_7_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_7_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6387 lines in total\n",
      "Total number of methylation call strings processed: 6387\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t106977\n",
      "\n",
      "Total methylated C's in CpG context:\t9312\n",
      "Total methylated C's in CHG context:\t821\n",
      "Total methylated C's in CHH context:\t2823\n",
      "\n",
      "Total C to T conversions in CpG context:\t4043\n",
      "Total C to T conversions in CHG context:\t22916\n",
      "Total C to T conversions in CHH context:\t67062\n",
      "\n",
      "C methylated in CpG context:\t69.7%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.0%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_7_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_7_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_7_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_7_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_8_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_8_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 5715 lines in total\n",
      "Total number of methylation call strings processed: 5715\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t94042\n",
      "\n",
      "Total methylated C's in CpG context:\t7628\n",
      "Total methylated C's in CHG context:\t699\n",
      "Total methylated C's in CHH context:\t2957\n",
      "\n",
      "Total C to T conversions in CpG context:\t4076\n",
      "Total C to T conversions in CHG context:\t20470\n",
      "Total C to T conversions in CHH context:\t58212\n",
      "\n",
      "C methylated in CpG context:\t65.2%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.8%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_7_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_8_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_8_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_8_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_8_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_8_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 5645 lines in total\n",
      "Total number of methylation call strings processed: 5645\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t92445\n",
      "\n",
      "Total methylated C's in CpG context:\t7495\n",
      "Total methylated C's in CHG context:\t721\n",
      "Total methylated C's in CHH context:\t2989\n",
      "\n",
      "Total C to T conversions in CpG context:\t3847\n",
      "Total C to T conversions in CHG context:\t19920\n",
      "Total C to T conversions in CHH context:\t57473\n",
      "\n",
      "C methylated in CpG context:\t66.1%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.9%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_8_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_8_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_8_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_8_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_9_s1_R1_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_9_s1_R1_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6115 lines in total\n",
      "Total number of methylation call strings processed: 6115\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t108734\n",
      "\n",
      "Total methylated C's in CpG context:\t11677\n",
      "Total methylated C's in CHG context:\t923\n",
      "Total methylated C's in CHH context:\t3066\n",
      "\n",
      "Total C to T conversions in CpG context:\t3948\n",
      "Total C to T conversions in CHG context:\t23839\n",
      "Total C to T conversions in CHH context:\t65281\n",
      "\n",
      "C methylated in CpG context:\t74.7%\n",
      "C methylated in CHG context:\t3.7%\n",
      "C methylated in CHH context:\t4.5%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_8_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_9_s1_R1_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_9_s1_R1_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_9_s1_R1_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R1_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "Checking file >>zr2096_9_s1_R2_bismark_bt2.bam<< for signs of file truncation...\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Now reading in Bismark result file zr2096_9_s1_R2_bismark_bt2.bam\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark -u 10000 --non_directional --score_min L,0,-1.2 --genome 2018-04-27-Bismark-Inputs/ /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_10_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_1_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_2_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_3_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_4_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_5_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_6_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_7_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_8_s1_R2.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R1.fastq.gz /Volumes/web/nightingales/C_virginica/zr2096_9_s1_R2.fastq.gz\"\n",
      "\n",
      "Processed 6280 lines in total\n",
      "Total number of methylation call strings processed: 6280\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t111835\n",
      "\n",
      "Total methylated C's in CpG context:\t11862\n",
      "Total methylated C's in CHG context:\t885\n",
      "Total methylated C's in CHH context:\t3081\n",
      "\n",
      "Total C to T conversions in CpG context:\t3985\n",
      "Total C to T conversions in CHG context:\t24536\n",
      "Total C to T conversions in CHH context:\t67486\n",
      "\n",
      "C methylated in CpG context:\t74.9%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.4%\n",
      "\n",
      "\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read length for M-Bias plot\n",
      "Maximum read length of Read 1: 100\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R1_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n",
      "2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: 2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_9_s1_R2_bismark_bt2.bedGraph.gz\n",
      "output directory:\t\t>/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Writing bedGraph to file: zr2096_9_s1_R2_bismark_bt2.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_9_s1_R2_bismark_bt2.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output/\n",
      "Now writing methylation information for file >>CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOT_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_CTOB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "Now writing methylation information for file >>CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz<< to individual files for each chromosome\n",
      "Finished writing out individual chromosome files for CpG_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz\n",
      "\n",
      "Collecting temporary chromosome file information... Processing the following input file(s):\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035782.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035784.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035786.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035781.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035780.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035783.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035788.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035785.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035787.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_035789.1.methXtractor.temp\n",
      "\n",
      "Sorting input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp by positions (using -S of 2G)\n",
      "Successfully deleted the temporary input file CpG_OT_zr2096_9_s1_R2_bismark_bt2.txt.gz.chrNC_007175.2.methXtractor.temp\n",
      "\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: 2018-04-29-Methylation-Extractor-Output/CHH_OB_zr2096_9_s1_R2_bismark_bt2.txt.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-o 2018-04-29-Methylation-Extractor-Output \\\n",
    "--gzip \\\n",
    "--bedGraph \\\n",
    "zr2096_*_bt2.bam \\"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once again, I got a `gunzip` error. However, I do have output files [in this folder](https://github.com/RobertsLab/project-virginica-oa/tree/master/analyses/2018-04-27-Bismark/2018-04-29-Methylation-Extractor-Output). I'll ignore the error and proceed."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. HTML Processing Report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "  SYNOPSIS:\r\n",
      "\r\n",
      "  This script uses a Bismark alignment report to generate a graphical HTML report page. Optionally, further reports of\r\n",
      "  the Bismark suite such as deduplication, methylation extractor splitting or M-bias reports can be specified as well. If several\r\n",
      "  Bismark reports are found in the same folder, a separate report will be generated for each of these, whereby the output filename\r\n",
      "  will be derived from the Bismark alignment report file. Bismark2report attempts to find optional reports automatically based\r\n",
      "  on the file basename.\r\n",
      "\r\n",
      "\r\n",
      "  USAGE: bismark2report [options]\r\n",
      "\r\n",
      "\r\n",
      "-o/--output <filename>     Name of the output file (optional). If not specified explicitly, the output filename will be derived\r\n",
      "                           from the Bismark alignment report file. Specifying an output filename only works if the HTML report is\r\n",
      "                           to be generated for a single Bismark alignment report (and potentially additional reports).\r\n",
      "\r\n",
      "--dir                      Output directory. Output is written to the current directory if not specified explicitly.\r\n",
      "\r\n",
      "\r\n",
      "--alignment_report FILE    If not specified explicitly, bismark2report attempts to find Bismark report file(s) in the current\r\n",
      "                           directory and produces a separate HTML report for each mapping report file. Based on the basename of\r\n",
      "                           the Bismark mapping report, bismark2report will also attempt to find the other Bismark reports (see below)\r\n",
      "                           for inclusion into the HTML report. Specifying a Bismark alignment report file is mandatory.\r\n",
      "\r\n",
      "--dedup_report FILE        If not specified explicitly, bismark2report attempts to find a deduplication report file with the same\r\n",
      "                           basename as the Bismark mapping report (generated by deduplicate_bismark) in the\r\n",
      "                           current working directory. Including a deduplication report is optional, and using the FILE 'none'\r\n",
      "                           will skip this step entirely.\r\n",
      "\r\n",
      "--splitting_report FILE    If not specified explicitly, bismark2report attempts to find a splitting report file with the same\r\n",
      "                           basename as the Bismark mapping report (generated by the Bismark methylation extractor) in the current\r\n",
      "                           working directory. Including a splitting report is optional, and using the FILE 'none' will skip this\r\n",
      "                           step entirely.\r\n",
      "\r\n",
      "--mbias_report FILE        If not specified explicitly, bismark2report attempts to find a single M-bias report file with the same\r\n",
      "                           basename as the Bismark mapping report (generated by the Bismark methylation extractor) in the current\r\n",
      "                           working directory. Including an M-Bias report is optional, and using the FILE 'none' will skip this step\r\n",
      "                           entirely.\r\n",
      "\r\n",
      "--nucleotide_report FILE   If not specified explicitly, bismark2report attempts to find a single nucleotide coverage report file\r\n",
      "                           with the same basename as the Bismark mapping report (generated by Bismark with the option\r\n",
      "                           '--nucleotide_coverage') in the current working directory. Including a nucleotide coverage statistics\r\n",
      "                           report is optional, and using the FILE 'none' will skip this report entirely.\r\n",
      "\r\n",
      "                           Script last modified: 13 May 2016\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are lots of ways to customize the `bismark2report` function, but unfortunately I don't understand what much of it means. I'll just run the command the simplest way possible to ensure it works.\n",
    "\n",
    "1. Path to `bismark2report`\n",
    "2. --dir + path to output directory"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before I run the command, I need to make a new output directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-04-29-Bismark2Report-Outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-04-27-Bismark-Inputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Bismark2Report-Outputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Methylation-Extractor-Output\u001b[m\u001b[m/\r\n",
      "zr2096_10_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_10_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R2_bismark_bt2_SE_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 20 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_10_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_10_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_10_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_10_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_10_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_10_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_1_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_1_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_1_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_1_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_1_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_1_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_2_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_2_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_2_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_2_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_2_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_2_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_3_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_3_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_3_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_3_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_3_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_3_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_4_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_4_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_4_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_4_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_4_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_4_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_5_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_5_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_5_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_5_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_5_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_5_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_6_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_6_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_6_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_6_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_6_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_6_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_7_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_7_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_7_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_7_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_7_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_7_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_8_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_8_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_8_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_8_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_8_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_8_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_9_s1_R1_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_9_s1_R1_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_9_s1_R1_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2018-04-29-Bismark2Report-Outputs/zr2096_9_s1_R2_bismark_bt2_SE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_9_s1_R2_bismark_bt2_SE_report.txt <\n",
      "Processing alignment report zr2096_9_s1_R2_bismark_bt2_SE_report.txt ...\n",
      "Complete\n",
      "\n",
      "No deduplication report file specified, skipping this step\n",
      "No splitting report file specified, skipping this step\n",
      "No M-bias report file specified, skipping this step\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2report \\\n",
    "--dir 2018-04-29-Bismark2Report-Outputs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The reports can be found [here](https://github.com/RobertsLab/project-virginica-oa/tree/master/analyses/2018-04-27-Bismark/2018-04-29-Bismark2Report-Outputs)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary Report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "  SYNOPSIS:\r\n",
      "\r\n",
      "  This script uses Bismark report files of several samples in a run folder to generate a graphical summary HTML report as well as\r\n",
      "  a whopping big table (tab delimited text) with all relevant alignment and methylation statistics which may be used for graphing\r\n",
      "  purposes in R, Excel or the like. Unless specific BAM files are specified, bismark2summary first identifies Bismark BAM files in\r\n",
      "  a folder (they need to use the Bismark naming conventions) and then automatically detects Bismark alignment, deduplication or\r\n",
      "  methylation extractor (splitting) reports based on the input file basename. If splitting reports are found they overwrite the\r\n",
      "  methylation statistics of the initial alignment report.\r\n",
      "\r\n",
      "\r\n",
      "  USAGE: bismark2summary [options] [<BAM file(s)>]\r\n",
      "\r\n",
      "  ARGUMENTS:\r\n",
      "\r\n",
      "  BAM file(s)                 Optional. If no BAM files are specified explicitly the current working directory is scanned for \r\n",
      "                              Bismark alignment files and their associated reports. \r\n",
      "\r\n",
      "  OPTIONS:\r\n",
      "\r\n",
      "  -o/--basename <filename>    Basename of the output file (optional). Generate a text file with all relevant extracted values\r\n",
      "                              ('basename.txt') as well as an HTML report ('basename.html'). If not specified explicitly, the\r\n",
      "                              the basename is 'bismark_summary_report'.\r\n",
      "\r\n",
      "  --title <string>            Optional HTML report title; use --title \"speech marks for text with spaces\". Default: \r\n",
      "                              'Bismark Summary Report'. \t\t\t      \r\n",
      "\r\n",
      "\r\n",
      "  --version                   Displays version information and exits.\r\n",
      "\t\t\t      \r\n",
      "  --help                      Displays this help message and exits.\r\n",
      "\r\n",
      "\r\n",
      "                                                Script last modified: 06 July 2016\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2summary -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are no required arguments, so I'm just going to run the command with nothing else."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found Bismark/Bowtie2 single-end files\n",
      "No Bismark/Bowtie2 paired-end BAM files detected\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 20 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: zr2096_10_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_10_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_1_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_1_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_2_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_2_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_3_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_3_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_4_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_4_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_5_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_5_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_6_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_6_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_7_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_7_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_8_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_8_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_9_s1_R1_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      ">> Reading from Bismark report: zr2096_9_s1_R2_bismark_bt2_SE_report.txt\n",
      "No deduplication report present, skipping...\n",
      "No methylation extractor report present, skipping...\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It seems to skip the methylation extractor reports because they're in a different directory. That's something I need to figure out how to fix if it ends up being important. Because we're using `methylkit` for methylation extraction, it may not be worth my time to figure out how to get it to read something from another directory.\n",
    "\n",
    "The report can be found as a [.txt file](https://github.com/RobertsLab/project-virginica-oa/blob/master/analyses/2018-04-27-Bismark/bismark_summary_report.txt) and [.html document](https://github.com/RobertsLab/project-virginica-oa/blob/master/analyses/2018-04-27-Bismark/bismark_summary_report.html).\n",
    "\n",
    "Overall, my `bismark` test pipeline seems to have yielded results! The next step is to understand what the outputs are, and then run the pipeline on the full extend of my sequencing data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Update: 2018-05-08"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `bismark` alignment step produced several .bam files. To understand these outputs, I will use the [Integrative Genomics Viewer](http://software.broadinstitute.org/software/igv/home). Before I can upload these files into IGV, I need to do some pre-processing. But first, I want to change my working directory! "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/yaamini/Documents/project-virginica-oa/notebooks'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark\n"
     ]
    }
   ],
   "source": [
    "cd ../analyses/2018-04-27-Bismark/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-04-27-Bismark-Inputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Bismark2Report-Outputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Methylation-Extractor-Output\u001b[m\u001b[m/\r\n",
      "2018-05-07-Subset-Alignment.xml\r\n",
      "bismark_summary_report.html\r\n",
      "bismark_summary_report.txt\r\n",
      "igv.log\r\n",
      "zr2096_10_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_10_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2_sort.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_sort.bam.bai\r\n",
      "zr2096_1_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R2_bismark_bt2_SE_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "According to the [`igvtools`](https://software.broadinstitute.org/software/igv/igvtools_commandline) website, I need to `sort` then `index` all of my .bam files. I will test this process with one file first. If it's successfull, I'll run it on all of my files. Here's what I need for the `sort` command:\n",
    "\n",
    "1. Path to `igvtools`\n",
    "2. `sort`\n",
    "3. Path to input .bam file\n",
    "4. Path to output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sorting zr2096_1_s1_R1_bismark_bt2.bam  -> zr2096_1_s1_R1_bismark_bt2_sort.bam\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "! /Users/Shared/Apps/IGVTools/igvtools sort \\\n",
    "zr2096_1_s1_R1_bismark_bt2.bam \\\n",
    "zr2096_1_s1_R1_bismark_bt2_sort.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Looks like the sorting completed! Now I'll use the `index` command:\n",
    "\n",
    "1. Path to `igvtools`\n",
    "2. `index`\n",
    "3. Path to input .bam file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\r\n"
     ]
    }
   ],
   "source": [
    "! /Users/Shared/Apps/IGVTools/igvtools index \\\n",
    "zr2096_1_s1_R1_bismark_bt2_sort.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-04-27-Bismark-Inputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Bismark2Report-Outputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Methylation-Extractor-Output\u001b[m\u001b[m/\r\n",
      "2018-05-07-Subset-Alignment.xml\r\n",
      "bismark_summary_report.html\r\n",
      "bismark_summary_report.txt\r\n",
      "igv.log\r\n",
      "zr2096_10_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_10_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2_sort.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_sort.bam.bai\r\n",
      "zr2096_1_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R2_bismark_bt2_SE_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I now have a sorted .bam file and an index file! Time to run these commands on the rest of my samples. [Sam suggested](https://github.com/RobertsLab/resources/issues/248#issuecomment-387471570) I use the code below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "for bam in *.bam\n",
    "do\n",
    "/Users/Shared/Apps/IGVTools/igvtools sort \\\n",
    "\"$bam\" \"$bam\".sorted\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-04-27-Bismark-Inputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Bismark2Report-Outputs\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-29-Methylation-Extractor-Output\u001b[m\u001b[m/\r\n",
      "2018-05-07-Subset-Alignment.xml\r\n",
      "bismark_summary_report.html\r\n",
      "bismark_summary_report.txt\r\n",
      "igv.log\r\n",
      "zr2096_10_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_10_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_10_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_1_s1_R1_bismark_bt2_sort.bam\r\n",
      "zr2096_1_s1_R1_bismark_bt2_sort.bam.bai\r\n",
      "zr2096_1_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_1_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_2_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_2_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_3_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_3_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_4_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_4_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_5_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_5_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_6_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_6_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_7_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_7_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_8_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_8_s1_R2_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R1_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R1_bismark_bt2_SE_report.txt\r\n",
      "zr2096_9_s1_R2_bismark_bt2.bam\r\n",
      "zr2096_9_s1_R2_bismark_bt2_SE_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
