{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# FastQC trimming for *C. virginica* gonad methylation data\n",
    "\n",
    "In this notebook, I'll use FastQC to examine file quality and trim sequences of my *C. virginica* gonad MBDSeq data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/yaamini/Documents/project-virginica-oa/notebooks'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First I will make a new directory for my analyses."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-26-Gonad-Methylation-FastQC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Confirm I made the directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-01-23-MBDSeq-Labwork\u001b[m\u001b[m/           README.md\r\n",
      "\u001b[34m2018-04-26-Gonad-Methylation-FastQC\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls ../analyses/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then change my working directory to the new folder I created."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-26-Gonad-Methylation-FastQC\n"
     ]
    }
   ],
   "source": [
    "cd ../analyses/2018-04-26-Gonad-Methylation-FastQC/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Identify path for FastQC command-line interface"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "            FastQC - A high throughput sequence QC analysis tool\r\n",
      "\r\n",
      "SYNOPSIS\r\n",
      "\r\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\r\n",
      "\r\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \r\n",
      "           [-c contaminant file] seqfile1 .. seqfileN\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\r\n",
      "    control report consisting of a number of different modules, each one of \r\n",
      "    which will help to identify a different potential type of problem in your\r\n",
      "    data.\r\n",
      "    \r\n",
      "    If no files to process are specified on the command line then the program\r\n",
      "    will start as an interactive graphical application.  If files are provided\r\n",
      "    on the command line then the program will run with no user interaction\r\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\r\n",
      "    analysis pipeline.\r\n",
      "    \r\n",
      "    The options for the program as as follows:\r\n",
      "    \r\n",
      "    -h --help       Print this help file and exit\r\n",
      "    \r\n",
      "    -v --version    Print the version of the program and exit\r\n",
      "    \r\n",
      "    -o --outdir     Create all output files in the specified output directory.\r\n",
      "                    Please note that this directory must exist as the program\r\n",
      "                    will not create it.  If this option is not set then the \r\n",
      "                    output file for each sequence file is created in the same\r\n",
      "                    directory as the sequence file which was processed.\r\n",
      "                    \r\n",
      "    --casava        Files come from raw casava output. Files in the same sample\r\n",
      "                    group (differing only by the group number) will be analysed\r\n",
      "                    as a set rather than individually. Sequences with the filter\r\n",
      "                    flag set in the header will be excluded from the analysis.\r\n",
      "                    Files must have the same names given to them by casava\r\n",
      "                    (including being gzipped and ending with .gz) otherwise they\r\n",
      "                    won't be grouped together correctly.\r\n",
      "                    \r\n",
      "    --nano          Files come from naopore sequences and are in fast5 format. In\r\n",
      "                    this mode you can pass in directories to process and the program\r\n",
      "                    will take in all fast5 files within those directories and produce\r\n",
      "                    a single output file from the sequences found in all files.                    \r\n",
      "                    \r\n",
      "    --nofilter      If running with --casava then don't remove read flagged by\r\n",
      "                    casava as poor quality when performing the QC analysis.\r\n",
      "                   \r\n",
      "    --extract       If set then the zipped output file will be uncompressed in\r\n",
      "                    the same directory after it has been created.  By default\r\n",
      "                    this option will be set if fastqc is run in non-interactive\r\n",
      "                    mode.\r\n",
      "                    \r\n",
      "    -j --java       Provides the full path to the java binary you want to use to\r\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\r\n",
      "                    your path.\r\n",
      "                   \r\n",
      "    --noextract     Do not uncompress the output file after creating it.  You\r\n",
      "                    should set this option if you do not wish to uncompress\r\n",
      "                    the output when running in non-interactive mode.\r\n",
      "                    \r\n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\r\n",
      "                    show data for every base in the read.  WARNING: Using this\r\n",
      "                    option will cause fastqc to crash and burn if you use it on\r\n",
      "                    really long reads, and your plots may end up a ridiculous size.\r\n",
      "                    You have been warned!\r\n",
      "                    \r\n",
      "    -f --format     Bypasses the normal sequence file format detection and\r\n",
      "                    forces the program to use the specified format.  Valid\r\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\r\n",
      "                    \r\n",
      "    -t --threads    Specifies the number of files which can be processed\r\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\r\n",
      "                    memory so you shouldn't run more threads than your\r\n",
      "                    available memory will cope with, and not more than\r\n",
      "                    6 threads on a 32 bit machine\r\n",
      "                  \r\n",
      "    -c              Specifies a non-default file which contains the list of\r\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\r\n",
      "                    The file must contain sets of named contaminants in the\r\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\r\n",
      "                    be ignored.\r\n",
      "\r\n",
      "    -a              Specifies a non-default file which contains the list of\r\n",
      "    --adapters      adapter sequences which will be explicity searched against\r\n",
      "                    the library. The file must contain sets of named adapters\r\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\r\n",
      "                    will be ignored.\r\n",
      "                    \r\n",
      "    -l              Specifies a non-default file which contains a set of criteria\r\n",
      "    --limits        which will be used to determine the warn/error limits for the\r\n",
      "                    various modules.  This file can also be used to selectively \r\n",
      "                    remove some modules from the output all together.  The format\r\n",
      "                    needs to mirror the default limits.txt file found in the\r\n",
      "                    Configuration folder.\r\n",
      "                    \r\n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\r\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\r\n",
      "                    length is 7 if not specified.\r\n",
      "                    \r\n",
      "   -q --quiet       Supress all progress messages on stdout and only report errors.\r\n",
      "   \r\n",
      "   -d --dir         Selects a directory to be used for temporary files written when\r\n",
      "                    generating report images. Defaults to system temp directory if\r\n",
      "                    not specified.\r\n",
      "                    \r\n",
      "BUGS\r\n",
      "\r\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\r\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\r\n",
      "                   \r\n",
      "    "
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/bioinformatics/FastQC/fastqc -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Use command-line interface\n",
    "\n",
    "Based on the help menu, I need the following information in my code:\n",
    "\n",
    "1. Path of FastQC application: I identified the path in the above cell\n",
    "2. Files to be analyzed: Sam said I can analyze the files from owl itself. The files can be found at [this link](http://owl.fish.washington.edu/nightingales/C_virginica/). \n",
    "3. Number of files to process simultaneously:I'm going to process 4 files simultaneously\n",
    "4. Directory for output files: 2018-04-26-Gonad-Methylation-FastQC (i.e. my current working directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of zr2096_10_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_10_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_1_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_1_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_10_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_2_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_2_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_1_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_3_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_3_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_2_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_2_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_4_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_4_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 100% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 100% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_3_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_5_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_5_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_4_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_4_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_6_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_6_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_7_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_5_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_7_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_6_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_6_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_8_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_8_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_7_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_9_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_9_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_8_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_9_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_9_s1_R2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/bioinformatics/FastQC/fastqc \\\n",
    "/Volumes/web/nightingales/C_virginica/zr2096_* \\\n",
    "-t 4 \\\n",
    "-o ../2018-04-26-Gonad-Methylation-FastQC/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "My FastQC files can be found [here](https://github.com/RobertsLab/project-virginica-oa/tree/master/analyses/2018-04-26-Gonad-Methylation-FastQC)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_fastqc.html  zr2096_5_s1_R1_fastqc.html\r\n",
      "zr2096_10_s1_R1_fastqc.zip   zr2096_5_s1_R1_fastqc.zip\r\n",
      "zr2096_10_s1_R2_fastqc.html  zr2096_5_s1_R2_fastqc.html\r\n",
      "zr2096_10_s1_R2_fastqc.zip   zr2096_5_s1_R2_fastqc.zip\r\n",
      "zr2096_1_s1_R1_fastqc.html   zr2096_6_s1_R1_fastqc.html\r\n",
      "zr2096_1_s1_R1_fastqc.zip    zr2096_6_s1_R1_fastqc.zip\r\n",
      "zr2096_1_s1_R2_fastqc.html   zr2096_6_s1_R2_fastqc.html\r\n",
      "zr2096_1_s1_R2_fastqc.zip    zr2096_6_s1_R2_fastqc.zip\r\n",
      "zr2096_2_s1_R1_fastqc.html   zr2096_7_s1_R1_fastqc.html\r\n",
      "zr2096_2_s1_R1_fastqc.zip    zr2096_7_s1_R1_fastqc.zip\r\n",
      "zr2096_2_s1_R2_fastqc.html   zr2096_7_s1_R2_fastqc.html\r\n",
      "zr2096_2_s1_R2_fastqc.zip    zr2096_7_s1_R2_fastqc.zip\r\n",
      "zr2096_3_s1_R1_fastqc.html   zr2096_8_s1_R1_fastqc.html\r\n",
      "zr2096_3_s1_R1_fastqc.zip    zr2096_8_s1_R1_fastqc.zip\r\n",
      "zr2096_3_s1_R2_fastqc.html   zr2096_8_s1_R2_fastqc.html\r\n",
      "zr2096_3_s1_R2_fastqc.zip    zr2096_8_s1_R2_fastqc.zip\r\n",
      "zr2096_4_s1_R1_fastqc.html   zr2096_9_s1_R1_fastqc.html\r\n",
      "zr2096_4_s1_R1_fastqc.zip    zr2096_9_s1_R1_fastqc.zip\r\n",
      "zr2096_4_s1_R2_fastqc.html   zr2096_9_s1_R2_fastqc.html\r\n",
      "zr2096_4_s1_R2_fastqc.zip    zr2096_9_s1_R2_fastqc.zip\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Compare FastQC results with MultiQC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MultiQC compiles existing FastQC analyses for easy comparison. But first, I need to install it. I used the code `conda install -c bioconda multiqc` in a separate Terminal window for the installation."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![multiqc-installation](https://raw.githubusercontent.com/RobertsLab/project-virginica-oa/master/images/2018-04-27-MultiQC-Installation.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: multiqc [OPTIONS] <analysis directory>\r\n",
      "\r\n",
      "  MultiQC aggregates results from bioinformatics analyses across many\r\n",
      "  samples into a single report.\r\n",
      "\r\n",
      "  It searches a given directory for analysis logs and compiles a HTML\r\n",
      "  report. It's a general use tool, perfect for summarising the output from\r\n",
      "  numerous bioinformatics tools.\r\n",
      "\r\n",
      "  To run, supply with one or more directory to scan for analysis results. To\r\n",
      "  run here, use 'multiqc .'\r\n",
      "\r\n",
      "  See http://multiqc.info for more details.\r\n",
      "\r\n",
      "  Author: Phil Ewels (http://phil.ewels.co.uk)\r\n",
      "\r\n",
      "Options:\r\n",
      "  -f, --force                     Overwrite any existing reports\r\n",
      "  -d, --dirs                      Prepend directory to sample names\r\n",
      "  -dd, --dirs-depth INTEGER       Prepend [INT] directories to sample names.\r\n",
      "                                  Negative number to take from start of path.\r\n",
      "  -s, --fullnames                 Do not clean the sample names (leave as full\r\n",
      "                                  file name)\r\n",
      "  -i, --title TEXT                Report title. Printed as page header, used\r\n",
      "                                  for filename if not otherwise specified.\r\n",
      "  -b, --comment TEXT              Custom comment, will be printed at the top\r\n",
      "                                  of the report.\r\n",
      "  -n, --filename TEXT             Report filename. Use 'stdout' to print to\r\n",
      "                                  standard out.\r\n",
      "  -o, --outdir TEXT               Create report in the specified output\r\n",
      "                                  directory.\r\n",
      "  -t, --template [geo|default|simple|default_dev]\r\n",
      "                                  Report template to use.\r\n",
      "  -x, --ignore TEXT               Ignore analysis files (glob expression)\r\n",
      "  -l, --file-list                 Supply a file containing a list of file\r\n",
      "                                  paths to be searched, one per row\r\n",
      "  -e, --exclude [module name]     Do not use this module. Can specify multiple\r\n",
      "                                  times.\r\n",
      "  -m, --module [module name]      Use only this module. Can specify multiple\r\n",
      "                                  times.\r\n",
      "  --data-dir                      Force the parsed data directory to be\r\n",
      "                                  created.\r\n",
      "  --no-data-dir                   Prevent the parsed data directory from being\r\n",
      "                                  created.\r\n",
      "  -k, --data-format [tsv|yaml|json]\r\n",
      "                                  Output parsed data in a different format.\r\n",
      "                                  Default: tsv\r\n",
      "  -z, --zip-data-dir              Compress the data directory.\r\n",
      "  -p, --export                    Export plots as static images in addition to\r\n",
      "                                  the report\r\n",
      "  -fp, --flat                     Use only flat plots (static images)\r\n",
      "  -ip, --interactive              Use only interactive plots (HighCharts\r\n",
      "                                  Javascript)\r\n",
      "  --pdf                           Creates PDF report with 'simple' template.\r\n",
      "                                  Requires Pandoc to be installed.\r\n",
      "  -c, --config PATH               Specific config file to load, after those in\r\n",
      "                                  MultiQC dir / home dir / working dir.\r\n",
      "  -v, --verbose                   Increase output verbosity.\r\n",
      "  -q, --quiet                     Only show log warnings\r\n",
      "  --version                       Show the version and exit.\r\n",
      "  -h, --help                      Show this message and exit.\r\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/anaconda/bin/multiqc -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To use MultiQC, I need to specify:\n",
    "\n",
    "1. Location of `multiqc` on my computer (see above) + directory with output from FastQC analysis (i.e. my current working directory)\n",
    "2. Specified MultiQC output directory (my current working directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[WARNING]         multiqc : MultiQC Version v1.5 now available!\n",
      "[INFO   ]         multiqc : This is MultiQC v1.0.dev0\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '../2018-04-26-Gonad-Methylation-FastQC/'\n",
      "[INFO   ]         multiqc : Searching '../2018-04-26-Gonad-Methylation-FastQC/'\n",
      "[INFO   ]          fastqc : Found 20 reports\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/anaconda/bin/multiqc ../2018-04-26-Gonad-Methylation-FastQC/ \\\n",
    "> -o ../2018-04-26-Gonad-Methylation-FastQC/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The results of the MultiQC analysis can be viewed [using .txt files in this folder](https://github.com/RobertsLab/project-virginica-oa/tree/master/analyses/2018-04-26-Gonad-Methylation-FastQC/multiqc_data) or an [interactive .html output](http://htmlpreview.github.io/?https://github.com/RobertsLab/project-virginica-oa/blob/master/analyses/2018-04-26-Gonad-Methylation-FastQC/multiqc_report.html)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
