{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bismark Parameter Testing\n",
    "\n",
    "In this notebook I'll test different `bismark` parameters using a subset of my gonad methylation data.\n",
    "\n",
    "1. Genome Preparation (already done in [this Jupyter notebook](https://github.com/RobertsLab/project-virginica-oa/blob/master/notebooks/2018-04-27-Gonad-Methylation-Bismark.ipynb))\n",
    "2. Alignment\n",
    "3. Deduplication\n",
    "4. Methlyation Extractor\n",
    "5. HTML Processing Report\n",
    "6. Summary Report\n",
    "\n",
    "Primarily, I'll test different alignments to increase mapping efficiency."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0. Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/yaamini/Documents/project-virginica-oa/notebooks'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses\n"
     ]
    }
   ],
   "source": [
    "cd ../analyses/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-01-23-MBDSeq-Labwork\u001b[m\u001b[m/              \u001b[34m2018-06-11-DML-Analysis\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-26-Gonad-Methylation-FastQC\u001b[m\u001b[m/    \u001b[34m2018-06-14-Gene-Enrichment-Analysis\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-04-27-Bismark\u001b[m\u001b[m/                     \u001b[34m2018-10-11-MethylKit-Parameter-Testing\u001b[m\u001b[m/\r\n",
      "\u001b[34m2018-05-01-MethylKit\u001b[m\u001b[m/                   README.md\r\n",
      "\u001b[34m2018-05-29-MethylKit-Full-Samples\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-10-03-Bismark-Parameter-Testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing\n"
     ]
    }
   ],
   "source": [
    "cd 2018-10-03-Bismark-Parameter-Testing/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Genome Preparation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This step was already completed in [this Jupyter notebook](https://github.com/RobertsLab/project-virginica-oa/blob/master/notebooks/2018-04-27-Gonad-Methylation-Bismark.ipynb). The genome only needs to be prepared once. I will move on to the second step, alignment."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "     This program is free software: you can redistribute it and/or modify\n",
      "     it under the terms of the GNU General Public License as published by\n",
      "     the Free Software Foundation, either version 3 of the License, or\n",
      "     (at your option) any later version.\n",
      "\n",
      "     This program is distributed in the hope that it will be useful,\n",
      "     but WITHOUT ANY WARRANTY; without even the implied warranty of\n",
      "     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n",
      "     GNU General Public License for more details.\n",
      "     You should have received a copy of the GNU General Public License\n",
      "     along with this program.  If not, see <http://www.gnu.org/licenses/>.\n",
      "\n",
      "\n",
      "\n",
      "DESCRIPTION\n",
      "\n",
      "\n",
      "The following is a brief description of command line options and arguments to control the Bismark\n",
      "bisulfite mapper and methylation caller. Bismark takes in FastA or FastQ files and aligns the\n",
      "reads to a specified bisulfite genome. Sequence reads are transformed into a bisulfite converted forward strand\n",
      "version (C->T conversion) or into a bisulfite treated reverse strand (G->A conversion of the forward strand).\n",
      "Each of these reads are then aligned to bisulfite treated forward strand index of a reference genome\n",
      "(C->T converted) and a bisulfite treated reverse strand index of the genome (G->A conversion of the\n",
      "forward strand, by doing this alignments will produce the same positions). These 4 instances of Bowtie (1 or 2)\n",
      "are run in parallel. The sequence file(s) are then read in again sequence by sequence to pull out the original\n",
      "sequence from the genome and determine if there were any protected C's present or not.\n",
      "\n",
      "As of version 0.7.0 Bismark will only run 2 alignment threads for OT and OB in parallel, the 4 strand mode can be\n",
      "re-enabled by using --non_directional.\n",
      "\n",
      "The final output of Bismark is in SAM format by default. For Bowtie 1 one can alos choose to report the old\n",
      "'vanilla' output format, which is a single tab delimited file with all sequences that have a unique best\n",
      "alignment to any of the 4 possible strands of a bisulfite PCR product. Both formats are described in more detail below.\n",
      "\n",
      "\n",
      "USAGE: bismark [options] <genome_folder> {-1 <mates1> -2 <mates2> | <singles>}\n",
      "\n",
      "\n",
      "ARGUMENTS:\n",
      "\n",
      "<genome_folder>          The path to the folder containing the unmodified reference genome\n",
      "                         as well as the subfolders created by the Bismark_Genome_Preparation\n",
      "                         script (/Bisulfite_Genome/CT_conversion/ and /Bisulfite_Genome/GA_conversion/).\n",
      "                         Bismark expects one or more fastA files in this folder (file extension: .fa\n",
      "                         or .fasta). The path can be relative or absolute. The path may also be set as\n",
      "                         '--genome_folder /path/to/genome/folder/'.\n",
      "\n",
      "-1 <mates1>              Comma-separated list of files containing the #1 mates (filename usually includes\n",
      "                         \"_1\"), e.g. flyA_1.fq,flyB_1.fq). Sequences specified with this option must\n",
      "                         correspond file-for-file and read-for-read with those specified in <mates2>.\n",
      "                         Reads may be a mix of different lengths. Bismark will produce one mapping result\n",
      "                         and one report file per paired-end input file pair.\n",
      "\n",
      "-2 <mates2>              Comma-separated list of files containing the #2 mates (filename usually includes\n",
      "                         \"_2\"), e.g. flyA_1.fq,flyB_1.fq). Sequences specified with this option must\n",
      "                         correspond file-for-file and read-for-read with those specified in <mates1>.\n",
      "                         Reads may be a mix of different lengths.\n",
      "\n",
      "<singles>                A comma- or space-separated list of files containing the reads to be aligned (e.g.\n",
      "                         lane1.fq,lane2.fq lane3.fq). Reads may be a mix of different lengths. Bismark will\n",
      "                         produce one mapping result and one report file per input file.\n",
      "\n",
      "\n",
      "OPTIONS:\n",
      "\n",
      "\n",
      "Input:\n",
      "\n",
      "--se/--single_end <list> Sets single-end mapping mode explicitly giving a list of file names as <list>.\n",
      "                         The filenames may be provided as a comma [,] or colon [:] separated list.\n",
      "\n",
      "-q/--fastq               The query input files (specified as <mate1>,<mate2> or <singles> are FASTQ\n",
      "                         files (usually having extension .fg or .fastq). This is the default. See also\n",
      "                         --solexa-quals.\n",
      "\n",
      "-f/--fasta               The query input files (specified as <mate1>,<mate2> or <singles> are FASTA\n",
      "                         files (usually having extensions .fa, .mfa, .fna or similar). All quality values\n",
      "                         are assumed to be 40 on the Phred scale. FASTA files are expected to contain both\n",
      "                         the read name and the sequence on a single line (and not spread over several lines).\n",
      "\n",
      "-s/--skip <int>          Skip (i.e. do not align) the first <int> reads or read pairs from the input.\n",
      "\n",
      "-u/--upto <int>          Only aligns the first <int> reads or read pairs from the input. Default: no limit.\n",
      "\n",
      "--phred33-quals          FASTQ qualities are ASCII chars equal to the Phred quality plus 33. Default: on.\n",
      "\n",
      "--phred64-quals          FASTQ qualities are ASCII chars equal to the Phred quality plus 64. Default: off.\n",
      "\n",
      "--solexa-quals           Convert FASTQ qualities from solexa-scaled (which can be negative) to phred-scaled\n",
      "                         (which can't). The formula for conversion is:\n",
      "                         phred-qual = 10 * log(1 + 10 ** (solexa-qual/10.0)) / log(10). Used with -q. This\n",
      "                         is usually the right option for use with (unconverted) reads emitted by the GA\n",
      "                         Pipeline versions prior to 1.3. Works only for Bowtie 1. Default: off.\n",
      "\n",
      "--solexa1.3-quals        Same as --phred64-quals. This is usually the right option for use with (unconverted)\n",
      "                         reads emitted by GA Pipeline version 1.3 or later. Default: off.\n",
      "\n",
      "--path_to_bowtie         The full path </../../> to the Bowtie (1 or 2) installation on your system. If not\n",
      "                         specified it is assumed that Bowtie (1 or 2) is in the PATH.\n",
      "\n",
      "\n",
      "Alignment:\n",
      "\n",
      "-n/--seedmms <int>       The maximum number of mismatches permitted in the \"seed\", i.e. the first L base pairs\n",
      "                         of the read (where L is set with -l/--seedlen). This may be 0, 1, 2 or 3 and the\n",
      "                         default is 1. This option is only available for Bowtie 1 (for Bowtie 2 see -N).\n",
      "\n",
      "-l/--seedlen             The \"seed length\"; i.e., the number of bases of the high quality end of the read to\n",
      "                         which the -n ceiling applies. The default is 28. Bowtie (and thus Bismark) is faster for\n",
      "                         larger values of -l. This option is only available for Bowtie 1 (for Bowtie 2 see -L).\n",
      "\n",
      "-e/--maqerr <int>        Maximum permitted total of quality values at all mismatched read positions throughout\n",
      "                         the entire alignment, not just in the \"seed\". The default is 70. Like Maq, bowtie rounds\n",
      "                         quality values to the nearest 10 and saturates at 30. This value is not relevant for\n",
      "                         Bowtie 2.\n",
      "\n",
      "--chunkmbs <int>         The number of megabytes of memory a given thread is given to store path descriptors in\n",
      "                         --best mode. Best-first search must keep track of many paths at once to ensure it is\n",
      "                         always extending the path with the lowest cumulative cost. Bowtie tries to minimize the\n",
      "                         memory impact of the descriptors, but they can still grow very large in some cases. If\n",
      "                         you receive an error message saying that chunk memory has been exhausted in --best mode,\n",
      "                         try adjusting this parameter up to dedicate more memory to the descriptors. This value\n",
      "                         is not relevant for Bowtie 2. Default: 512.\n",
      "\n",
      "-I/--minins <int>        The minimum insert size for valid paired-end alignments. E.g. if -I 60 is specified and\n",
      "                         a paired-end alignment consists of two 20-bp alignments in the appropriate orientation\n",
      "                         with a 20-bp gap between them, that alignment is considered valid (as long as -X is also\n",
      "                         satisfied). A 19-bp gap would not be valid in that case. Default: 0.\n",
      "\n",
      "-X/--maxins <int>        The maximum insert size for valid paired-end alignments. E.g. if -X 100 is specified and\n",
      "                         a paired-end alignment consists of two 20-bp alignments in the proper orientation with a\n",
      "                         60-bp gap between them, that alignment is considered valid (as long as -I is also satisfied).\n",
      "                         A 61-bp gap would not be valid in that case. Default: 500.\n",
      "\n",
      "--parallel <int>         (May also be --multicore <int>) Sets the number of parallel instances of Bismark to be run concurrently.\n",
      "                         This forks the Bismark alignment step very early on so that each individual Spawn of Bismark processes\n",
      "                         only every n-th sequence (n being set by --parallel). Once all processes have completed,\n",
      "                         the individual BAM files, mapping reports, unmapped or ambiguous FastQ files are merged\n",
      "                         into single files in very much the same way as they would have been generated running Bismark\n",
      "                         conventionally with only a single instance.\n",
      "\n",
      "                         If system resources are plentiful this is a viable option to speed up the alignment process\n",
      "                         (we observed a near linear speed increase for up to --parallel 8 tested). However, please note\n",
      "                         that a typical Bismark run will use several cores already (Bismark itself, 2 or 4 threads of\n",
      "                         Bowtie/Bowtie2, Samtools, gzip etc...) and ~10-16GB of memory depending on the choice of aligner\n",
      "                         and genome. WARNING: Bismark Parallel (BP?) is resource hungry! Each value of --parallel specified\n",
      "                         will effectively lead to a linear increase in compute and memory requirements, so --parallel 4 for\n",
      "                         e.g. the GRCm38 mouse genome will probably use ~20 cores and eat ~40GB or RAM, but at the same time\n",
      "                         reduce the alignment time to ~25-30%. You have been warned.\n",
      "\n",
      "\n",
      "\n",
      "Bowtie 1 Reporting:\n",
      "\n",
      "-k <2>                   Due to the way Bismark works Bowtie will report up to 2 valid alignments. This option\n",
      "                         will be used by default.\n",
      "\n",
      "--best                   Make Bowtie guarantee that reported singleton alignments are \"best\" in terms of stratum\n",
      "                         (i.e. number of mismatches, or mismatches in the seed in the case if -n mode) and in\n",
      "                         terms of the quality; e.g. a 1-mismatch alignment where the mismatch position has Phred\n",
      "                         quality 40 is preferred over a 2-mismatch alignment where the mismatched positions both\n",
      "                         have Phred quality 10. When --best is not specified, Bowtie may report alignments that\n",
      "                         are sub-optimal in terms of stratum and/or quality (though an effort is made to report\n",
      "                         the best alignment). --best mode also removes all strand bias. Note that --best does not\n",
      "                         affect which alignments are considered \"valid\" by Bowtie, only which valid alignments\n",
      "                         are reported by Bowtie. Bowtie is about 1-2.5 times slower when --best is specified.\n",
      "                         Default: on.\n",
      "\n",
      "--no_best                Disables the --best option which is on by default. This can speed up the alignment process,\n",
      "                         e.g. for testing purposes, but for credible results it is not recommended to disable --best.\n",
      "\n",
      "\n",
      "Output:\n",
      "\n",
      "--non_directional        The sequencing library was constructed in a non strand-specific manner, alignments to all four\n",
      "                         bisulfite strands will be reported. Default: OFF.\n",
      "\n",
      "                         (The current Illumina protocol for BS-Seq is directional, in which case the strands complementary\n",
      "                         to the original strands are merely theoretical and should not exist in reality. Specifying directional\n",
      "                         alignments (which is the default) will only run 2 alignment threads to the original top (OT)\n",
      "                         or bottom (OB) strands in parallel and report these alignments. This is the recommended option\n",
      "                         for sprand-specific libraries).\n",
      "\n",
      "--pbat                   This options may be used for PBAT-Seq libraries (Post-Bisulfite Adapter Tagging; Kobayashi et al.,\n",
      "                         PLoS Genetics, 2012). This is essentially the exact opposite of alignments in 'directional' mode,\n",
      "                         as it will only launch two alignment threads to the CTOT and CTOB strands instead of the normal OT\n",
      "                         and OB ones. Use this option only if you are certain that your libraries were constructed following\n",
      "                         a PBAT protocol (if you don't know what PBAT-Seq is you should not specify this option). The option\n",
      "                         --pbat works only for FastQ files (in both Bowtie and Bowtie 2 mode) and using uncompressed\n",
      "                         temporary files only).\n",
      "\n",
      "--sam-no-hd              Suppress SAM header lines (starting with @). This might be useful when very large input files are\n",
      "                         split up into several smaller files to run concurrently and the output files are to be merged.\n",
      "\n",
      "--rg_tag                 Write out a Read Group tag to the resulting SAM/BAM file. This will write the following line to the\n",
      "                         SAM header: @RG PL: ILLUMINA ID:SAMPLE SM:SAMPLE ; to set ID and SM see --rg_id and --rg_sample.\n",
      "                         In addition each read receives an RG:Z:RG-ID tag. Default: OFF.\n",
      "\n",
      "--rg_id <string>         Sets the ID field in the @RG header line. The default is 'SAMPLE'.\n",
      "\n",
      "--rg_sample <string>     Sets the SM field in the @RG header line; can't be set without setting --rg_id as well. The default is\n",
      "                         'SAMPLE'.\n",
      "\n",
      "--quiet                  Print nothing besides alignments.\n",
      "\n",
      "--vanilla                Performs bisulfite mapping with Bowtie 1 and prints the 'old' output (as in Bismark 0.5.X) instead\n",
      "                         of SAM format output.\n",
      "\n",
      "-un/--unmapped           Write all reads that could not be aligned to a file in the output directory. Written reads will\n",
      "                         appear as they did in the input, without any translation of quality values that may have\n",
      "                         taken place within Bowtie or Bismark. Paired-end reads will be written to two parallel files with _1\n",
      "                         and _2 inserted in their filenames, i.e. _unmapped_reads_1.txt and unmapped_reads_2.txt. Reads\n",
      "                         with more than one valid alignment with the same number of lowest mismatches (ambiguous mapping)\n",
      "                         are also written to _unmapped_reads.txt unless the option --ambiguous is specified as well.\n",
      "\n",
      "--ambiguous              Write all reads which produce more than one valid alignment with the same number of lowest\n",
      "                         mismatches or other reads that fail to align uniquely to a file in the output directory.\n",
      "                         Written reads will appear as. they did in the input, without any of the translation of quality\n",
      "                         values that may have taken place within Bowtie or Bismark. Paired-end reads will be written to two\n",
      "                         parallel files with _1 and _2 inserted in their filenames, i.e. _ambiguous_reads_1.txt and\n",
      "                         _ambiguous_reads_2.txt. These reads are not written to the file specified with --un.\n",
      "\n",
      "-o/--output_dir <dir>    Write all output files into this directory. By default the output files will be written into\n",
      "                         the same folder as the input file(s). If the specified folder does not exist, Bismark will attempt\n",
      "                         to create it first. The path to the output folder can be either relative or absolute.\n",
      "\n",
      "--temp_dir <dir>         Write temporary files to this directory instead of into the same directory as the input files. If\n",
      "                         the specified folder does not exist, Bismark will attempt to create it first. The path to the\n",
      "                         temporary folder can be either relative or absolute.\n",
      "\n",
      "--non_bs_mm              Optionally outputs an extra column specifying the number of non-bisulfite mismatches a read during the\n",
      "                         alignment step. This option is only available for SAM format. In Bowtie 2 context, this value is\n",
      "                         just the number of actual non-bisulfite mismatches and ignores potential insertions or deletions.\n",
      "                         The format for single-end reads and read 1 of paired-end reads is 'XA:Z:number of mismatches'\n",
      "                         and 'XB:Z:number of mismatches' for read 2 of paired-end reads.\n",
      "\n",
      "--gzip                   Temporary bisulfite conversion files will be written out in a GZIP compressed form to save disk\n",
      "                         space. This option is available for most alignment modes but is not available for paired-end FastA\n",
      "                         files. This option might be somewhat slower than writing out uncompressed files, but this awaits\n",
      "                         further testing.\n",
      "\n",
      "--sam                    The output will be written out in SAM format instead of the default BAM format. Bismark will\n",
      "                         attempt to use the path to Samtools that was specified with '--samtools_path', or, if it hasn't\n",
      "                         been specified, attempt to find Samtools in the PATH. If no installation of Samtools can be found,\n",
      "                         the SAM output will be compressed with GZIP instead (yielding a .sam.gz output file).\n",
      "\n",
      "--cram                   Writes the output to a CRAM file instead of BAM. This requires the use of Samtools 1.2 or higher.\n",
      "\n",
      "--cram_ref <ref_file>    CRAM output requires you to specify a reference genome as a single FastA file. If this single-FastA\n",
      "                         reference file is not supplied explicitly it will be regenerated from the genome .fa sequence(s)\n",
      "                         used for the Bismark run and written to a file called 'Bismark_genome_CRAM_reference.mfa' into the\n",
      "                         oputput directory.\n",
      "\n",
      "--samtools_path          The path to your Samtools installation, e.g. /home/user/samtools/. Does not need to be specified\n",
      "                         explicitly if Samtools is in the PATH already.\n",
      "\n",
      "--prefix <prefix>        Prefixes <prefix> to the output filenames. Trailing dots will be replaced by a single one. For\n",
      "                         example, '--prefix test' with 'file.fq' would result in the output file 'test.file.fq_bismark.sam' etc.\n",
      "\n",
      "-B/--basename <basename> Write all output to files starting with this base file name. For example, '--basename foo'\n",
      "                         would result in the files 'foo.bam' and 'foo_SE_report.txt' (or its paired-end equivalent). Takes\n",
      "                         precedence over --prefix.\n",
      "\n",
      "--old_flag               Only in paired-end SAM mode, uses the FLAG values used by Bismark v0.8.2 and before. In addition,\n",
      "                         this options appends /1 and /2 to the read IDs for reads 1 and 2 relative to the input file. Since\n",
      "                         both the appended read IDs and custom FLAG values may cause problems with some downstream tools\n",
      "                         such as Picard, new defaults were implemented as of version 0.8.3.\n",
      "\n",
      "\n",
      "                                             default                         old_flag\n",
      "                                       ===================              ===================\n",
      "                                       Read 1       Read 2              Read 1       Read 2\n",
      "\n",
      "                              OT:         99          147                  67          131\n",
      "\n",
      "                              OB:         83          163                 115          179\n",
      "\n",
      "                              CTOT:      147           99                  67          131\n",
      "\n",
      "                              CTOB:      163           83                 115          179\n",
      "\n",
      "--ambig_bam              For reads that have multiple alignments a random alignment is written out to a special file ending in\n",
      "                         '.ambiguous.bam'. The alignments are in Bowtie2 format and do not any contain Bismark specific\n",
      "                         entries such as the methylation call etc. These ambiguous BAM files are intended to be used as\n",
      "                         coverage estimators for variant callers.\n",
      "\n",
      "--nucleotide_coverage    Calculates the mono- and di-nucleotide sequence composition of covered positions in the analysed BAM\n",
      "                         file and compares it to the genomic average composition once alignments are complete by calling 'bam2nuc'.\n",
      "                         Since this calculation may take a while, bam2nuc attempts to write the genomic sequence composition\n",
      "                         into a file called 'genomic_nucleotide_frequencies.txt' indside the reference genome folder so it can\n",
      "                         be re-used the next time round instead of calculating it once again. If a file 'nucleotide_stats.txt' is\n",
      "                         found with the Bismark reports it will be automatically detected and used for the Bismark HTML report.\n",
      "                         This option works only for BAM or CRAM files.\n",
      "\n",
      "\n",
      "Other:\n",
      "\n",
      "-h/--help                Displays this help file.\n",
      "\n",
      "-v/--version             Displays version information.\n",
      "\n",
      "\n",
      "BOWTIE 2 SPECIFIC OPTIONS\n",
      "\n",
      "--bowtie1                Uses Bowtie 1 instead of Bowtie 2, which might be a good choice for faster and very short\n",
      "                         alignments. Bismark assumes that raw sequence data is adapter and/or quality trimmed where\n",
      "                         appropriate. Default: off.\n",
      "\n",
      "--bowtie2                Default: ON. Uses Bowtie 2 instead of Bowtie 1. Bismark limits Bowtie 2 to only perform end-to-end\n",
      "                         alignments, i.e. searches for alignments involving all read characters (also called\n",
      "                         untrimmed or unclipped alignments). Bismark assumes that raw sequence data is adapter\n",
      "                         and/or quality trimmed where appropriate. Both small (.bt2) and large (.bt2l) Bowtie 2\n",
      "                         indexes are supported.\n",
      "\n",
      "Bowtie 2 alignment options:\n",
      "\n",
      "-N <int>                 Sets the number of mismatches to allowed in a seed alignment during multiseed alignment.\n",
      "                         Can be set to 0 or 1. Setting this higher makes alignment slower (often much slower)\n",
      "                         but increases sensitivity. Default: 0. This option is only available for Bowtie 2 (for\n",
      "                         Bowtie 1 see -n).\n",
      "\n",
      "-L <int>                 Sets the length of the seed substrings to align during multiseed alignment. Smaller values\n",
      "                         make alignment slower but more senstive. Default: the --sensitive preset of Bowtie 2 is\n",
      "                         used by default, which sets -L to 20. maximum of L can be set to 32. The length of the seed\n",
      "                         would effect the alignment speed dramatically while the larger L, the faster the aligment.\n",
      "                         This option is only available for Bowtie 2 (for Bowtie 1 see -l).\n",
      "\n",
      "--ignore-quals           When calculating a mismatch penalty, always consider the quality value at the mismatched\n",
      "                         position to be the highest possible, regardless of the actual value. I.e. input is treated\n",
      "                         as though all quality values are high. This is also the default behavior when the input\n",
      "                         doesn't specify quality values (e.g. in -f mode). This option is invariable and on by default.\n",
      "\n",
      "\n",
      "Bowtie 2 paired-end options:\n",
      "\n",
      "--no-mixed               This option disables Bowtie 2's behavior to try to find alignments for the individual mates if\n",
      "                         it cannot find a concordant or discordant alignment for a pair. This option is invariable and\n",
      "                         and on by default.\n",
      "\n",
      "--no-discordant          Normally, Bowtie 2 looks for discordant alignments if it cannot find any concordant alignments.\n",
      "                         A discordant alignment is an alignment where both mates align uniquely, but that does not\n",
      "                         satisfy the paired-end constraints (--fr/--rf/--ff, -I, -X). This option disables that behavior\n",
      "                         and it is on by default.\n",
      "\n",
      "--no_dovetail            It is possible, though unusual, for the mates to \"dovetail\", with the mates seemingly extending\n",
      "                         \"past\" each other as in this example:\n",
      "\n",
      "                         Mate 1:                 GTCAGCTACGATATTGTTTGGGGTGACACATTACGC\n",
      "                         Mate 2:            TATGAGTCAGCTACGATATTGTTTGGGGTGACACAT\n",
      "                         Reference: GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGCGTCTTTGAC\n",
      "\n",
      "                         By default, dovetailing is considered inconsistent with concordant alignment, but by default\n",
      "                         Bismark calls Bowtie 2 with --dovetail, causing it to consider dovetailing alignments as\n",
      "                         concordant. This becomes relevant whenever reads are clipped from their 5' end prior to mapping,\n",
      "                         e.g. because of quality or bias issues.\n",
      "\n",
      "                         Specify --no_dovetail to turn off this behaviour for paired-end libraries. Default: OFF.\n",
      "\n",
      "\n",
      "Bowtie 2 effort options:\n",
      "\n",
      "-D <int>                 Up to <int> consecutive seed extension attempts can \"fail\" before Bowtie 2 moves on, using\n",
      "                         the alignments found so far. A seed extension \"fails\" if it does not yield a new best or a\n",
      "                         new second-best alignment. Default: 15.\n",
      "\n",
      "-R <int>                 <int> is the maximum number of times Bowtie 2 will \"re-seed\" reads with repetitive seeds.\n",
      "                         When \"re-seeding,\" Bowtie 2 simply chooses a new set of reads (same length, same number of\n",
      "                         mismatches allowed) at different offsets and searches for more alignments. A read is considered\n",
      "                         to have repetitive seeds if the total number of seed hits divided by the number of seeds\n",
      "                         that aligned at least once is greater than 300. Default: 2.\n",
      "\n",
      "Bowtie 2 parallelization options:\n",
      "\n",
      "\n",
      "-p NTHREADS              Launch NTHREADS parallel search threads (default: 1). Threads will run on separate processors/cores\n",
      "                         and synchronize when parsing reads and outputting alignments. Searching for alignments is highly\n",
      "                         parallel, and speedup is close to linear. Increasing -p increases Bowtie 2's memory footprint.\n",
      "                         E.g. when aligning to a human genome index, increasing -p from 1 to 8 increases the memory footprint\n",
      "                         by a few hundred megabytes. This option is only available if bowtie is linked with the pthreads\n",
      "                         library (i.e. if BOWTIE_PTHREADS=0 is not specified at build time). In addition, this option will\n",
      "                         automatically use the option '--reorder', which guarantees that output SAM records are printed in\n",
      "                         an order corresponding to the order of the reads in the original input file, even when -p is set\n",
      "                         greater than 1 (Bismark requires the Bowtie 2 output to be this way). Specifying --reorder and\n",
      "                         setting -p greater than 1 causes Bowtie 2 to run somewhat slower and use somewhat more memory then\n",
      "                         if --reorder were not specified. Has no effect if -p is set to 1, since output order will naturally\n",
      "                         correspond to input order in that case.\n",
      "\n",
      "Bowtie 2 Scoring options:\n",
      "\n",
      "--score_min <func>       Sets a function governing the minimum alignment score needed for an alignment to be considered\n",
      "                         \"valid\" (i.e. good enough to report). This is a function of read length. For instance, specifying\n",
      "                         L,0,-0.2 sets the minimum-score function f to f(x) = 0 + -0.2 * x, where x is the read length.\n",
      "                         See also: setting function options at http://bowtie-bio.sourceforge.net/bowtie2. The default is\n",
      "                         L,0,-0.2.\n",
      "\n",
      "--rdg <int1>,<int2>      Sets the read gap open (<int1>) and extend (<int2>) penalties. A read gap of length N gets a penalty\n",
      "                         of <int1> + N * <int2>. Default: 5, 3.\n",
      "\n",
      "--rfg <int1>,<int2>      Sets the reference gap open (<int1>) and extend (<int2>) penalties. A reference gap of length N gets\n",
      "                         a penalty of <int1> + N * <int2>. Default: 5, 3.\n",
      "\n",
      "\n",
      "Bowtie 2 Reporting options:\n",
      "\n",
      "-most_valid_alignments <int> This used to be the Bowtie 2 parameter -M. As of Bowtie 2 version 2.0.0 beta7 the option -M is\n",
      "                         deprecated. It will be removed in subsequent versions. What used to be called -M mode is still the\n",
      "                         default mode, but adjusting the -M setting is deprecated.  Use the -D and -R options to adjust the\n",
      "                         effort expended to find valid alignments.\n",
      "\n",
      "                         For reference, this used to be the old (now deprecated) description of -M:\n",
      "                         Bowtie 2 searches for at most <int>+1 distinct, valid alignments for each read. The search terminates when it\n",
      "                         can't find more distinct valid alignments, or when it finds <int>+1 distinct alignments, whichever\n",
      "                         happens first. Only the best alignment is reported. Information from the other alignments is used to\n",
      "                         estimate mapping quality and to set SAM optional fields, such as AS:i and XS:i. Increasing -M makes\n",
      "                         Bowtie 2 slower, but increases the likelihood that it will pick the correct alignment for a read that\n",
      "                         aligns many places. For reads that have more than <int>+1 distinct, valid alignments, Bowtie 2 does not\n",
      "                         guarantee that the alignment reported is the best possible in terms of alignment score. -M is\n",
      "                         always used and its default value is set to 10.\n",
      "\n",
      "\n",
      "'VANILLA' Bismark  OUTPUT:\n",
      "\n",
      "Single-end output format (tab-separated):\n",
      "\n",
      " (1) <seq-ID>\n",
      " (2) <read alignment strand>\n",
      " (3) <chromosome>\n",
      " (4) <start position>\n",
      " (5) <end position>\n",
      " (6) <observed bisulfite sequence>\n",
      " (7) <equivalent genomic sequence>\n",
      " (8) <methylation call>\n",
      " (9) <read conversion\n",
      "(10) <genome conversion>\n",
      "(11) <read quality score (Phred33)>\n",
      "\n",
      "\n",
      "Paired-end output format (tab-separated):\n",
      " (1) <seq-ID>\n",
      " (2) <read 1 alignment strand>\n",
      " (3) <chromosome>\n",
      " (4) <start position>\n",
      " (5) <end position>\n",
      " (6) <observed bisulfite sequence 1>\n",
      " (7) <equivalent genomic sequence 1>\n",
      " (8) <methylation call 1>\n",
      " (9) <observed bisulfite sequence 2>\n",
      "(10) <equivalent genomic sequence 2>\n",
      "(11) <methylation call 2>\n",
      "(12) <read 1 conversion\n",
      "(13) <genome conversion>\n",
      "(14) <read 1 quality score (Phred33)>\n",
      "(15) <read 2 quality score (Phred33)>\n",
      "\n",
      "\n",
      "Bismark SAM OUTPUT (default):\n",
      "\n",
      " (1) QNAME  (seq-ID)\n",
      " (2) FLAG   (this flag tries to take the strand a bisulfite read originated from into account (this is different from ordinary DNA alignment flags!))\n",
      " (3) RNAME  (chromosome)\n",
      " (4) POS    (start position)\n",
      " (5) MAPQ   (always 255 for use with Bowtie)\n",
      " (6) CIGAR\n",
      " (7) RNEXT\n",
      " (8) PNEXT\n",
      " (9) TLEN\n",
      "(10) SEQ\n",
      "(11) QUAL   (Phred33 scale)\n",
      "(12) NM-tag (edit distance to the reference)\n",
      "(13) MD-tag (base-by-base mismatches to the reference (handles indels)\n",
      "(14) XM-tag (methylation call string)\n",
      "(15) XR-tag (read conversion state for the alignment)\n",
      "(16) XG-tag (genome conversion state for the alignment)\n",
      "(17) XA/XB-tag (non-bisulfite mismatches) (optional!)\n",
      "\n",
      "Each read of paired-end alignments is written out in a separate line in the above format.\n",
      "\n",
      "\n",
      "Last modified on 10 October 2017\n"
     ]
    }
   ],
   "source": [
    "! ../../../../../Shared/Apps/Bismark_v0.19.0/bismark -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " The trimmed *C. virginica* sequences I need to use are in [this OWL folder](http://owl.fish.washington.edu/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/). Information about trimming and QC can be found in [Sam's notebook](http://onsnetwork.org/kubu4/2018/04/11/trimgalorefastqcmultiqc-trim-10bp-53-ends-c-virginica-mbd-bs-seq-fastq-data/). To run the `find` + `xargs` commands I want, I need to specify this path. I'm going to run this piece of code first just to confirm that I'm selecting the right files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_10_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_1_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_2_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_3_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_4_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_5_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_6_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_7_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_8_s1_R1_val_1.fq.gz\n",
      "/Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_9_s1_R1_val_1.fq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_*R1*.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10\n",
      "zr2096_1\n",
      "zr2096_2\n",
      "zr2096_3\n",
      "zr2096_4\n",
      "zr2096_5\n",
      "zr2096_6\n",
      "zr2096_7\n",
      "zr2096_8\n",
      "zr2096_9\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_*R1*.fq.gz \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### What I need for this command:\n",
    "\n",
    "1. Find all the *C. virginica* sequence files using `find`\n",
    "2. Isolate the base name from each of these files using `xargs` that I will then pipe into...\n",
    "\n",
    "*drumroll*\n",
    "\n",
    "1. ...The path to `bismark`\n",
    "2. --non_directional: See [this issue](https://github.com/RobertsLab/resources/issues/216)\n",
    "3. -p 4: Number of simultaneous threads to run\n",
    "4. -u 10000: Only aligning the first 10,000 reads\n",
    "5. -score_min: This is the main paramter I want to test. Running this command with the default setting (L,0,-0,2) is stringent and led to ~20% mapping efficiency, if not lower. I will run this with three different options: L,0,-0.6; L,0,-0.9; L,0,-1.2\n",
    "5. --genome + path to the folder with the .fa genome, which also has all of the bisulfite genome directories\n",
    "6. -1 + Path to first paired file in the Athaliana folder\n",
    "7. -2 + Path to second paired file in the Athaliana folder\n",
    "8. Path to redirect standard error output.\n",
    "\n",
    "Note that I'm not including a `-score_min` option, since I want to just run this alignment with the default setting."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2a. -score_min = L,0,-0.6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3664\n",
      "Mapping efficiency:\t36.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4659\n",
      "Sequence pairs did not map uniquely:\t1677\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t972\t((converted) top strand)\n",
      "GA/CT/CT:\t890\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t929\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t873\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1554\n",
      "Mapping efficiency:\t15.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7710\n",
      "Sequence pairs did not map uniquely:\t736\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t397\t((converted) top strand)\n",
      "GA/CT/CT:\t390\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t374\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t393\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3235\n",
      "Mapping efficiency:\t32.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5100\n",
      "Sequence pairs did not map uniquely:\t1665\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t829\t((converted) top strand)\n",
      "GA/CT/CT:\t807\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t817\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t782\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3723\n",
      "Mapping efficiency:\t37.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4510\n",
      "Sequence pairs did not map uniquely:\t1767\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t892\t((converted) top strand)\n",
      "GA/CT/CT:\t934\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t971\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t926\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3596\n",
      "Mapping efficiency:\t36.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4569\n",
      "Sequence pairs did not map uniquely:\t1835\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t929\t((converted) top strand)\n",
      "GA/CT/CT:\t887\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t856\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t924\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3458\n",
      "Mapping efficiency:\t34.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4975\n",
      "Sequence pairs did not map uniquely:\t1567\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t864\t((converted) top strand)\n",
      "GA/CT/CT:\t886\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t835\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t873\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3672\n",
      "Mapping efficiency:\t36.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4498\n",
      "Sequence pairs did not map uniquely:\t1830\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t989\t((converted) top strand)\n",
      "GA/CT/CT:\t856\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t864\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t963\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3460\n",
      "Mapping efficiency:\t34.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5043\n",
      "Sequence pairs did not map uniquely:\t1497\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t881\t((converted) top strand)\n",
      "GA/CT/CT:\t850\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t853\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t876\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3173\n",
      "Mapping efficiency:\t31.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5380\n",
      "Sequence pairs did not map uniquely:\t1447\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t795\t((converted) top strand)\n",
      "GA/CT/CT:\t787\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t782\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t809\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3295\n",
      "Mapping efficiency:\t33.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5111\n",
      "Sequence pairs did not map uniquely:\t1594\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t844\t((converted) top strand)\n",
      "GA/CT/CT:\t795\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t710\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t946\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "find /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_*R1*.fq.gz \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Users/Shared/Apps/Bismark_v0.19.0/bismark \\\n",
    "--non_directional \\\n",
    "-p 4 \\\n",
    "-u 10000 \\\n",
    "-score_min L,0,-0.6 \\\n",
    "--genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ \\\n",
    "-1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark-1003-L006.err"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mapping efficiency was 30-40%, except for sample 1 (15%). I'm going to move this output to a new subdirectory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-10-03-L006-Output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-10-03-L006-Output\u001b[m\u001b[m/\r\n",
      "bismark-1003-L006.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L006.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F *.*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "! mv *.* 2018-10-03-L006-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L006.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F 2018-10-03-L006-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-10-03-L006-Output\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2b. -score_min = L,0,-0.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4487\n",
      "Mapping efficiency:\t44.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3611\n",
      "Sequence pairs did not map uniquely:\t1902\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1170\t((converted) top strand)\n",
      "GA/CT/CT:\t1102\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1124\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1091\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2025\n",
      "Mapping efficiency:\t20.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7131\n",
      "Sequence pairs did not map uniquely:\t844\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t509\t((converted) top strand)\n",
      "GA/CT/CT:\t515\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t478\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t523\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4019\n",
      "Mapping efficiency:\t40.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4124\n",
      "Sequence pairs did not map uniquely:\t1857\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1047\t((converted) top strand)\n",
      "GA/CT/CT:\t992\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1007\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t973\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4532\n",
      "Mapping efficiency:\t45.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3458\n",
      "Sequence pairs did not map uniquely:\t2010\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1080\t((converted) top strand)\n",
      "GA/CT/CT:\t1140\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1177\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1135\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4469\n",
      "Mapping efficiency:\t44.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3493\n",
      "Sequence pairs did not map uniquely:\t2038\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1159\t((converted) top strand)\n",
      "GA/CT/CT:\t1083\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1068\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1159\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4289\n",
      "Mapping efficiency:\t42.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3927\n",
      "Sequence pairs did not map uniquely:\t1784\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1072\t((converted) top strand)\n",
      "GA/CT/CT:\t1068\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1039\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1110\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4501\n",
      "Mapping efficiency:\t45.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3432\n",
      "Sequence pairs did not map uniquely:\t2067\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1229\t((converted) top strand)\n",
      "GA/CT/CT:\t1025\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1044\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1203\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4291\n",
      "Mapping efficiency:\t42.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4000\n",
      "Sequence pairs did not map uniquely:\t1709\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1081\t((converted) top strand)\n",
      "GA/CT/CT:\t1052\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1068\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1090\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3900\n",
      "Mapping efficiency:\t39.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4461\n",
      "Sequence pairs did not map uniquely:\t1639\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t953\t((converted) top strand)\n",
      "GA/CT/CT:\t988\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t949\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1010\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4119\n",
      "Mapping efficiency:\t41.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4062\n",
      "Sequence pairs did not map uniquely:\t1819\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1069\t((converted) top strand)\n",
      "GA/CT/CT:\t989\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t888\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1173\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "find /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_*R1*.fq.gz \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Users/Shared/Apps/Bismark_v0.19.0/bismark \\\n",
    "--non_directional \\\n",
    "-p 4 \\\n",
    "-u 10000 \\\n",
    "-score_min L,0,-0.9 \\\n",
    "--genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ \\\n",
    "-1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark-1003-L009.err"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mapping efficiency was 40-50%, except for sample 1 (20%)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-10-03-L009-Output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L009.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F *.*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mv *.* 2018-10-03-L009-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L009.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F 2018-10-03-L009-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-10-03-L006-Output\u001b[m\u001b[m/ \u001b[34m2018-10-03-L009-Output\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2c. -score_min = L,0,-1.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5299\n",
      "Mapping efficiency:\t53.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2607\n",
      "Sequence pairs did not map uniquely:\t2094\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1386\t((converted) top strand)\n",
      "GA/CT/CT:\t1293\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1328\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1292\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2878\n",
      "Mapping efficiency:\t28.8%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t6110\n",
      "Sequence pairs did not map uniquely:\t1012\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t710\t((converted) top strand)\n",
      "GA/CT/CT:\t755\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t692\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t721\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4978\n",
      "Mapping efficiency:\t49.8%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2974\n",
      "Sequence pairs did not map uniquely:\t2048\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1295\t((converted) top strand)\n",
      "GA/CT/CT:\t1242\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1251\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1190\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5359\n",
      "Mapping efficiency:\t53.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2484\n",
      "Sequence pairs did not map uniquely:\t2157\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1299\t((converted) top strand)\n",
      "GA/CT/CT:\t1327\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1398\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1335\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5293\n",
      "Mapping efficiency:\t52.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2473\n",
      "Sequence pairs did not map uniquely:\t2234\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1374\t((converted) top strand)\n",
      "GA/CT/CT:\t1269\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1281\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1369\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5173\n",
      "Mapping efficiency:\t51.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2846\n",
      "Sequence pairs did not map uniquely:\t1981\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1303\t((converted) top strand)\n",
      "GA/CT/CT:\t1304\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1251\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1315\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5375\n",
      "Mapping efficiency:\t53.8%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2366\n",
      "Sequence pairs did not map uniquely:\t2259\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1444\t((converted) top strand)\n",
      "GA/CT/CT:\t1228\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1259\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1444\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5144\n",
      "Mapping efficiency:\t51.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2966\n",
      "Sequence pairs did not map uniquely:\t1890\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1297\t((converted) top strand)\n",
      "GA/CT/CT:\t1237\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1285\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1325\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4760\n",
      "Mapping efficiency:\t47.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3417\n",
      "Sequence pairs did not map uniquely:\t1823\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1196\t((converted) top strand)\n",
      "GA/CT/CT:\t1200\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1168\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1196\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4994\n",
      "Mapping efficiency:\t49.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2979\n",
      "Sequence pairs did not map uniquely:\t2027\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1324\t((converted) top strand)\n",
      "GA/CT/CT:\t1206\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1070\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1394\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "find /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_*R1*.fq.gz \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Users/Shared/Apps/Bismark_v0.19.0/bismark \\\n",
    "--non_directional \\\n",
    "-p 4 \\\n",
    "-u 10000 \\\n",
    "-score_min L,0,-1.2 \\\n",
    "--genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ \\\n",
    "-1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark-1003-L012.err"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mapping efficiency was 50-55%, except for sample 1 (28%)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 2018-10-03-L012-Output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L012.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F *.*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mv *.* 2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L012.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F 2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-10-03-L006-Output\u001b[m\u001b[m/ \u001b[34m2018-10-03-L009-Output\u001b[m\u001b[m/ \u001b[34m2018-10-03-L012-Output\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "!ls -F"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "e| **Sample** | **L,0,-0.6** | **L,0,-0.9** | **L,0,-1.2** |\n",
    "|:----------:|:------------:|:------------:|:------------:|\n",
    "|      1     |     15.5     |     20.2     |     28.8     |\n",
    "|      2     |     32.4     |     40.2     |     49.8     |\n",
    "|      3     |     37.2     |     45.3     |     53.6     |\n",
    "|      4     |     36.0     |     44.7     |     52.9     |\n",
    "|      5     |     34.6     |     42.9     |     51.7     |\n",
    "|      6     |     36.7     |     45.0     |     53.8     |\n",
    "|      7     |     34.6     |     42.9     |     51.4     |\n",
    "|      8     |     31.7     |     39.0     |     47.6     |\n",
    "|      9     |     33.0     |     41.2     |     49.9     |\n",
    "|     10     |     36.6     |     44.9     |     53.0     |"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To optimize mapping efficiency, I will use `-score_min L,0,-1.2`. Sample 1 is concerning, but I will keep an eye on it. All associated files can now be found on [gannett](http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Deduplication, Sorting, and Indexing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We deemed the `-score_min L,0,-1.2` option was the best moving forward. I will use the subset data generated with this parameter to test out options in `methylKit`. To do this, I need to finish the `bismark` pipeline, starting with deduplication, sorting, and indexing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3a. Download files from gannet"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since I have gannet mounted on this machine, I could directly alter the files on gannet. However, I don't want to permanently change files on accident. Instead, I'll move files from gannet, work on them locally, then put the output back on gannet. If I didn't have gannet mounted, I could `curl` the files from the above link."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2018-10-11 14:16:58--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/\n",
      "Resolving gannet.fish.washington.edu... 128.95.149.52\n",
      "Connecting to gannet.fish.washington.edu|128.95.149.52|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:16:58 (95.9 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html' saved [6433/6433]\n",
      "\n",
      "Loading robots.txt; please ignore errors.\n",
      "--2018-10-11 14:16:58--  http://gannet.fish.washington.edu/robots.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 404 Not Found\n",
      "2018-10-11 14:16:58 ERROR 404: Not Found.\n",
      "\n",
      "--2018-10-11 14:16:58--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=N;O=D\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=N;O=D'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:16:59 (80.7 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=N;O=D' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:16:59--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=M;O=A\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=M;O=A'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:16:59 (66.0 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=M;O=A' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:16:59--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=S;O=A\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=S;O=A'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (131 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=S;O=A' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=D;O=A\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=D;O=A'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (81.8 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=D;O=A' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/2018-10-03-L012-Output-checksum.sha\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1776 (1.7K)\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/2018-10-03-L012-Output-checksum.sha'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.73K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (130 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/2018-10-03-L012-Output-checksum.sha' saved [1776/1776]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/bismark-1003-L012.err\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 90343 (88K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/bismark-1003-L012.err'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]  88.23K  --.-KB/s    in 0.002s  \n",
      "\n",
      "2018-10-11 14:17:00 (48.5 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/bismark-1003-L012.err' saved [90343/90343]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1928 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.88K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (131 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1928/1928]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 501948 (490K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 490.18K  --.-KB/s    in 0.007s  \n",
      "\n",
      "2018-10-11 14:17:00 (64.8 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam' saved [501948/501948]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (263 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 826534 (807K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 807.16K  --.-KB/s    in 0.02s   \n",
      "\n",
      "2018-10-11 14:17:00 (45.4 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam' saved [826534/826534]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (230 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 949058 (927K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 926.81K  --.-KB/s    in 0.01s   \n",
      "\n",
      "2018-10-11 14:17:00 (95.3 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam' saved [949058/949058]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (154 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 906350 (885K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 885.11K  --.-KB/s    in 0.02s   \n",
      "\n",
      "2018-10-11 14:17:00 (47.3 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam' saved [906350/906350]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (263 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 903187 (882K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 882.02K  --.-KB/s    in 0.02s   \n",
      "\n",
      "2018-10-11 14:17:00 (44.0 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam' saved [903187/903187]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (184 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 913162 (892K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 891.76K  --.-KB/s    in 0.03s   \n",
      "\n",
      "2018-10-11 14:17:00 (30.1 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam' saved [913162/913162]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:00 (230 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:00--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 917284 (896K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 895.79K  --.-KB/s    in 0.02s   \n",
      "\n",
      "2018-10-11 14:17:01 (37.1 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam' saved [917284/917284]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:01 (205 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 831282 (812K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 811.80K  --.-KB/s    in 0.04s   \n",
      "\n",
      "2018-10-11 14:17:01 (21.6 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam' saved [831282/831282]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1933 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:01 (263 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1933/1933]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 916727 (895K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 895.24K  --.-KB/s    in 0.01s   \n",
      "\n",
      "2018-10-11 14:17:01 (83.1 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam' saved [916727/916727]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1935 (1.9K) [text/plain]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   1.89K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:01 (185 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt' saved [1935/1935]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 980570 (958K) [application/octet-stream]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam'\n",
      "\n",
      "gannet.fish.washing 100%[===================>] 957.59K  --.-KB/s    in 0.01s   \n",
      "\n",
      "2018-10-11 14:17:01 (69.5 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam' saved [980570/980570]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=N;O=A\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=N;O=A'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:01 (150 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=N;O=A' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:17:01--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=M;O=D\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=M;O=D'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:02 (123 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=M;O=D' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:17:02--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=S;O=D\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=S;O=D'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:02 (204 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=S;O=D' saved [6433/6433]\n",
      "\n",
      "--2018-10-11 14:17:02--  http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/?C=D;O=D\n",
      "Reusing existing connection to gannet.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 6433 (6.3K) [text/html]\n",
      "Saving to: 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=D;O=D'\n",
      "\n",
      "gannet.fish.washing 100%[===================>]   6.28K  --.-KB/s    in 0s      \n",
      "\n",
      "2018-10-11 14:17:03 (204 MB/s) - 'gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/index.html?C=D;O=D' saved [6433/6433]\n",
      "\n",
      "FINISHED --2018-10-11 14:17:03--\n",
      "Total wall clock time: 5.2s\n",
      "Downloaded: 31 files, 8.4M in 0.2s (44.8 MB/s)\n"
     ]
    }
   ],
   "source": [
    "#Use wget to download the entire directory\n",
    "! wget --no-parent -r http://gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34mgannet.fish.washington.edu\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "! ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/gannet.fish.washington.edu\n"
     ]
    }
   ],
   "source": [
    "cd gannet.fish.washington.edu/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34mspartina\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "! ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output\n"
     ]
    }
   ],
   "source": [
    "cd spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2018-10-03-L012-Output-checksum.sha\r\n",
      "bismark-1003-L012.err\r\n",
      "index.html\r\n",
      "index.html?C=D;O=A\r\n",
      "index.html?C=D;O=D\r\n",
      "index.html?C=M;O=A\r\n",
      "index.html?C=M;O=D\r\n",
      "index.html?C=N;O=A\r\n",
      "index.html?C=N;O=D\r\n",
      "index.html?C=S;O=A\r\n",
      "index.html?C=S;O=D\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "! ls -F"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Looks like contents I wanted are in their own folder, based on the folder structure on gannet. I'll have to move the files out of the folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing\n"
     ]
    }
   ],
   "source": [
    "cd ../../../../../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Make a new directory to hold all of the files from gannet\n",
    "mkdir 2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-10-03-L012-Output\u001b[m\u001b[m/     \u001b[34mgannet.fish.washington.edu\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "! ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Move files from one directory to another\n",
    "! mv gannet.fish.washington.edu/spartina/2018-10-10-project-virginica-oa-Large-Files/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/* \\\n",
    "2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output\n"
     ]
    }
   ],
   "source": [
    "cd 2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2018-10-03-L012-Output-checksum.sha\r\n",
      "bismark-1003-L012.err\r\n",
      "index.html\r\n",
      "index.html?C=D;O=A\r\n",
      "index.html?C=D;O=D\r\n",
      "index.html?C=M;O=A\r\n",
      "index.html?C=M;O=D\r\n",
      "index.html?C=N;O=A\r\n",
      "index.html?C=N;O=D\r\n",
      "index.html?C=S;O=A\r\n",
      "index.html?C=S;O=D\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "#Check to see that all of the files were moved\n",
    "! ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Now I want to remove the empty directory that all of the files are from\n",
    "! rm -r ../gannet.fish.washington.edu/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m2018-10-03-L012-Output\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "! ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#I also want to remove the index.html files from the directory.\n",
    "! rm 2018-10-03-L012-Output/index.html*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output\n"
     ]
    }
   ],
   "source": [
    "cd 2018-10-03-L012-Output/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: shasum [OPTION]... [FILE]...\r\n",
      "Print or check SHA checksums.\r\n",
      "With no FILE, or when FILE is -, read standard input.\r\n",
      "\r\n",
      "  -a, --algorithm   1 (default), 224, 256, 384, 512, 512224, 512256\r\n",
      "  -b, --binary      read in binary mode\r\n",
      "  -c, --check       read SHA sums from the FILEs and check them\r\n",
      "  -t, --text        read in text mode (default)\r\n",
      "  -U, --UNIVERSAL   read in Universal Newlines mode\r\n",
      "                        produces same digest on Windows/Unix/Mac\r\n",
      "  -0, --01          read in BITS mode\r\n",
      "                        ASCII '0' interpreted as 0-bit,\r\n",
      "                        ASCII '1' interpreted as 1-bit,\r\n",
      "                        all other characters ignored\r\n",
      "  -p, --portable    read in portable mode (to be deprecated)\r\n",
      "\r\n",
      "The following two options are useful only when verifying checksums:\r\n",
      "  -s, --status      don't output anything, status code shows success\r\n",
      "  -w, --warn        warn about improperly formatted checksum lines\r\n",
      "\r\n",
      "  -h, --help        display this help and exit\r\n",
      "  -v, --version     output version information and exit\r\n",
      "\r\n",
      "When verifying SHA-512/224 or SHA-512/256 checksums, indicate the\r\n",
      "algorithm explicitly using the -a option, e.g.\r\n",
      "\r\n",
      "  shasum -a 512224 -c checksumfile\r\n",
      "\r\n",
      "The sums are computed as described in FIPS PUB 180-4.  When checking,\r\n",
      "the input should be a former output of this program.  The default\r\n",
      "mode is to print a line with checksum, a character indicating type\r\n",
      "(`*' for binary, ` ' for text, `U' for UNIVERSAL, `^' for BITS, `?'\r\n",
      "for portable), and name for each FILE.\r\n",
      "\r\n",
      "Report shasum bugs to mshelor@cpan.org\r\n"
     ]
    }
   ],
   "source": [
    "!shasum -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark-1003-L012.err: OK\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam: OK\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt: OK\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam: OK\n"
     ]
    }
   ],
   "source": [
    "#Finally, I'll verify the checksums of the downloaded files match the original checksums\n",
    "!shasum -c 2018-10-03-L012-Output-checksum.sha"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Yay everything matches! Now I can move on with deduplication."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3b. Deduplication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "===============================================================================================================\r\n",
      "\r\n",
      "This script is supposed to remove alignments to the same position in the genome from the Bismark mapping output\r\n",
      "(both single and paired-end SAM files), which can arise by e.g. excessive PCR amplification. If sequences align\r\n",
      "to the same genomic position but on different strands they will be scored individually.\r\n",
      "\r\n",
      "Note that deduplication is not recommended for RRBS-type experiments!\r\n",
      "\r\n",
      "In the default mode, the first alignment to a given position will be used irrespective of its methylation call\r\n",
      "(this is the fastest option, and as the alignments are not ordered in any way this is also near enough random).\r\n",
      "\r\n",
      "For single-end alignments only use the start coordinate of a read will be used for deduplication.\r\n",
      "\r\n",
      "For paired-end alignments the start-coordinate of the first read and the end coordinate of the second\r\n",
      "read will be used for deduplication. This script expects the Bismark output to be in SAM format\r\n",
      "(Bismark v0.6.x or higher). To deduplicate the old custom Bismark output please specify '--vanilla'.\r\n",
      "\r\n",
      "*** Please note that for paired-end BAM files the deduplication script expects Read1 and Read2 to\r\n",
      "follow each other in consecutive lines! If the file has been sorted by position make sure that you resort it\r\n",
      "by read name first (e.g. using samtools sort -n)  ***\r\n",
      "\r\n",
      "===============================================================================================================\r\n",
      "\r\n",
      ">>> USAGE: ./deduplicate_bismark_alignment_output.pl [options] filename(s) <<<\r\n",
      "\r\n",
      "-s/--single\t\tdeduplicate single-end Bismark files (default format: SAM)\r\n",
      "-p/--paired\t\tdeduplicate paired-end Bismark files (default format: SAM)\r\n",
      "\r\n",
      "--output_dir [path]\tOutput directory, either relative or absolute. Output is written to the current directory if not\r\n",
      "\t\t\tspecified explicitly.\r\n",
      "\r\n",
      "--vanilla\t\tThe input file is in the old custom Bismark format and not in SAM format\r\n",
      "\r\n",
      "--barcode\t\tIn addition to chromosome, start position and orientation this will also take a potential barcode into\r\n",
      "                        consideration while deduplicating. The barcode needs to be the last element of the read ID and separated\r\n",
      "                        by a ':', e.g.: MISEQ:14:000000000-A55D0:1:1101:18024:2858_1:N:0:CTCCT\r\n",
      "\r\n",
      "--bam\t\t\tThe output will be written out in BAM format instead of the default SAM format. This script will\r\n",
      "\t\t\tattempt to use the path to Samtools that was specified with '--samtools_path', or, if it hasn't\r\n",
      "\t\t\tbeen specified, attempt to find Samtools in the PATH. If no installation of Samtools can be found,\r\n",
      "\t\t\tthe SAM output will be compressed with GZIP instead (yielding a .sam.gz output file)\r\n",
      "\r\n",
      "--multiple\t\tAll specified input files are treated as one sample and concatenated together for deduplication.\r\n",
      "\t\t\tThis uses Unix 'cat' for SAM files and 'samtools cat' for BAM files. Additional notes for BAM files:\r\n",
      "\t\t\tAlthough this works on either BAM or CRAM, all input files must be the same format as each other.\r\n",
      "\t\t\tThe sequence dictionary of each input file must be identical, although this command does not check this.\r\n",
      "\t\t\tBy default the header is taken from the first file to be concatenated.\r\n",
      "\r\n",
      "--samtools_path\t\tThe path to your Samtools installation, e.g. /home/user/samtools/. Does not need to be specified\r\n",
      "\t\t\texplicitly if Samtools is in the PATH already\r\n",
      "\r\n",
      "--version\t\tPrint version information and exit\r\n",
      "===============================================================================================================\r\n",
      "\r\n",
      "This script was last modified on 07 September 2017\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! /Users/Shared/Apps/Bismark_v0.19.0/deduplicate_bismark -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find zr2096_*_s1_R1_val_1_bismark_bt2_pe.bam \\"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To deduplicate:\n",
    "\n",
    "1. Path to deduplicate_bismark\n",
    "2. -p: Input data is paired\n",
    "3. --bam: Write the output as a .bam file\n",
    "4. Path to input files. It's important to specify only the files from the alignment! If I just used .bam, the process would become recursive, and the command will start to deduplicate the .deduplicated.bam files that are already deduplicated!\n",
    "5. Path to standard error output."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_10_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_10_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_1_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_1_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_2_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_2_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_3_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_3_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_4_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_4_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_5_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_5_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_6_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_6_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_7_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_7_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_8_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_8_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --non_directional -p 4 -u 10000 -score_min L,0,-1.2 --genome /Users/yaamini/Documents/project-virginica-oa/analyses/2018-04-27-Bismark/2018-04-27-Bismark-Inputs/ -1 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_9_s1_R1_val_1.fq.gz -2 /Volumes/web/Athaliana/20180411_trimgalore_10bp_Cvirginica_MBD/zr2096_9_s1_R2_val_2.fq.gz\"\n"
     ]
    }
   ],
   "source": [
    "!/Users/Shared/Apps/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "-p \\\n",
    "--bam \\\n",
    "zr2096_*_s1_R1_val_1_bismark_bt2_pe.bam \\\n",
    "2> dedup-L012-1011.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dedup-L012-1011.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "! find *dedup*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n"
     ]
    }
   ],
   "source": [
    "! find *deduplicated.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3b. Sorting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The next two steps will allow me to use my .bam files in IGV. According to the [`igvtools`](https://software.broadinstitute.org/software/igv/igvtools_commandline) website, I need to `sort` then `index` all of my .bam files. Instead of `igvtools`, I will use `samtools`, since `igvtools` may not have the capacity to deal with paired-end read data (ie. my data).\n",
    "\n",
    "I will use a similar `find` and `xargs` approach to loop through all of my files.\n",
    "\n",
    "1. Identify all deduplicated .bam files using `find`\n",
    "2. Isolate the basename with `xargs` to be fed into `igvtools`\n",
    "\n",
    "THEN\n",
    "\n",
    "1. Path to `samtools`\n",
    "2. `sort`\n",
    "3. Path to input .bam file\n",
    "4. -o + Path to output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "find *deduplicated.bam \\\n",
    "| xargs basename -s _s1_R1_val_1_bismark_bt2_pe.deduplicated.bam | xargs -I{} /Users/Shared/Apps/samtools-1.8/samtools \\\n",
    "sort {}_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam \\\n",
    "-o {}_dedup.sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_dedup.sorted.bam\r\n",
      "zr2096_1_dedup.sorted.bam\r\n",
      "zr2096_2_dedup.sorted.bam\r\n",
      "zr2096_3_dedup.sorted.bam\r\n",
      "zr2096_4_dedup.sorted.bam\r\n",
      "zr2096_5_dedup.sorted.bam\r\n",
      "zr2096_6_dedup.sorted.bam\r\n",
      "zr2096_7_dedup.sorted.bam\r\n",
      "zr2096_8_dedup.sorted.bam\r\n",
      "zr2096_9_dedup.sorted.bam\r\n"
     ]
    }
   ],
   "source": [
    "!find *dedup.sorted.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3c. Indexing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, I'll index my files since each .bam in IGV needs an associated index.\n",
    "\n",
    "1. Identify all deduplicated .bam files using `find`\n",
    "2. Isolate the basename with `xargs` to be fed into `igvtools`\n",
    "\n",
    "THEN\n",
    "\n",
    "1. Path to `samtools`\n",
    "2. `index`\n",
    "3. Path to input .bam file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "find *dedup.sorted.bam \\\n",
    "| xargs basename -s _dedup.sorted.bam | xargs -I{} /Users/Shared/Apps/samtools-1.8/samtools \\\n",
    "index {}_dedup.sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_dedup.sorted.bam.bai\r\n",
      "zr2096_1_dedup.sorted.bam.bai\r\n",
      "zr2096_2_dedup.sorted.bam.bai\r\n",
      "zr2096_3_dedup.sorted.bam.bai\r\n",
      "zr2096_4_dedup.sorted.bam.bai\r\n",
      "zr2096_5_dedup.sorted.bam.bai\r\n",
      "zr2096_6_dedup.sorted.bam.bai\r\n",
      "zr2096_7_dedup.sorted.bam.bai\r\n",
      "zr2096_8_dedup.sorted.bam.bai\r\n",
      "zr2096_9_dedup.sorted.bam.bai\r\n"
     ]
    }
   ],
   "source": [
    "! find *.bai"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Methylation Extractor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I'll complete the methylation extraction and report steps so I can compare these results with what I get from `methylKit`.\n",
    "\n",
    "To use `methylKit`:\n",
    "\n",
    "1. Path to `bismark_methylation_extractor`\n",
    "2. -p: Indicate I have paired reads\n",
    "3. --bedGraph: Create `bedGraph` file with results\n",
    "4. --counts\n",
    "5. --scaffolds: Bypass limitation of open filehandles\n",
    "6. --multicore 8\n",
    "7. --gzip: Gunzip files to save space\n",
    "8. Path to .bam files generated in Step 2\n",
    "9. Path for standard error output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/yaamini/Documents/project-virginica-oa/analyses/2018-10-03-Bismark-Parameter-Testing/2018-10-03-L012-Output/CpG_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! /Users/Shared/Apps/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p \\\n",
    "--bedGraph \\\n",
    "--counts \\\n",
    "--scaffolds \\\n",
    "--multicore 8 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme-L012-1011.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!gunzip *bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\r\n"
     ]
    }
   ],
   "source": [
    "!ls *bedGraph*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. HTML Processing Reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 10 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "Redundant argument in sprintf at /Users/Shared/Apps/Bismark_v0.19.0/bismark2report line 130.\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! /Users/Shared/Apps/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Summary Reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 10 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! /Users/Shared/Apps/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Create Checksums"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#I'm overwriting the old checksum file since all of the checksums matched\n",
    "! shasum * > 2018-10-03-L012-Output-checksum.sha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "da39a3ee5e6b4b0d3255bfef95601890afd80709  2018-10-03-L012-Output-checksum.sha\n",
      "14f1254a3c4f0f27b9f6a222c260e6fb396c4864  CHG_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "425a5ecd623f6c83d3feb160c70781e18c31b232  CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "33df1472528f5c94732282617855b1115b18057b  CHG_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "5f5ef3831c5dc07a5c8151a057c29fcad8ed19fe  CHG_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "2f66b7f061d53fa89f016ff43692051b0849b201  CHG_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "ed1af63988afcaf0ab3665b047a564956ffb1423  CHG_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "f3cf582099a8b7649eee8cbf644b635b21850da0  CHG_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "8a822ccb8dc7972f5e1ece646101898509c06d5d  CHG_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "bf8eee200c9af841bd7b66b9770f5e0ce0028dd4  CHG_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "4f22227c8fdb76ad93d3e335ea7b9d699a48db16  CHG_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "4173f6b5ea913e5e09ce1f75a32aa7950dca5a10  CHG_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "6d5b9f048491c2c879717a36113e6c3b6a3f817b  CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "0c9e94cac4e76ed67f2d32d79d4ff09158859866  CHG_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "e68c163f2ef17cada9246b93f828d560bd0ba211  CHG_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d39ea60a01b114ece2a12cef4ce7786a707aa8b0  CHG_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "f78d11fbf37859a3470dedb0ecc1076866ab42a7  CHG_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "dc70b452bdb54355a52c22e3d9ff1fdef68ab5c7  CHG_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "18628530062ec322babbea3d7daf8c1812e0dd5c  CHG_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "92561bd8ba813e61c749f3e314dc8faf4e9b72d8  CHG_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "9b514a6726be662b17547f6d0ab1e05b0784c04d  CHG_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "64ee5e8f94915201018db13107b345e282c1c27e  CHG_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "20ee1b449eeb0adab98e5ed9e9009f79f79909f1  CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "8a59d2cc13c27b0a280011e1dc19adfdd07003ce  CHG_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7802902ecf43865f00b8c62ea5835bbfdc4c3e48  CHG_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "210698ea23599fe8983ada29c007a1404bdbb669  CHG_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "b3b7625de43cf9cc3d45dd172c06dda837c579e5  CHG_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7f86637018bcee5de14a9b063cc0c07bd8657aae  CHG_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "a454e3729ad79c5529b493a3c9edf4cf78891f64  CHG_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "9d3c01ab822d259d595b0de6432f364e7b240337  CHG_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "efc52e551ea57753a8ab5f73f3b12528f5180125  CHG_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "e82c8601d71846a08d9956c3cf11dd84cc4d3b0a  CHG_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "a4e3998d00aa91e00e101595109f830869a59fc1  CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "5d19bb6db9ea0a794b8731cab564585e79382eaf  CHG_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "0f7b8388ec85ad6e13d3dc0983bf379e1cea5d6f  CHG_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "6583d2a0e7eb6bf9e778a7d0d4af994c750f5c72  CHG_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "64f9ae029b6398572de2957ed7ddbb7ee6cb1bcf  CHG_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "c303d2adb63e0623ac96fbf65ec6f9533e20d545  CHG_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "1d648f58932312655941c728930b7f636589a83b  CHG_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "abc610067e5e3a1eaf9cbf6e899c61ef73da90cb  CHG_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7bff2777d52e246c577d5ec6265b75b4dda78cf7  CHG_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "b60237117fb8852b5d8de62d1b12d6b3cdf4af4f  CHH_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "6ca2218c25c9ee8963469e2dd2635d52aacbe5e8  CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "85c90e5da142f4871e3d04d597d787f025c77c12  CHH_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "ba2e8ea9ec11f4fba64a0a41628a22d51f8d0af4  CHH_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "697a06ca4e51ee569666af048dd6282684db2455  CHH_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "e944dd3b3d55956cedc99be15fbeb085c8672680  CHH_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "ac85bbbeab8a1b98abc4576279911ede17f2461f  CHH_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "008a7ff8a9695a672f71db26f91ba5cce2c49628  CHH_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "3925ceba683359c994586f6b697d2dfe88acdba3  CHH_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "4b31885ebafbf40c668b42ac3194b0a788edf7d8  CHH_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "f4708f84ce2ae2caaf07514fbd7fb8bf2ea767c3  CHH_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "8b4561e316adf260dec12bf69368759c32fe6cf8  CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "33bcbc07d3490ced1a1124963b3733f4adcc6bd5  CHH_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "3067f0263892cbb16ed2edbcac492fb6f2527e84  CHH_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "9284646308b95a0dbe6b1cb56007fe18a546774d  CHH_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "3a8c464e726b83b7a3563767f6ce898c3b54bb17  CHH_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "47d6d18e32afd19cfe533af850ae32b7a03d4f41  CHH_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "b7a53e8290d0de2038e3f8334984739fd1f0d440  CHH_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "4550a652a641e668017e0b0f86fbe74b56024d73  CHH_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "9e1782a838b69be81123a685137990b67978255f  CHH_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "5154f8c37f5dab5f71f3ea386d699d6ae5af9684  CHH_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "111581ab41291c32c01d44ce173e617535d6cbe8  CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "1cfd01c78e641ab324a1329dbd739ca097aada07  CHH_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "cee75a114dddec1c27f2b115548ac96ee3fb101c  CHH_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d48f19851a60ac4bc017098efc583d42f2d0eca1  CHH_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d81a5e11fd246be9f6cc43fefc68c6293ad26596  CHH_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "38c5c7fd6e2e1d18915bc145bcdf5c3073851763  CHH_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "8f03fc4de418c971bb9d5476d44ec089c07881ce  CHH_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "ab5378e1cdf1de10e99f00a93c187df39ea1f0c7  CHH_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "990394b90fbbc273023c918ed3c756ba9b6a9aa9  CHH_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "858553063845bba4434103a9863e8c2c0da111c6  CHH_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "43930dc7940553f8ee719a1a7607b4bc263a85c1  CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "a95802877b2bbe63e3f34d9ff8e46df2446ce60c  CHH_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7bd7094ccc63a765a2953119a75e901825463510  CHH_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "14c938a7e1dc744c9baa50df6a4bcc713cb06be9  CHH_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "1a881bf08e6d43d8880edbb552661a1df51646d9  CHH_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "dd02b4b2089cd5cd801e5b6e6fec66dd5a33111f  CHH_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "2dcadaaa69e1d5ae38df01645c511b9f72d303e3  CHH_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "65b26d49b4fafc98674d9138350b02097c0c18f9  CHH_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "40aeda719e334df65eede1364bf4fb3b44516a66  CHH_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "f0c7aa4b79bb95b43e16d68c43600dcf805bea28  CpG_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "fc0512a0bb1fdbd513d713231aade25fd90e8a87  CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "50ea7afccf3e795b5319bb8a1b1bdbc5592216d8  CpG_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "c84821afe27e71deaaa2302f6bb33e415051b0f7  CpG_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d9c10b4e47134d07ab7724c9df9df47efc74a03c  CpG_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "8f0ec51453593ecd32b65ec88968f4dd232ce0d1  CpG_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "782769503af0a659797c95b7fde21366b3f04940  CpG_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "b168e37975e7d6368a596ae1ba5f0b4ae9d9d6ae  CpG_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "12f85fd3940791c96457d98b1f9ba35bc1ae4376  CpG_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "4e0f6b7db536a9dcbbe8a6177ca4186d78673701  CpG_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "00969a2cd23ecb7d78c1d1ac64f2127f86e6b170  CpG_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "ec00b2273c4cdcbaf2df3c39bd298cc8a9ac36e3  CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "77ae3ee58b5aaba028b084dbde30297d7ad4f151  CpG_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "8f8b763ec84fff3612d886e8f43e9b47a1cb1f87  CpG_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "76a2b971b356933d99f2ce625a763546cdea752d  CpG_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "c32a9ce6faef1447f07fb61f47cc561549a7f04b  CpG_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "830d409c48b14edf25ed299f6e49e9c83b24a9cc  CpG_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d043805b6366ca2ec10956dee9b079652d092cbc  CpG_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7489f6ac151f285693c51384f8bb90fb9119c948  CpG_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "10eeca5f912b248023102cff3c7e60aa788082d1  CpG_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "19bcccbd9a553198434252a95a9b22dfd080e7a7  CpG_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "a28d7af4310b67d5a43d4de0401764330e09b05e  CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d9eb8211aec86fca1fb65fe2fdca31ea04ecc880  CpG_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "e775c9b33c64f3ea5c3a98416e863865502ee4fe  CpG_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "4c3216d6faf964727246c16be94bb3ed79364c8a  CpG_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "b7e90551f2485d4ea117884dbc2ea012f22b18c5  CpG_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "933698f4a68777a33697a4d14c168748915b42c3  CpG_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "b76ca02f0843d6ceb15cf83ebe1687f8a684728d  CpG_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7d40cf6925420b60c688eeda990279b23d8b55d6  CpG_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "478c29644178b3512fc3bf5b0f7d25897ce377fb  CpG_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "7192616ee6ca75e189720d8f91e3c4fc047a946f  CpG_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "48ff4fa3520d7d445f899b6414bb402a4d93643d  CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "dc935a1af3a91a953dd8ad2de8aab6f0b62db7fe  CpG_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "cb729daa9ef3502ff9cf5f30e6c12264fe0bb9c5  CpG_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "a973a97bafe1a3c1db21f997c51f42d68c1ef271  CpG_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "dd3b67d1809f41517fc91b86ed5c5a3c64807593  CpG_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "30298a2ea798a97752027c0e832fdab4e293eb6d  CpG_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "d72f684a3d54802185ef7082a007a3618f4f2c13  CpG_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "6104df2dc34a3fc403cb915d2e388339eb1ff69b  CpG_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "53c51e2ab35f677cb65afcaaf128b05b6ace3b61  CpG_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "3544280bf396b6d4d143a821d4b265376da7ac39  bismark-1003-L012.err\n",
      "56d3a425bd4c57473cc61426126e2a0260441310  bismark_summary_report.html\n",
      "2179123903a2017ebc7f030e08bfda7d548f5cf7  bismark_summary_report.txt\n",
      "d3436e0ea004138b59a7813ff322777c6e0fbf12  bme-L012-1011.err\n",
      "0d05e94565d3140166f8bab74bc0c987fb7f8f1b  dedup-L012-1011.err\n",
      "0a51ec60db19f01be37815a98101865eb75752ec  zr2096_10_dedup.sorted.bam\n",
      "4ed9a6c318dde1c15b97e6df16396527e9dd24a5  zr2096_10_dedup.sorted.bam.bai\n",
      "0a71d91f5b4709e69930e976a39bd4c7c94176a4  zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "e9a24f689a1be9d9debe326ee1dfb60e8f16af88  zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "ee7eb103fc08e99b341673753b6f4ba1efc1463f  zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "8eca30fb7f4a4ddcd77d3e064dde4012c7f360ed  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "ee7eb103fc08e99b341673753b6f4ba1efc1463f  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "72e3cce716b89976b0535f96ba0ee947f5d7654f  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "993ccdaaa8ea43dd8931962fffaa84ae69dd28d7  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "a9c13d02478cd00d728d6d210d118c02e99d7c76  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "f1d808fd25a0a0d77e3c70c7e2ce8a8dd2a1157f  zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "9a3724395be6c5490ccea8d95908d54c5a54d1fe  zr2096_1_dedup.sorted.bam\n",
      "8c01e2b29e6375cd6d4e65dac58ed364cbdfb000  zr2096_1_dedup.sorted.bam.bai\n",
      "518b3a992acc240aa662608da2355829323499c9  zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "b5f170beb7db499c34cef0d1a0acf18033da89da  zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "e66a5b7bdeeea5b7c396993704e83153d1ef3cb9  zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "bf431b01c323a08caf1b3c0fe1d992bc3e6db591  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "9ba42668aaf0afb6a651c72f1e250f2b0e1c5c1c  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "a04587607c4e4bcd73c2db46e5dd059f309f00c7  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "bf5585099cf81fbb2e994254b29796c6cfe1d557  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "6fd2ea4b5a6a1ceb69bde6197d41a8f24d016dfb  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "8698d30da5d670725f8f9e8e2438867d6d52e35b  zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "4ed4004dc1b1579ad015929c78e8b6dd4623b57b  zr2096_2_dedup.sorted.bam\n",
      "f85539da6855419c6c11987b5b0b7d71498abc58  zr2096_2_dedup.sorted.bam.bai\n",
      "ee8b891b0578811bfe9d172acf95aa68ca751d2e  zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "b26ef3b52bc5854a21225056a03933a0169e49ac  zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "22d9bfc2032fad32c8d230febfcd0162e28539a3  zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "d6e6a2dd12975287a1683885a54955478b6e0f4b  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "ca07388bcc12865613ed60146a1c1a5786e17364  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "788cedeb003a8da2e2056d34b2e2cec17427a163  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "5bd5d3983c8dd0288cdf471071b8909b3cd13694  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "341387874d8b2b15478eb27112de27f7035f2812  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "2c8296697b181c4e48719cfe503282215aa8527b  zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "6f158bff05001d0259688950f099d8c0bea0ec1b  zr2096_3_dedup.sorted.bam\n",
      "bb1d4c20be4a82edcee71bbf48ad0dc915688350  zr2096_3_dedup.sorted.bam.bai\n",
      "d20d288bd3e80e86a6c61c3b4abffad20b572ea8  zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "c8d8ebfd68e50d561e3552d0e459c26f00b2d744  zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "f52996cf7de1311123a6818215e9fe0e50f4f62d  zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "fba53cb5e8617d1161d53e10684c769e10b5260e  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "5d85628c9130348021320cc8926a6e9b76e6237d  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "0da9e6a4b891cbfaa87fa19090f023c32da03747  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "85b4fee0906a96611c45c71268af734bcf8f1976  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "cde010f54e30d31d1c3f1be2a0180cff17970cb1  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "ffedd9de4bd244a07da4f170ef545d71c35228e8  zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "1af9ba9f2dfb73b6ffad7c0d4081865d5eb1264d  zr2096_4_dedup.sorted.bam\n",
      "009852f20fc4210252435cbd969c27ccd840bac2  zr2096_4_dedup.sorted.bam.bai\n",
      "1744f9da61419940c1ac30f0dce2a9398b9d615c  zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "f516c160571074b62142f378d8b2084aafb9145c  zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "aabd92fa6a511d940ddf366590329aca76173844  zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "809af15f16fe3f652592b0d8f30ef3c57187ab32  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "aabd92fa6a511d940ddf366590329aca76173844  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "b1114cf0afa24d14d9b1939b79b6f71f17952743  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "48e65a42ace82140dba21c615d9b405e6604c6b5  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "19d2a0ae7bbdbdc91713fda119c2f4f988ca8f91  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "d88bab80b2024549c6e82b6fedd950cd422dff3d  zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "aba3e01ec5f21b2cc336614d6581dd93f1d578ff  zr2096_5_dedup.sorted.bam\n",
      "eec0dac976e1874b7cfd7b80fb1353afb055b056  zr2096_5_dedup.sorted.bam.bai\n",
      "6ad796381505e76e6ffc8f35c9d279f23cbdc7b9  zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "a0dfd55ac1f36302c69c15cbc92c318a4f9c0acd  zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "a52c9984f628d775cb160070ce6e04ff938356d4  zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "7af2bb82b23f49efa0302cb68c613098f9b93758  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "a52c9984f628d775cb160070ce6e04ff938356d4  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "445580d57a9220fdd8880f5c40d1ceec0c812469  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "20991d3c72445eac24b852f9cd4152cbaefdbca0  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "e224626e2ad532562d39ec74aca3b40fc51995a2  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "436d621a8df7c3f174f4e438ef166ca46ad0499c  zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "1b09cdc8f89a5cde1c3b44b1d2d2cdf6fd54dd03  zr2096_6_dedup.sorted.bam\n",
      "388521846130ec1b73ee91cedb951d6f7c97ff0e  zr2096_6_dedup.sorted.bam.bai\n",
      "3d8484c149e4f81ae95f45e9a6ccb01b379a7f4e  zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "2e4100ea9578d13bb33f1b9af6ca7f221df3506c  zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "2fa21c80569bae410b1c062f1ad3f38cf1749e4b  zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "b561a535e06e87ead3c92de9b187630c5c0d3629  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "2fa21c80569bae410b1c062f1ad3f38cf1749e4b  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "03768e39a47c31c31d0e3079c514a6fd3b114279  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "adb44e06981bd15def77febac7e7d3e5d11bd5a0  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "3891b43c5b8b5a5f690e9ab1040a369de2e23b11  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "69b065f05205e32e0f81b67d05a043ab5bc0213b  zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "0078da6b38ef66653138b7692687542acbda2871  zr2096_7_dedup.sorted.bam\n",
      "d578ac2aff640c72ed7f83be74e488ac74d7ec15  zr2096_7_dedup.sorted.bam.bai\n",
      "2614f92c526939940096931deccf261e01f7e94f  zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "07c7907a6d3b00638db1d7ca967e248ec761f971  zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "97a7c347c33139bf559c777a825b66aba405b658  zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "584e02a5a14a8716fecfa1dee55d3edfe2adb128  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "97a7c347c33139bf559c777a825b66aba405b658  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "f275bc798ef6fae30745692de70ca9dbe2be5a8c  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "a31cdd45e87a3e13be2091e4cd3a395f9b327bf7  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "219a0aa7b7cbb19d1f7c7147756c46b727a9dfa9  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "ac25012143e97bd710804896cf15694285c90bcb  zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "d40cfb9c8da4bb6d44780970ca6d337a12bee12a  zr2096_8_dedup.sorted.bam\n",
      "5cd8716edcffef52177241d48c78563d3b37089f  zr2096_8_dedup.sorted.bam.bai\n",
      "9f8f93aa3aa56a5fe1308ef675c613d8c8e435b5  zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "264f6cac2c9693b8e1658c2bf70828ec0fc08416  zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "94f03bbae2996bf1afb133e4e9ceef1d77087b78  zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "641625b9d247040d56eb7b104b006686000ada9f  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "94f03bbae2996bf1afb133e4e9ceef1d77087b78  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "dafd209eb2fa44f4a8f2bb2afa84b1724f00c0b1  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "037563d093117765b33d40d98c5a1901c057129c  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "6580d3538d4b46aa2275590146966573122a0376  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "6e13c5e36f9dfe0b6a8be255914f8990d2f92c3f  zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n",
      "2acfa1288195015469b79a7eb7d4da45c4c35e99  zr2096_9_dedup.sorted.bam\n",
      "ca1528fe096fb5001e729a7715f9f13c7081588e  zr2096_9_dedup.sorted.bam.bai\n",
      "edd39b22c5024738aea9df2afb2a1402c2771878  zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.html\n",
      "0793a1518e1fdd5f1f45af4044e7f9c4f8ca5f52  zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "709450e906de22fd08f979cd8a0a953e0cedd178  zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "47aa8b76c381586399364f32887aebc96a08eb0c  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
      "709450e906de22fd08f979cd8a0a953e0cedd178  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "23ca49269399abf3856c145d735c3c00088fff92  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph\n",
      "6e729253707f72b038939f3ef69559dde17a65f2  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "b5b38b9a896f47fbe4bb12a76f524de37927f79d  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "e99a2df03b3916e96b3188cdb0c76e3beb7ce302  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt\n",
      "6d7b3ad6d62427c1902f881fc73e799210e8b540  zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt\n"
     ]
    }
   ],
   "source": [
    "! cat 2018-10-03-L012-Output-checksum.sha"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I'll move the files over to gannet manually."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
